<!DOCTYPE html>
<!-- saved from url=(0076)https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559 -->
<html lang="en" class="js gr__bmcbioinformatics_biomedcentral_com" style="--font-family-sans:Europa; --font-family-serif:Noto Serif;"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <title>WGCNA: an R package for weighted correlation network analysis | BMC Bioinformatics | Full Text</title>

        

    <meta name="citation_abstract" content="Correlation networks are increasingly being used in bioinformatics applications. For example, weighted gene co-expression network analysis is a systems biology method for describing the correlation patterns among genes across microarray samples. Weighted correlation network analysis (WGCNA) can be used for finding clusters (modules) of highly correlated genes, for summarizing such clusters using the module eigengene or an intramodular hub gene, for relating modules to one another and to external sample traits (using eigengene network methodology), and for calculating module membership measures. Correlation networks facilitate network based gene screening methods that can be used to identify candidate biomarkers or therapeutic targets. These methods have been successfully applied in various biological contexts, e.g. cancer, mouse genetics, yeast genetics, and analysis of brain imaging data. While parts of the correlation network methodology have been described in separate publications, there is a need to provide a user-friendly, comprehensive, and consistent software implementation and an accompanying tutorial. The WGCNA R software package is a comprehensive collection of R functions for performing various aspects of weighted correlation network analysis. The package includes functions for network construction, module detection, gene selection, calculations of topological properties, data simulation, visualization, and interfacing with external software. Along with the R package we also present R software tutorials. While the methods development was motivated by gene expression data, the underlying data mining approach can be applied to a variety of different settings. The WGCNA package provides R functions for weighted correlation network analysis, e.g. co-expression network analysis of gene expression data. The R package along with its source code and additional material are freely available at 
                    http://www.genetics.ucla.edu/labs/horvath/CoexpressionNetwork/Rpackages/WGCNA
                    
                  .">

    <meta name="description" content="Correlation networks are increasingly being used in bioinformatics applications. For example, weighted gene co-expression network analysis is a systems biology method for describing the correlation patterns among genes across microarray samples. Weighted correlation network analysis (WGCNA) can be used for finding clusters (modules) of highly correlated genes, for summarizing such clusters using the module eigengene or an intramodular hub gene, for relating modules to one another and to external sample traits (using eigengene network methodology), and for calculating module membership measures. Correlation networks facilitate network based gene screening methods that can be used to identify candidate biomarkers or therapeutic targets. These methods have been successfully applied in various biological contexts, e.g. cancer, mouse genetics, yeast genetics, and analysis of brain imaging data. While parts of the correlation network methodology have been described in separate publications, there is a need to provide a user-friendly, comprehensive, and consistent software implementation and an accompanying tutorial. The WGCNA R software package is a comprehensive collection of R functions for performing various aspects of weighted correlation network analysis. The package includes functions for network construction, module detection, gene selection, calculations of topological properties, data simulation, visualization, and interfacing with external software. Along with the R package we also present R software tutorials. While the methods development was motivated by gene expression data, the underlying data mining approach can be applied to a variety of different settings. The WGCNA package provides R functions for weighted correlation network analysis, e.g. co-expression network analysis of gene expression data. The R package along with its source code and additional material are freely available at 
                    http://www.genetics.ucla.edu/labs/horvath/CoexpressionNetwork/Rpackages/WGCNA
                    
                  .">

    <meta name="journal_id" content="12859">

    <meta name="dc.title" content="WGCNA: an R package for weighted correlation network analysis">

    <meta name="dc.source" content="BMC Bioinformatics 2008 9:1">

    <meta name="dc.format" content="text/html">

    <meta name="dc.publisher" content="BioMed Central">

    <meta name="dc.date" content="2008-12-29">

    <meta name="dc.type" content="OriginalPaper">

    <meta name="dc.language" content="En">

    <meta name="dc.copyright" content="2008 Langfelder and Horvath; licensee BioMed Central Ltd.">

    <meta name="dc.rights" content="http://creativecommons.org/licenses/by/2.0">

    <meta name="dc.rightsAgent" content="reprints@biomedcentral.com">

    <meta name="dc.description" content="Correlation networks are increasingly being used in bioinformatics applications. For example, weighted gene co-expression network analysis is a systems biology method for describing the correlation patterns among genes across microarray samples. Weighted correlation network analysis (WGCNA) can be used for finding clusters (modules) of highly correlated genes, for summarizing such clusters using the module eigengene or an intramodular hub gene, for relating modules to one another and to external sample traits (using eigengene network methodology), and for calculating module membership measures. Correlation networks facilitate network based gene screening methods that can be used to identify candidate biomarkers or therapeutic targets. These methods have been successfully applied in various biological contexts, e.g. cancer, mouse genetics, yeast genetics, and analysis of brain imaging data. While parts of the correlation network methodology have been described in separate publications, there is a need to provide a user-friendly, comprehensive, and consistent software implementation and an accompanying tutorial. The WGCNA R software package is a comprehensive collection of R functions for performing various aspects of weighted correlation network analysis. The package includes functions for network construction, module detection, gene selection, calculations of topological properties, data simulation, visualization, and interfacing with external software. Along with the R package we also present R software tutorials. While the methods development was motivated by gene expression data, the underlying data mining approach can be applied to a variety of different settings. The WGCNA package provides R functions for weighted correlation network analysis, e.g. co-expression network analysis of gene expression data. The R package along with its source code and additional material are freely available at 
                    http://www.genetics.ucla.edu/labs/horvath/CoexpressionNetwork/Rpackages/WGCNA
                    
                  .">

    <meta name="prism.issn" content="1471-2105">

    <meta name="prism.publicationName" content="BMC Bioinformatics">

    <meta name="prism.publicationDate" content="2008-12-29">

    <meta name="prism.volume" content="9">

    <meta name="prism.number" content="1">

    <meta name="prism.section" content="OriginalPaper">

    <meta name="prism.startingPage" content="559">

    <meta name="prism.copyright" content="2008 Langfelder and Horvath; licensee BioMed Central Ltd.">

    <meta name="prism.rightsAgent" content="reprints@biomedcentral.com">

    <meta name="prism.url" content="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559">

    <meta name="prism.doi" content="doi:10.1186/1471-2105-9-559">

    <meta name="citation_pdf_url" content="https://bmcbioinformatics.biomedcentral.com/track/pdf/10.1186/1471-2105-9-559">

    <meta name="citation_fulltext_html_url" content="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559">

    <meta name="citation_journal_title" content="BMC Bioinformatics">

    <meta name="citation_publisher" content="BioMed Central">

    <meta name="citation_issn" content="1471-2105">

    <meta name="citation_title" content="WGCNA: an R package for weighted correlation network analysis">

    <meta name="citation_volume" content="9">

    <meta name="citation_issue" content="1">

    <meta name="citation_publication_date" content="2008/12">

    <meta name="citation_online_date" content="2008/12/29">

    <meta name="citation_firstpage" content="559">

    <meta name="citation_article_type" content="Software">

    <meta name="citation_fulltext_world_readable" content="">

    <meta name="dc.identifier" content="doi:10.1186/1471-2105-9-559">

    <meta name="DOI" content="10.1186/1471-2105-9-559">

    <meta name="citation_doi" content="10.1186/1471-2105-9-559">

    <meta name="dc.creator" content="Peter Langfelder">

    <meta name="dc.creator" content="Steve Horvath">

    <meta name="dc.subject" content="Bioinformatics">

    <meta name="dc.subject" content="Microarrays">

    <meta name="dc.subject" content="Computational Biology/Bioinformatics">

    <meta name="dc.subject" content="Computer Appl. in Life Sciences">

    <meta name="dc.subject" content="Combinatorial Libraries">

    <meta name="dc.subject" content="Algorithms">

    <meta name="citation_reference" content="citation_journal_title=Metron; citation_title=On the &#39;probable error&#39; of a coefficient of correlation deduced from a small sample; citation_author=RA Fisher; citation_volume=1; citation_publication_date=1915; citation_pages=1-32; citation_id=CR1">

    <meta name="citation_reference" content="citation_journal_title=Proc Natl Acad Sci USA; citation_title=Transitive Functional Annotation by Shortest-path Analysis of Gene Expression Data; citation_author=X Zhou, MC Kao, W Wong; citation_volume=99; citation_issue=20; citation_publication_date=2002; citation_pages=12783-12788; citation_doi=; citation_id=CR2">

    <meta name="citation_reference" content="citation_journal_title=BMC Bioinformatics; citation_title=Automated modelling of signal transduction networks; citation_author=M Steffen, A Petti, J Aach, P D&#39;haeseleer, G Church; citation_volume=3; citation_publication_date=2002; citation_pages=34; citation_doi=; citation_id=CR3">

    <meta name="citation_reference" content="citation_journal_title=Science; citation_title=A Gene-Coexpression Network for Global Discovery of Conserved Genetic Modules; citation_author=JM Stuart, E Segal, D Koller, SK Kim; citation_volume=302; citation_issue=5643; citation_publication_date=2003; citation_pages=249-255; citation_doi=; citation_id=CR4">

    <meta name="citation_reference" content="citation_journal_title=Stat Appl Genet Mol Biol; citation_title=A General Framework for Weighted Gene Co-expression Network Analysis; citation_author=B Zhang, S Horvath; citation_volume=4; citation_publication_date=2005; citation_pages=Article 17; citation_id=CR5">

    <meta name="citation_reference" content="citation_journal_title=Bioinformatics; citation_title=Network structures and algorithms in Bioconductor; citation_author=VJ Carey, J Gentry, E Whalen, R Gentleman; citation_volume=21; citation_publication_date=2005; citation_pages=135-136; citation_doi=/bth458; citation_id=CR6">

    <meta name="citation_reference" content="citation_journal_title=Bioinformatics; citation_title=An empirical Bayes approach to inferring large-scale gene association networks; citation_author=J Schaefer, K Strimmer; citation_volume=21; citation_issue=6; citation_publication_date=2005; citation_pages=754-764; citation_doi=/bti062; citation_id=CR7">

    <meta name="citation_reference" content="citation_journal_title=Bioinformatics; citation_title=A pattern recognition approach to infer time-lagged genetic interactions; citation_author=CL Chuang, CH Jen, CM Chen, GS Shieh; citation_volume=24; citation_issue=9; citation_publication_date=2008; citation_pages=1183-1190; citation_doi=/btn098; citation_id=CR8">

    <meta name="citation_reference" content="citation_journal_title=BMC Bioinformatics; citation_title=Modelling the network of cell cycle transcription factors in the yeast Saccharomyces cerevisiae; citation_author=S Cokus, S Rose, D Haynor, N Gronbech-Jensen, M Pellegrini; citation_volume=7; citation_publication_date=2006; citation_pages=381; citation_doi=; citation_id=CR9">

    <meta name="citation_reference" content="citation_journal_title=Proc Natl Acad Sci USA; citation_title=Analysis of Oncogenic Signaling Networks in Glioblastoma Identifies ASPM as a Novel Molecular Target; citation_author=S Horvath, B Zhang, M Carlson, K Lu, S Zhu, R Felciano, M Laurance, W Zhao, Q Shu, Y Lee, A Scheck, L Liau, H Wu, D Geschwind, P Febbo, H Kornblum, T Cloughesy, S Nelson, P Mischel; citation_volume=103; citation_issue=46; citation_publication_date=2006; citation_pages=17402-17407; citation_doi=; citation_id=CR10">

    <meta name="citation_reference" content="citation_title=Geometric interpretation of Gene Co-expression Network Analysis; citation_inbook_title=PLoS Computational Biology; citation_publication_date=2008; citation_id=CR11; citation_author=S Horvath; citation_author=J Dong">

    <meta name="citation_reference" content="citation_journal_title=BMC Systems Biology; citation_title=Eigengene networks for studying the relationships between co-expression modules; citation_author=P Langfelder, S Horvath; citation_volume=1; citation_publication_date=2007; citation_pages=54; citation_doi=; citation_id=CR12">

    <meta name="citation_reference" content="citation_title=Gene Connectivity, Function, and Sequence Conservation: Predictions from Modular Yeast Co-expression Networks; citation_inbook_title=BMC Genomics; citation_publication_date=2006; citation_id=CR13; citation_author=MR Carlson; citation_author=B Zhang; citation_author=Z Fang; citation_author=S Horvath; citation_author=PS Mishel; citation_author=SF Nelson">

    <meta name="citation_reference" content="citation_journal_title=PloS Genetics; citation_title=Integrating Genetics and Network Analysis to Characterize Genes Related to Mouse Weight; citation_author=A Ghazalpour, S Doss, B Zhang, C Plaisier, S Wang, E Schadt, A Thomas, T Drake, A Lusis, S Horvath; citation_volume=2; citation_issue=8; citation_publication_date=2006; citation_pages=e130; citation_doi=; citation_id=CR14">

    <meta name="citation_reference" content="citation_journal_title=Mammalian Genome; citation_title=Weighted Gene Co-expression Network Analysis Strategies Applied to Mouse Weight; citation_author=T Fuller, A Ghazalpour, J Aten, T Drake, A Lusis, S Horvath; citation_volume=6; citation_issue=18; citation_publication_date=2007; citation_pages=463-472; citation_doi=; citation_id=CR15">

    <meta name="citation_reference" content="citation_journal_title=Nature; citation_title=Genetics of gene expression and its effect on disease; citation_author=V Emilsson, G Thorleifsson, B Zhang, A Leonardson, F Zink, J Zhu, S Carlson, A Helgason, G Walters, S Gunnarsdottir, M Mouy, V Steinthorsdottir, G Eiriksdottir, G Bjornsdottir, I Reynisdottir, D Gudbjartsson, A Helgadottir, A Jonasdottir, A Jonasdottir, U Styrkarsdottir, S Gretarsdottir, K Magnusson, H Stefansson, R Fossdal, K Kristjansson, H Gislason, T Stefansson, B Leifsson, U Thorsteinsdottir, J Lamb, MJ Gulcher,  Reitman, A Kong, E Schadt, K Stefansson; citation_volume=452; citation_issue=7186; citation_publication_date=2008; citation_pages=423-8; citation_doi=; citation_id=CR16">

    <meta name="citation_reference" content="citation_title=Elucidating the Role of Gonadal Hormones in Sexually Dimorphic Gene Co-Expression Networks; citation_inbook_title=Endocrinology; citation_publication_date=2008; citation_id=CR17; citation_author=A van Nas; citation_author=D Guhathakurta; citation_author=S Wang; citation_author=S Yehya; citation_author=S Horvath; citation_author=B Zhang; citation_author=L Ingram Drake; citation_author=G Chaudhuri; citation_author=E Schadt; citation_author=T Drake; citation_author=A Arnold; citation_author=A Lusis">

    <meta name="citation_reference" content="citation_journal_title=Proc Natl Acad Sci USA; citation_title=Conservation and Evolution of Gene Co-expression Networks in Human and Chimpanzee Brains; citation_author=M Oldham, S Horvath, D Geschwind; citation_volume=103; citation_issue=47; citation_publication_date=2006; citation_pages=17973-17978; citation_doi=; citation_id=CR18">

    <meta name="citation_reference" content="citation_journal_title=J Neurosci; citation_title=A Systems Level Analysis of Transcriptional Changes in Alzheimer&#39;s Disease and Normal Aging; citation_author=JA Miller, MC Oldham, DH Geschwind; citation_volume=28; citation_issue=6; citation_publication_date=2008; citation_pages=1410-1420; citation_doi=; citation_id=CR19">

    <meta name="citation_reference" content="citation_journal_title=Nature Neuroscience; citation_title=Functional organization of the transcriptome in human brain; citation_author=MC Oldham, G Konopka, K Iwamoto, P Langfelder, T Kato, S Horvath, DH Geschwind; citation_volume=11; citation_issue=11; citation_publication_date=2008; citation_pages=1271-1282; citation_doi=; citation_id=CR20">

    <meta name="citation_reference" content="citation_journal_title=Genome Res; citation_title=A gene expression network model of type 2 diabetes links cell cycle regulation in islets with diabetes susceptibility; citation_author=MP Keller, Y Choi, P Wang, D Belt Davis, ME Rabaglia, AT Oler, DS Stapleton, C Argmann, KL Schueler, S Edwards, HA Steinberg, E Chaibub Neto, R Kleinhanz, S Turner, MK Hellerstein, EE Schadt, BS Yandell, C Kendziorski, AD Attie; citation_volume=18; citation_issue=5; citation_publication_date=2008; citation_pages=706-716; citation_doi=; citation_id=CR21">

    <meta name="citation_reference" content="citation_title=Integrated weighted gene co-expression network analysis with an application to chronic fatigue syndrome; citation_inbook_title=BMC Systems Biology; citation_publication_date=2008; citation_id=CR22; citation_author=A Presson; citation_author=E Sobel; citation_author=J Papp; citation_author=C Suarez; citation_author=T Whistler; citation_author=M Rajeevan; citation_author=S Vernon; citation_author=S Horvath">

    <meta name="citation_reference" content="citation_title=Connecting genes, coexpression modules, and molecular signatures to environmental stress phenotypes in plants; citation_inbook_title=BMC Systems Biology; citation_publication_date=2008; citation_id=CR23; citation_author=D Weston; citation_author=L Gunter; citation_author=A Rogers; citation_author=S Wullschleger">

    <meta name="citation_reference" content="citation_title=Introduction to Robust Estimation and Hypothesis Testing; citation_publication_date=1997; citation_id=CR24; citation_author=RR Wilcox">

    <meta name="citation_reference" content="citation_journal_title=BMC Bioinformatics; citation_title=Gene network interconnectedness and the generalized topological overlap measure; citation_author=A Yip, S Horvath; citation_volume=8; citation_publication_date=2007; citation_pages=22; citation_doi=; citation_id=CR25">

    <meta name="citation_reference" content="citation_journal_title=Science; citation_title=Hierarchical Organization of Modularity in Metabolic Networks; citation_author=E Ravasz, A Somera, D Mongru, Z Oltvai, A Barabási; citation_volume=297; citation_issue=5586; citation_publication_date=2002; citation_pages=1551-1555; citation_doi=; citation_id=CR26">

    <meta name="citation_reference" content="citation_journal_title=Bioinformatics; citation_title=Network Neighborhood Analysis With the Multi-node Topological Overlap Measure; citation_author=A Li, S Horvath; citation_volume=23; citation_issue=2; citation_publication_date=2007; citation_pages=222-231; citation_doi=/btl581; citation_id=CR27">

    <meta name="citation_reference" content="citation_title=Finding Groups in Data: An Introduction to Cluster Analysis; citation_publication_date=1990; citation_id=CR28; citation_author=L Kaufman; citation_author=P Rousseeuw; citation_publisher=John Wiley &amp; Sons, Inc">

    <meta name="citation_reference" content="citation_journal_title=Bioinformatics; citation_title=Defining clusters from a hierarchical cluster tree: the Dynamic Tree Cut package for R; citation_author=P Langfelder, B Zhang, S Horvath; citation_volume=24; citation_issue=5; citation_publication_date=2008; citation_pages=719-720; citation_doi=/btm563; citation_id=CR29">

    <meta name="citation_reference" content="citation_journal_title=Genome Biol; citation_title=A prediction-based resampling method for estimating the number of clusters in a dataset; citation_author=S Dudoit, J Fridlyand; citation_volume=3; citation_issue=7; citation_publication_date=2002; citation_pages=RESEARCH0036; citation_doi=; citation_id=CR30">

    <meta name="citation_reference" content="citation_title=Imputing Missing Data for Gene Expression Arrays; citation_inbook_title=Technical report, Stanford Statistics Department; citation_publication_date=1999; citation_id=CR31; citation_author=T Hastie; citation_author=R Tibshirani; citation_author=G Sherlock; citation_author=M Eisen; citation_author=P Brown; citation_author=D Botstein">

    <meta name="citation_reference" content="citation_journal_title=Bioinformatics; citation_title=Missing value estimation methods for DNA microarrays; citation_author=O Troyanskaya, M Cantor, G Sherlock, P Brown, T Hastie, R Tibshirani, D Botstein, RB Altman; citation_volume=17; citation_issue=6; citation_publication_date=2001; citation_pages=520-525; citation_doi=/17.6.520; citation_id=CR32">

    <meta name="citation_reference" content="citation_journal_title=BMC Systems Biology; citation_title=Understanding network concepts in modules; citation_author=J Dong, S Horvath; citation_volume=1; citation_publication_date=2007; citation_pages=24; citation_doi=; citation_id=CR33">

    <meta name="citation_reference" content="citation_journal_title=Nature; citation_title=Collective dynamics of &#39;small-world&#39; networks; citation_author=DJ Watts, SH Strogatz; citation_volume=393; citation_issue=6684; citation_publication_date=1998; citation_pages=440-2; citation_doi=; citation_id=CR34">

    <meta name="citation_reference" content="citation_title=Statistical methods for identifying differentially expressed genes in replicated cDNA microarray experiments; citation_inbook_title=Statistica Sinica; citation_publication_date=2002; citation_id=CR35; citation_author=S Dudoit; citation_author=Y Yang; citation_author=M Callow; citation_author=T Speed">

    <meta name="citation_reference" content="citation_title=fields: Tools for Spatial Data; citation_inbook_title=Tech. rep., National Center for Atmospheric Research, Boulder, CO; citation_publication_date=2007; citation_id=CR36">

    <meta name="citation_reference" content="citation_journal_title=Brief Bioinform; citation_title=VisANT: an integrative framework for networks in systems biology; citation_author=Z Hu, ES Snitkin, C DeLisi; citation_volume=9; citation_issue=4; citation_publication_date=2008; citation_pages=317-325; citation_doi=/bbn020; citation_id=CR37">

    <meta name="citation_reference" content="citation_journal_title=Genome Research; citation_title=Cytoscape: A Software Environment for Integrated Models of Biomolecular Interaction Networks; citation_author=P Shannon, A Markiel, O Ozier, NS Baliga, JT Wang, D Ramage, N Amin, B Schwikowski, T Ideker; citation_volume=13; citation_issue=11; citation_publication_date=2003; citation_pages=2498-2504; citation_doi=; citation_id=CR38">

    <meta name="citation_reference" content="citation_title=GOSim – an R-package for computation of information theoretic GO similarities between terms and gene products; citation_inbook_title=BMC Bioinformatics; citation_publication_date=2007; citation_id=CR39; citation_author=H Frohlich; citation_author=N Speer; citation_author=A Poustka; citation_author=T BeiSZbarth">

    <meta name="citation_reference" content="citation_journal_title=Genome Biol; citation_title=DAVID: Database for Annotation, Visualization, and Integrated Discovery; citation_author=G Dennis, B Sherman, D Hosack, J Yang, W Gao, H Lane, R Lempicki; citation_volume=4; citation_issue=5; citation_publication_date=2003; citation_pages=P3; citation_doi=; citation_id=CR40">

    <meta name="citation_reference" content="citation_journal_title=Nat Genet; citation_title=Gene Ontology: tool for the unification of biology; citation_author=M Ashburner, CA Ball, JA Blake, D Botstein, H Butler, JM Cherry, AP Davis, K Dolinski, SS Dwight, JT Eppig, MA Harris, DP Hill, L Issel-Tarver, A Kasarskis, S Lewis, JC Matese, JE Richardson, M Ringwald, GM Rubin, G Sherlock; citation_volume=25; citation_publication_date=2000; citation_pages=25-29; citation_doi=; citation_id=CR41">

    <meta name="citation_reference" content="citation_journal_title=Nucleic Acids Res; citation_title=WebGestalt: an integrated system for exploring gene sets in various biological contexts; citation_author=B Zhang, S Kirov, J Snoddy; citation_volume=33; citation_issue=Web Server issue; citation_publication_date=2005; citation_pages=W741-W748; citation_doi=/gki475; citation_id=CR42">

    <meta name="citation_reference" content="citation_journal_title=PLoS Genet; citation_title=Network-Based Analysis of Affected Biological Processes in Type 2 Diabetes Models; citation_author=M Liu, A Liberzon, SW Kong, WR Lai, PJ Park, IS Kohane, S Kasif; citation_volume=3; citation_issue=6; citation_publication_date=2007; citation_pages=e96; citation_doi=; citation_id=CR43">

    <meta name="citation_reference" content="citation_title=Unsupervised Multiple-Instance Learning for Functional Profiling of Genomic Data; citation_inbook_title=Machine Learning: ECML 2006; citation_publication_date=2006; citation_pages=186-197; citation_id=CR44; citation_author=C Henegar; citation_author=K Clement; citation_author=JD Zucker; citation_publisher=Springer Berlin">

    <meta name="citation_reference" content="citation_title=Bioinformatics and Computational Biology Solutions Using R and Bioconductor; citation_inbook_title=Book; citation_publication_date=2005; citation_id=CR45; citation_author=R Gentleman; citation_author=W Huber; citation_author=V Carey; citation_author=R Irizarry; citation_author=S Dudoit; citation_publisher=Springer-Verlag">

    <meta name="citation_reference" content="citation_title=From correlation to causation networks: a simple approximate learning algorithm and its application to high-dimensional plant gene expression data; citation_inbook_title=BMC Systems Biology; citation_publication_date=2007; citation_id=CR46; citation_author=R Opgen-Rhein; citation_author=K Strimmer">

    <meta name="citation_reference" content="citation_title=Using genetic markers to orient the edges in quantitative trait networks: The NEO software; citation_inbook_title=BMC Systems Biology; citation_publication_date=2008; citation_id=CR47; citation_author=J Aten; citation_author=T Fuller; citation_author=A Lusis; citation_author=S Horvath">

    <meta name="citation_reference" content="citation_journal_title=Genetics; citation_title=Inferring Causal Phenotype Networks From Segregating Populations; citation_author=E Chaibub Neto, CT Ferrara, AD Attie, BS Yandell; citation_volume=179; citation_issue=2; citation_publication_date=2008; citation_pages=1089-1100; citation_doi=; citation_id=CR48">

    <meta name="citation_author" content="Peter Langfelder">

    <meta name="citation_author_institution" content="Department of Human Genetics, University of California, Los Angeles, USA">

    <meta name="citation_author" content="Steve Horvath">

    <meta name="citation_author_institution" content="Department of Human Genetics and Department of Biostatistics, University of California, Los Angeles, USA">



        
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="format-detection" content="telephone=no">

        <link rel="shortcut icon" href="https://bmcbioinformatics.biomedcentral.com/static/images/favicons/bmc/favicon-9ff1bf1161.ico">
<link rel="apple-touch-icon" sizes="114x114" href="https://bmcbioinformatics.biomedcentral.com/static/images/favicons/bmc/app-icon-114x114-aa3514ee38.png">
<link rel="apple-touch-icon" sizes="144x144" href="https://bmcbioinformatics.biomedcentral.com/static/images/favicons/bmc/app-icon-144x144-67f5b60136.png">
<link rel="apple-touch-icon" sizes="180x180" href="https://bmcbioinformatics.biomedcentral.com/static/images/favicons/bmc/app-icon-180x180-230f795aee.png">
<link rel="icon" type="image/png" sizes="16x16" href="https://bmcbioinformatics.biomedcentral.com/static/images/favicons/bmc/favicon-16x16-25a0239eec.png">
<link rel="icon" type="image/png" sizes="32x32" href="https://bmcbioinformatics.biomedcentral.com/static/images/favicons/bmc/favicon-32x32-637712e609.png">
<link rel="icon" type="image/png" sizes="96x96" href="https://bmcbioinformatics.biomedcentral.com/static/images/favicons/bmc/favicon-96x96-9f5dc535c7.png">
<link rel="icon" type="image/png" sizes="194x194" href="https://bmcbioinformatics.biomedcentral.com/static/images/favicons/bmc/favicon-194x194-d2a2617de2.png">
<meta name="msapplication-TileColor" content="#e6e6e6">
<meta name="msapplication-TileImage" content="/static/images/favicons/bmc/app-icon-144x144-67f5b60136.png">
<meta name="theme-color" content="#1b3051">

        <script type="text/javascript" src="https://cdn.krxd.net/userdata/get?pub=bd339c69-af54-4a21-b4f1-654bcfcd83ca&amp;kxfp=82c410ba84f8addc29ea15bb0fcd47a5cda6b2e6&amp;technographics=1&amp;callback=Krux.ns.macmillan.kxjsonp_userdata"></script><script type="text/javascript" src="https://beacon.krxd.net/optout_check?callback=Krux.ns.macmillan.kxjsonp_optOutCheck"></script><script src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/osd.js.download"></script><script type="text/javascript" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/bd339c69-af54-4a21-b4f1-654bcfcd83ca(1)"></script><script type="text/javascript" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/bd339c69-af54-4a21-b4f1-654bcfcd83ca"></script><script src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/pubads_impl_rendering_2019041801.js.download"></script><script async="" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/controltag.js.9791902f4f5b02b08a4a694e7fe1b073"></script><script type="text/javascript" async="" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/MathJax.js.download"></script><script type="text/javascript" async="" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/analytics.js.download"></script><script type="text/javascript" async="" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/KDqylSLE.js.download"></script><script async="" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/gtm.js.download"></script><script>(function(H){H.className=H.className.replace(/\bno-js\b/,'js')})(document.documentElement)</script>
        <link rel="stylesheet" media="screen" href="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/basic-79235ff3ca.css">
<link rel="stylesheet" media="print" href="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/print-7680b1a80b.css">

<link rel="stylesheet" id="js-mustard" href="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/main-f06148b04e.css" media="only screen and (-webkit-min-device-pixel-ratio:0) and (min-color-index:0), (-ms-high-contrast: none), only all and (min--moz-device-pixel-ratio:0) and (min-resolution: 3e1dpcm)">


        

        <script>
            window.airbrakeErrors = [];
            window.onerror = function(error) {window.airbrakeErrors.push(error);}
        </script>

        
    <script type="text/javascript">
        config = {
            env: 'live',
            site: 'bmcbioinformatics.biomedcentral.com',
            siteWithPath: 'bmcbioinformatics.biomedcentral.com' + window.location.pathname,
            twitterHashtag: 'bmcbioinformatics',
            cmsPrefix: 'https://studio-cms.springernature.com/studio/',
            
            doi: '10.1186/1471-2105-9-559',
            figshareScriptUrl: 'https://widgets.figshare.com/static/figshare.js',
            hasFigshareInvoked: false,
            publisherBrand: 'BioMed Central',
            mustardcut: false
        };
    </script>

        <!-- SpringerLink's event tracker, as per: https://github.com/springernature/springerlink-event-tracker -->

<script type="text/javascript">
document.addEventListener('dataLayerCreated', function() {
    var script = document.createElement('script');
    script.id = 'springerlink-event-tracker';
    script.src = 'https://event-tracker.springernature.com/dist/eventTracker.js';
    script.onload = function () {
        var dL;
        var doi;
        var imprint;
        var bpids;
        var publisher;
        var pageType;
        var eventObject;

        for (var i = 0; i < window.dataLayer.length; i++) {
            if (window.dataLayer[i].event === "dataLayerCreated") {
                dL = window.dataLayer[i];
            }
        }

        if (!dL || !dL.content.article || !dL.content.article.doi) { return false;}

        doi = dL.content.article.doi;
        imprint = dL.content.contentInfo.imprint;
        bpids = dL.session.authentication.authenticationID;
        pageType = dL.page.category.pageType;

        if (pageType) {
            pageType = pageType.charAt(0).toUpperCase() + pageType.substring(1);
        }

        eventObject = {'content_type': pageType, 'doi': doi};

        if (imprint === 'BioMed Central') {
            publisher = 'BMC';
        }
        else if (imprint === 'SpringerOpen') {
            publisher = 'SpringerOpen';
        }
        else {
            throw new Error('Event Tracker Error: The publisher has an unexpected value');
        }

        if (bpids.length > 0) {
            eventObject['business_partner_ids'] = bpids;
        }

        new EventTracker({'platform': publisher}).sendEvent('display', eventObject);
    };
    document.head.appendChild(script);
});
</script>

        <!--Polyfills CustomEvent constructor in IE. Allows us to use events to manage race conditions in client side js-->
<script>
    (function () {
        if (typeof window.CustomEvent === "function") { return false; } // If IE polyfill not needed just return

        function CustomEvent ( event, params ) {
            params = params || { bubbles: false, cancelable: false, detail: undefined };
            var evt = document.createEvent( 'CustomEvent' );
            evt.initCustomEvent( event, params.bubbles, params.cancelable, params.detail );
            return evt;
        }

        CustomEvent.prototype = window.Event.prototype;

        window.CustomEvent = CustomEvent;
    })();
</script> 
        
    <script type="text/javascript">
        window.dataLayer = [{"content":{"article":{"doi":"10.1186/1471-2105-9-559","articleType":"Software","peerReviewType":"Closed","supplement":null,"keywords":"Correlation Network;Module Membership;Module Eigengene;Brown Module;Hierarchical Cluster Dendrogram"},"contentInfo":{"imprint":"BioMed Central","title":"WGCNA: an R package for weighted correlation network analysis","publishedAt":1230508800000,"publishedAtDate":"2008-12-29","author":["Peter Langfelder","Steve Horvath"],"collection":[]},"attributes":{"deliveryPlatform":"oscar","template":"rebrand","cms":null,"copyright":{"creativeCommonsType":null,"openAccess":false},"environment":"live"},"journal":{"siteKey":"bmcbioinformatics.biomedcentral.com","volume":"9","issue":"1","title":"BMC Bioinformatics","type":"SERIES","journalID":12859,"gaCode":"UA-62065396-3","section":[]},"category":{"pmc":{"primarySubject":"Life Sciences"},"contentType":"Software","publishingSegment":"BMC-series Biology"}},"session":{"authentication":{"authenticationID":[]}},"version":"1.0.0","page":{"category":{"pageType":"article"}},"event":"dataLayerCreated"}];
        var event = new CustomEvent('dataLayerCreated');
        document.dispatchEvent(event);
    </script><script id="springerlink-event-tracker" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/eventTracker.js.download"></script>


        <script>
            
            var mustardcutlink = document.getElementById('js-mustard');
            if (mustardcutlink && window.matchMedia && window.matchMedia(mustardcutlink.media).matches) {
                window.config.mustardcut = true;
            }
        </script>
        <script>
            (function(d){
                var config = window.config;
                if (config && config.mustardcut && config.publisherBrand && sessionStorage.fontsLoaded === 'true') {
                    if (window.config.publisherBrand === 'BioMed Central') {
                        d.style.setProperty('--font-family-sans', 'Europa');
                        d.style.setProperty('--font-family-serif', 'Noto Serif')
                    } else if (window.config.publisherBrand === 'SpringerOpen' || window.config.publisherBrand === 'Springer') {
                        d.style.setProperty('--font-family-sans', 'Source Sans Pro');
                    }
                }
            })(document.documentElement);
        </script>
        <script src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/7d6f4b4f-77fb-43b9-b429-eb1e1eeac6aa.js.download" id="onetrust-bmc-control" charset="UTF-8"></script><script type="text/javascript" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/jquery.min.js.download"></script><link type="text/css" href="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/optanon.css" rel="stylesheet"><style>#optanon ul#optanon-menu li { background-color: #F8F8F8 !important }#optanon ul#optanon-menu li.menu-item-selected { background-color: #C5F6F9 !important }#optanon #optanon-popup-wrapper .optanon-white-button-middle { background-color: #1B3051 !important }.optanon-alert-box-wrapper .optanon-alert-box-button-middle { background-color: #1B3051 !important; border-color: #1B3051 !important; }#optanon #optanon-popup-wrapper .optanon-white-button-middle a { color: #ffffff !important }.optanon-alert-box-wrapper .optanon-alert-box-button-middle a { color: #ffffff !important }#optanon #optanon-popup-bottom { background-color: #FFFFFF !important }#optanon.modern #optanon-popup-top, #optanon.modern #optanon-popup-body-left-shading { background-color: #FFFFFF !important }.optanon-alert-box-wrapper { background-color:#EFF6FC !important }.optanon-alert-box-wrapper .optanon-alert-box-bg p { color:#333333 !important }</style>
        
    
        
    

        
    
        
            
                <!-- Google Tag Manager -->
                <script>
                    (function (w, d, s, l, i) {
                    w[l] = w[l] || [];
                    w[l].push({'gtm.start': new Date().getTime(), event: 'gtm.js'});
                    var f = d.getElementsByTagName(s)[0],
                            j = d.createElement(s),
                            dl = l != 'dataLayer' ? '&l=' + l : '';
                    j.async = true;
                    j.src = 'https://www.googletagmanager.com/gtm.js?id=' + i + dl;
                    f.parentNode.insertBefore(j, f);
                })(window, document, 'script', 'dataLayer', 'GTM-TDGJHK');
                </script>
                <!-- End Google Tag Manager -->
            
        
    


        
    
    <link rel="canonical" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559">
    

        
        
        <meta property="og:url" content="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559">
        <meta property="og:type" content="article">
        <meta property="og:site_name" content="BMC Bioinformatics">
        <meta property="og:title" content="WGCNA: an R package for weighted correlation network analysis">
        <meta property="og:description" content="Correlation networks are increasingly being used in bioinformatics applications. For example, weighted gene co-expression network analysis is a systems biology method for describing the correlation patterns among genes across microarray samples. Weighted correlation network analysis (WGCNA) can be used for finding clusters (modules) of highly correlated genes, for summarizing such clusters using the module eigengene or an intramodular hub gene, for relating modules to one another and to external sample traits (using eigengene network methodology), and for calculating module membership measures. Correlation networks facilitate network based gene screening methods that can be used to identify candidate biomarkers or therapeutic targets. These methods have been successfully applied in various biological contexts, e.g. cancer, mouse genetics, yeast genetics, and analysis of brain imaging data. While parts of the correlation network methodology have been described in separate publications, there is a need to provide a user-friendly, comprehensive, and consistent software implementation and an accompanying tutorial. The WGCNA R software package is a comprehensive collection of R functions for performing various aspects of weighted correlation network analysis. The package includes functions for network construction, module detection, gene selection, calculations of topological properties, data simulation, visualization, and interfacing with external software. Along with the R package we also present R software tutorials. While the methods development was motivated by gene expression data, the underlying data mining approach can be applied to a variety of different settings. The WGCNA package provides R functions for weighted correlation network analysis, e.g. co-expression network analysis of gene expression data. The R package along with its source code and additional material are freely available at http://www.genetics.ucla.edu/labs/horvath/CoexpressionNetwork/Rpackages/WGCNA .">
        <meta property="og:image" content="https://images.springer.com/sgw/journals/medium/12859.jpg">
    
    <script src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/figshare.js.download"></script><link rel="preload" href="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/f(2).txt" as="script"><script type="text/javascript" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/f(2).txt"></script><script src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/pubads_impl_2019041801.js.download" async=""></script><link rel="prefetch" href="https://tpc.googlesyndication.com/safeframe/1-0-33/html/container.html"><script type="text/javascript" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/jquery.min.js(1).download"></script><style type="text/css">.MathJax_Hover_Frame {border-radius: .25em; -webkit-border-radius: .25em; -moz-border-radius: .25em; -khtml-border-radius: .25em; box-shadow: 0px 0px 15px #83A; -webkit-box-shadow: 0px 0px 15px #83A; -moz-box-shadow: 0px 0px 15px #83A; -khtml-box-shadow: 0px 0px 15px #83A; border: 1px solid #A6D ! important; display: inline-block; position: absolute}
.MathJax_Menu_Button .MathJax_Hover_Arrow {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 4px; -webkit-border-radius: 4px; -moz-border-radius: 4px; -khtml-border-radius: 4px; font-family: 'Courier New',Courier; font-size: 9px; color: #F0F0F0}
.MathJax_Menu_Button .MathJax_Hover_Arrow span {display: block; background-color: #AAA; border: 1px solid; border-radius: 3px; line-height: 0; padding: 4px}
.MathJax_Hover_Arrow:hover {color: white!important; border: 2px solid #CCC!important}
.MathJax_Hover_Arrow:hover span {background-color: #CCC!important}
</style><style type="text/css">#MathJax_About {position: fixed; left: 50%; width: auto; text-align: center; border: 3px outset; padding: 1em 2em; background-color: #DDDDDD; color: black; cursor: default; font-family: message-box; font-size: 120%; font-style: normal; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; border-radius: 15px; -webkit-border-radius: 15px; -moz-border-radius: 15px; -khtml-border-radius: 15px; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
#MathJax_About.MathJax_MousePost {outline: none}
.MathJax_Menu {position: absolute; background-color: white; color: black; width: auto; padding: 2px; border: 1px solid #CCCCCC; margin: 0; cursor: default; font: menu; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
.MathJax_MenuItem {padding: 2px 2em; background: transparent}
.MathJax_MenuArrow {position: absolute; right: .5em; padding-top: .25em; color: #666666; font-size: .75em}
.MathJax_MenuActive .MathJax_MenuArrow {color: white}
.MathJax_MenuArrow.RTL {left: .5em; right: auto}
.MathJax_MenuCheck {position: absolute; left: .7em}
.MathJax_MenuCheck.RTL {right: .7em; left: auto}
.MathJax_MenuRadioCheck {position: absolute; left: 1em}
.MathJax_MenuRadioCheck.RTL {right: 1em; left: auto}
.MathJax_MenuLabel {padding: 2px 2em 4px 1.33em; font-style: italic}
.MathJax_MenuRule {border-top: 1px solid #CCCCCC; margin: 4px 1px 0px}
.MathJax_MenuDisabled {color: GrayText}
.MathJax_MenuActive {background-color: Highlight; color: HighlightText}
.MathJax_MenuDisabled:focus, .MathJax_MenuLabel:focus {background-color: #E8E8E8}
.MathJax_ContextMenu:focus {outline: none}
.MathJax_ContextMenu .MathJax_MenuItem:focus {outline: none}
#MathJax_AboutClose {top: .2em; right: .2em}
.MathJax_Menu .MathJax_MenuClose {top: -10px; left: -10px}
.MathJax_MenuClose {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; font-family: 'Courier New',Courier; font-size: 24px; color: #F0F0F0}
.MathJax_MenuClose span {display: block; background-color: #AAA; border: 1.5px solid; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; line-height: 0; padding: 8px 0 6px}
.MathJax_MenuClose:hover {color: white!important; border: 2px solid #CCC!important}
.MathJax_MenuClose:hover span {background-color: #CCC!important}
.MathJax_MenuClose:hover:focus {outline: none}
</style><style type="text/css">.MJX_Assistive_MathML {position: absolute!important; top: 0; left: 0; clip: rect(1px, 1px, 1px, 1px); padding: 1px 0 0 0!important; border: 0!important; height: 1px!important; width: 1px!important; overflow: hidden!important; display: block!important; -webkit-touch-callout: none; -webkit-user-select: none; -khtml-user-select: none; -moz-user-select: none; -ms-user-select: none; user-select: none}
.MJX_Assistive_MathML.MJX_Assistive_MathML_Block {width: 100%!important}
</style><style type="text/css">#MathJax_Zoom {position: absolute; background-color: #F0F0F0; overflow: auto; display: block; z-index: 301; padding: .5em; border: 1px solid black; margin: 0; font-weight: normal; font-style: normal; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; -webkit-box-sizing: content-box; -moz-box-sizing: content-box; box-sizing: content-box; box-shadow: 5px 5px 15px #AAAAAA; -webkit-box-shadow: 5px 5px 15px #AAAAAA; -moz-box-shadow: 5px 5px 15px #AAAAAA; -khtml-box-shadow: 5px 5px 15px #AAAAAA; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
#MathJax_ZoomOverlay {position: absolute; left: 0; top: 0; z-index: 300; display: inline-block; width: 100%; height: 100%; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}
#MathJax_ZoomFrame {position: relative; display: inline-block; height: 0; width: 0}
#MathJax_ZoomEventTrap {position: absolute; left: 0; top: 0; z-index: 302; display: inline-block; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}
</style><style type="text/css">.MathJax_Preview {color: #888}
#MathJax_Message {position: fixed; left: 1em; bottom: 1.5em; background-color: #E6E6E6; border: 1px solid #959595; margin: 0px; padding: 2px 8px; z-index: 102; color: black; font-size: 80%; width: auto; white-space: nowrap}
#MathJax_MSIE_Frame {position: absolute; top: 0; left: 0; width: 0px; z-index: 101; border: 0px; margin: 0px; padding: 0px}
.MathJax_Error {color: #CC0000; font-style: italic}
</style><style type="text/css">.MJXp-script {font-size: .8em}
.MJXp-right {-webkit-transform-origin: right; -moz-transform-origin: right; -ms-transform-origin: right; -o-transform-origin: right; transform-origin: right}
.MJXp-bold {font-weight: bold}
.MJXp-italic {font-style: italic}
.MJXp-scr {font-family: MathJax_Script,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-frak {font-family: MathJax_Fraktur,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-sf {font-family: MathJax_SansSerif,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-cal {font-family: MathJax_Caligraphic,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-mono {font-family: MathJax_Typewriter,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-largeop {font-size: 150%}
.MJXp-largeop.MJXp-int {vertical-align: -.2em}
.MJXp-math {display: inline-block; line-height: 1.2; text-indent: 0; font-family: 'Times New Roman',Times,STIXGeneral,serif; white-space: nowrap; border-collapse: collapse}
.MJXp-display {display: block; text-align: center; margin: 1em 0}
.MJXp-math span {display: inline-block}
.MJXp-box {display: block!important; text-align: center}
.MJXp-box:after {content: " "}
.MJXp-rule {display: block!important; margin-top: .1em}
.MJXp-char {display: block!important}
.MJXp-mo {margin: 0 .15em}
.MJXp-mfrac {margin: 0 .125em; vertical-align: .25em}
.MJXp-denom {display: inline-table!important; width: 100%}
.MJXp-denom > * {display: table-row!important}
.MJXp-surd {vertical-align: top}
.MJXp-surd > * {display: block!important}
.MJXp-script-box > *  {display: table!important; height: 50%}
.MJXp-script-box > * > * {display: table-cell!important; vertical-align: top}
.MJXp-script-box > *:last-child > * {vertical-align: bottom}
.MJXp-script-box > * > * > * {display: block!important}
.MJXp-mphantom {visibility: hidden}
.MJXp-munderover, .MJXp-munder {display: inline-table!important}
.MJXp-over {display: inline-block!important; text-align: center}
.MJXp-over > * {display: block!important}
.MJXp-munderover > *, .MJXp-munder > * {display: table-row!important}
.MJXp-mtable {vertical-align: .25em; margin: 0 .125em}
.MJXp-mtable > * {display: inline-table!important; vertical-align: middle}
.MJXp-mtr {display: table-row!important}
.MJXp-mtd {display: table-cell!important; text-align: center; padding: .5em 0 0 .5em}
.MJXp-mtr > .MJXp-mtd:first-child {padding-left: 0}
.MJXp-mtr:first-child > .MJXp-mtd {padding-top: 0}
.MJXp-mlabeledtr {display: table-row!important}
.MJXp-mlabeledtr > .MJXp-mtd:first-child {padding-left: 0}
.MJXp-mlabeledtr:first-child > .MJXp-mtd {padding-top: 0}
.MJXp-merror {background-color: #FFFF88; color: #CC0000; border: 1px solid #CC0000; padding: 1px 3px; font-style: normal; font-size: 90%}
.MJXp-scale0 {-webkit-transform: scaleX(.0); -moz-transform: scaleX(.0); -ms-transform: scaleX(.0); -o-transform: scaleX(.0); transform: scaleX(.0)}
.MJXp-scale1 {-webkit-transform: scaleX(.1); -moz-transform: scaleX(.1); -ms-transform: scaleX(.1); -o-transform: scaleX(.1); transform: scaleX(.1)}
.MJXp-scale2 {-webkit-transform: scaleX(.2); -moz-transform: scaleX(.2); -ms-transform: scaleX(.2); -o-transform: scaleX(.2); transform: scaleX(.2)}
.MJXp-scale3 {-webkit-transform: scaleX(.3); -moz-transform: scaleX(.3); -ms-transform: scaleX(.3); -o-transform: scaleX(.3); transform: scaleX(.3)}
.MJXp-scale4 {-webkit-transform: scaleX(.4); -moz-transform: scaleX(.4); -ms-transform: scaleX(.4); -o-transform: scaleX(.4); transform: scaleX(.4)}
.MJXp-scale5 {-webkit-transform: scaleX(.5); -moz-transform: scaleX(.5); -ms-transform: scaleX(.5); -o-transform: scaleX(.5); transform: scaleX(.5)}
.MJXp-scale6 {-webkit-transform: scaleX(.6); -moz-transform: scaleX(.6); -ms-transform: scaleX(.6); -o-transform: scaleX(.6); transform: scaleX(.6)}
.MJXp-scale7 {-webkit-transform: scaleX(.7); -moz-transform: scaleX(.7); -ms-transform: scaleX(.7); -o-transform: scaleX(.7); transform: scaleX(.7)}
.MJXp-scale8 {-webkit-transform: scaleX(.8); -moz-transform: scaleX(.8); -ms-transform: scaleX(.8); -o-transform: scaleX(.8); transform: scaleX(.8)}
.MJXp-scale9 {-webkit-transform: scaleX(.9); -moz-transform: scaleX(.9); -ms-transform: scaleX(.9); -o-transform: scaleX(.9); transform: scaleX(.9)}
.MathJax_PHTML .noError {vertical-align: ; font-size: 90%; text-align: left; color: black; padding: 1px 3px; border: 1px solid}
</style><style type="text/css">.MathJax_Display {text-align: center; margin: 0; position: relative; display: block!important; text-indent: 0; max-width: none; max-height: none; min-width: 0; min-height: 0; width: 100%}
.MathJax .merror {background-color: #FFFF88; color: #CC0000; border: 1px solid #CC0000; padding: 1px 3px; font-style: normal; font-size: 90%}
.MathJax .MJX-monospace {font-family: monospace}
.MathJax .MJX-sans-serif {font-family: sans-serif}
#MathJax_Tooltip {background-color: InfoBackground; color: InfoText; border: 1px solid black; box-shadow: 2px 2px 5px #AAAAAA; -webkit-box-shadow: 2px 2px 5px #AAAAAA; -moz-box-shadow: 2px 2px 5px #AAAAAA; -khtml-box-shadow: 2px 2px 5px #AAAAAA; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true'); padding: 3px 4px; z-index: 401; position: absolute; left: 0; top: 0; width: auto; height: auto; display: none}
.MathJax {display: inline; font-style: normal; font-weight: normal; line-height: normal; font-size: 100%; font-size-adjust: none; text-indent: 0; text-align: left; text-transform: none; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; direction: ltr; max-width: none; max-height: none; min-width: 0; min-height: 0; border: 0; padding: 0; margin: 0}
.MathJax:focus, body :focus .MathJax {display: inline-table}
.MathJax.MathJax_FullWidth {text-align: center; display: table-cell!important; width: 10000em!important}
.MathJax img, .MathJax nobr, .MathJax a {border: 0; padding: 0; margin: 0; max-width: none; max-height: none; min-width: 0; min-height: 0; vertical-align: 0; line-height: normal; text-decoration: none}
img.MathJax_strut {border: 0!important; padding: 0!important; margin: 0!important; vertical-align: 0!important}
.MathJax span {display: inline; position: static; border: 0; padding: 0; margin: 0; vertical-align: 0; line-height: normal; text-decoration: none; box-sizing: content-box}
.MathJax nobr {white-space: nowrap!important}
.MathJax img {display: inline!important; float: none!important}
.MathJax * {transition: none; -webkit-transition: none; -moz-transition: none; -ms-transition: none; -o-transition: none}
.MathJax_Processing {visibility: hidden; position: fixed; width: 0; height: 0; overflow: hidden}
.MathJax_Processed {display: none!important}
.MathJax_test {font-style: normal; font-weight: normal; font-size: 100%; font-size-adjust: none; text-indent: 0; text-transform: none; letter-spacing: normal; word-spacing: normal; overflow: hidden; height: 1px}
.MathJax_test.mjx-test-display {display: table!important}
.MathJax_test.mjx-test-inline {display: inline!important; margin-right: -1px}
.MathJax_test.mjx-test-default {display: block!important; clear: both}
.MathJax_ex_box {display: inline-block!important; position: absolute; overflow: hidden; min-height: 0; max-height: none; padding: 0; border: 0; margin: 0; width: 1px; height: 60ex}
.MathJax_em_box {display: inline-block!important; position: absolute; overflow: hidden; min-height: 0; max-height: none; padding: 0; border: 0; margin: 0; width: 1px; height: 60em}
.mjx-test-inline .MathJax_left_box {display: inline-block; width: 0; float: left}
.mjx-test-inline .MathJax_right_box {display: inline-block; width: 0; float: right}
.mjx-test-display .MathJax_right_box {display: table-cell!important; width: 10000em!important; min-width: 0; max-width: none; padding: 0; border: 0; margin: 0}
.MathJax .MathJax_HitBox {cursor: text; background: white; opacity: 0; filter: alpha(opacity=0)}
.MathJax .MathJax_HitBox * {filter: none; opacity: 1; background: transparent}
#MathJax_Tooltip * {filter: none; opacity: 1; background: transparent}
@font-face {font-family: MathJax_Main; src: url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/woff/MathJax_Main-Regular.woff?V=2.7.5') format('woff'), url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/otf/MathJax_Main-Regular.otf?V=2.7.5') format('opentype')}
@font-face {font-family: MathJax_Main-bold; src: url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/woff/MathJax_Main-Bold.woff?V=2.7.5') format('woff'), url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/otf/MathJax_Main-Bold.otf?V=2.7.5') format('opentype')}
@font-face {font-family: MathJax_Main-italic; src: url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/woff/MathJax_Main-Italic.woff?V=2.7.5') format('woff'), url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/otf/MathJax_Main-Italic.otf?V=2.7.5') format('opentype')}
@font-face {font-family: MathJax_Math-italic; src: url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/woff/MathJax_Math-Italic.woff?V=2.7.5') format('woff'), url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/otf/MathJax_Math-Italic.otf?V=2.7.5') format('opentype')}
@font-face {font-family: MathJax_Caligraphic; src: url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/woff/MathJax_Caligraphic-Regular.woff?V=2.7.5') format('woff'), url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/otf/MathJax_Caligraphic-Regular.otf?V=2.7.5') format('opentype')}
@font-face {font-family: MathJax_Size1; src: url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/woff/MathJax_Size1-Regular.woff?V=2.7.5') format('woff'), url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/otf/MathJax_Size1-Regular.otf?V=2.7.5') format('opentype')}
@font-face {font-family: MathJax_Size2; src: url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/woff/MathJax_Size2-Regular.woff?V=2.7.5') format('woff'), url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/otf/MathJax_Size2-Regular.otf?V=2.7.5') format('opentype')}
@font-face {font-family: MathJax_Size3; src: url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/woff/MathJax_Size3-Regular.woff?V=2.7.5') format('woff'), url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/otf/MathJax_Size3-Regular.otf?V=2.7.5') format('opentype')}
@font-face {font-family: MathJax_Size4; src: url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/woff/MathJax_Size4-Regular.woff?V=2.7.5') format('woff'), url('https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/fonts/HTML-CSS/TeX/otf/MathJax_Size4-Regular.otf?V=2.7.5') format('opentype')}
#MathJax_Message {margin: 0}
.MathJax .noError {vertical-align: ; font-size: 90%; text-align: left; color: black; padding: 1px 3px; border: 1px solid}
</style></head>

    <body class="journal journal-fulltext" data-gr-c-s-loaded="true"><div style="visibility: hidden; overflow: hidden; position: absolute; top: 0px; height: 1px; width: auto; padding: 0px; border: 0px; margin: 0px; text-align: left; text-indent: 0px; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal;"><div id="MathJax_Hidden"></div></div><div id="MathJax_Message" style="display: none;"></div>
        
    
        
            
                <!-- Google Tag Manager (noscript) -->
                <noscript>
                    <iframe src="https://www.googletagmanager.com/ns.html?id=GTM-TDGJHK"
                            height="0" width="0" style="display:none;visibility:hidden"></iframe>
                </noscript>
                <!-- End Google Tag Manager (noscript) -->
            
        
    

        <div class="u-visually-hidden" aria-hidden="true">
    
    <!--?xml version="1.0" encoding="UTF-8"?--><svg xmlns="http://www.w3.org/2000/svg"><defs><filter id="a" filterUnits="userSpaceOnUse" x="0" y="0" width="60.643" height="31.01"><fecolormatrix values="1 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 1 0"></fecolormatrix></filter></defs><symbol id="icon-arrow-down" viewBox="0 0 16 8"><path d="M8 6l6-6h2L8 8 0 0h2l6 6"></path></symbol><symbol id="icon-arrow-left-bullet" viewBox="0 0 8 16"><path d="M3 8l5 5v3L0 8l8-8v3L3 8z"></path></symbol><symbol id="icon-arrow-right-bullet" viewBox="0 0 8 16"><path d="M5 8L0 3V0l8 8-8 8v-3l5-5z"></path></symbol><symbol id="icon-arrow-right" viewBox="0 0 32 64"><path d="M0 0v64l32-32z"></path></symbol><symbol id="icon-close" viewBox="0 0 23.998 23.999"><path d="M20.998 0l-8.999 9L3 0 0 3l8.999 9L0 20.999l3 3L11.999 15l8.999 8.999 3-3L14.999 12l8.999-9z"></path></symbol><symbol id="icon-community" viewBox="0 0 60.643 31.01"><mask maskUnits="userSpaceOnUse" x="0" y="0" width="60.643" height="31.01" id="b"><path fill="#FFF" d="M0 31.01h60.643V0H0z" filter="url(#a)"></path></mask><path mask="url(#b)" d="M60.643 25.497L50.005 14.86a7.925 7.925 0 1 0-7.669 0l-6.847 6.847-1.334-1.334a7.925 7.925 0 0 0-3.834-14.861 7.925 7.925 0 0 0-3.834 14.861l-1.334 1.334-6.847-6.847a7.925 7.925 0 1 0-7.669 0L0 25.497h21.362L15.85 31.01h28.943l-5.514-5.513h21.364z"></path></symbol><symbol id="icon-download" viewBox="-301 390 9 14"><path d="M-301 395.6l4.5 5.1 4.5-5.1h-3V390h-3v5.6h-3zm0 6.5h9v1.9h-9z"></path></symbol><symbol id="icon-ext-link" viewBox="0 0 16 16"><path fill="#676767" d="M12.9 16H3.1C1.4 16 0 14.6 0 12.9V3.2C0 1.4 1.4 0 3.1 0h3.7v1H3.1C2 1 1 2 1 3.2v9.7C1 14 2 15 3.1 15h9.7c1.2 0 2.1-1 2.1-2.1V8.7h1v4.2c.1 1.7-1.3 3.1-3 3.1z"></path><path fill="#676767" d="M12.8 2.5l.7.7-9 8.9-.7-.7 9-8.9z"></path><path fill="#676767" d="M9.7 0L16 6.2V0z"></path></symbol><symbol id="icon-info-bordered" viewBox="470.812 270.868 18 18"><path d="M479.812 270.868c-4.972 0-9 4.029-9 9s4.028 9 9 9c4.971 0 9-4.029 9-9s-4.029-9-9-9zm0 16.875a7.875 7.875 0 0 1-7.875-7.875 7.875 7.875 0 1 1 15.75 0 7.875 7.875 0 0 1-7.875 7.875z"></path><path d="M479.284 279.586c.089-.238-.024-.361-.13-.361-.489 0-1.123 1.16-1.359 1.16-.093 0-.173-.092-.173-.174 0-.238.581-.801.751-.971.526-.506 1.217-.895 1.979-.895.567 0 1.174.346.703 1.639l-.952 2.604c-.079.199-.224.531-.224.746 0 .092.054.186.159.186.395 0 1.121-1.135 1.304-1.135.067 0 .158.086.158.199 0 .385-1.542 2.043-2.874 2.043-.477 0-.804-.225-.804-.732 0-.641.447-1.734.538-1.963l.924-2.346zm.727-3.41c0-.586.498-1.066 1.082-1.066.527 0 .91.357.91.906 0 .615-.501 1.068-1.098 1.068-.541-.002-.894-.361-.894-.908z"></path></symbol><symbol id="icon-opr" viewBox="0 0 18 18"><path d="M.9 2.5c-.2-.2-.4-.3-.6-.2-.3.2-.3.4-.3.6 0 .8 0 1.7.1 2.5 0 .5.3.7.7.6.9 0 1.7-.2 2.5-.3h.2c.5-.2.6-.5.3-.8-.2-.2-.4-.4-.6-.5-.1-.1-.2-.2-.4-.3.5-.6 1-1.1 1.7-1.5 4.6-3 10.9-.5 12 4.9.9 4.4-2.2 8.6-6.7 9.2-3.7.5-7.2-1.6-8.4-5.1 0-.1-.1-.2-.1-.3-.1-.3-.5-.5-.8-.4-.3.1-.5.4-.4.8.4 1.3 1.1 2.4 2 3.4 1.4 1.5 3.1 2.4 5.1 2.8 3 .5 5.6-.2 7.8-2.2 2.1-1.9 3-4.2 3-6.9-.1-4-3.1-7.6-7.1-8.4-3.3-.9-6.2 0-8.6 2.3-.2.1-.3.3-.5.5-.3-.2-.6-.5-.9-.7z"></path><path d="M13 4.7c-2.6 1.6-4.5 3.6-5.3 4.6L5.6 7.7l-.9.7L8.3 12c.6-1.6 2.6-4.6 4.9-6.8l-.2-.5z"></path></symbol><symbol id="icon-quotation" viewBox="0 0.333 55.932 35.619"><path d="M43.738.333L31.545 12.207v23.745h24.387V12.207H43.738zm-31.545 0L0 12.207v23.745h24.386V12.207H12.193z"></path></symbol><symbol id="icon-remove" viewBox="-296 388 18 18"><path d="M-291.7 396.1h9v2h-9z"></path><path d="M-287 405.5c-4.7 0-8.5-3.8-8.5-8.5s3.8-8.5 8.5-8.5 8.5 3.8 8.5 8.5-3.8 8.5-8.5 8.5zm0-16c-4.1 0-7.5 3.4-7.5 7.5s3.4 7.5 7.5 7.5 7.5-3.4 7.5-7.5-3.4-7.5-7.5-7.5z"></path></symbol><symbol id="icon-ribbon" viewBox="0 0 36 72"><path d="M0 0v72h.053L18 54.054 35.945 72H36V0H0zm25.8 11.4l3.001 3L15 28.2l-7.801-7.8 3-3L15 22.201 25.8 11.4z"></path></symbol><symbol id="icon-rss" viewBox="0 0 24 24"><g fill="#F26822"><ellipse cx="3.305" cy="20.702" rx="3.306" ry="3.298"></ellipse><path d="M15.978 24h-4.684c0-6.224-5.057-11.27-11.294-11.27V8.058c8.824 0 15.978 7.137 15.978 15.942z"></path><path d="M19.2 23.95C19.2 13.366 10.604 4.79 0 4.79V0c13.255 0 24 10.723 24 23.95h-4.8z"></path></g></symbol><symbol id="icon-search" viewBox="0 0 16 16"><path d="M6.6 10.6c2.2 0 4-1.7 4-4s-1.7-4-4-4-4 1.7-4 4 1.8 4 4 4zm4 1.4l1.5-1.5 3.7 3.6c.4.5.4 1.2 0 1.6-.4.4-1.1.4-1.6 0L10.6 12zm-4 1.2C2.9 13.2 0 10.3 0 6.6S2.9 0 6.6 0s6.6 2.9 6.6 6.6-2.9 6.6-6.6 6.6z"></path></symbol><symbol id="icon-speech-bubble" viewBox="0 0 40 54"><path d="M0 0v40h6.667v13.334L20 40h20V0z"></path></symbol><symbol id="icon-error" viewBox="2.002 0 14 14"><path d="M16.002 11.949L13.951 14 9.002 9.051 4.053 14l-2.051-2.051L6.951 7 2.002 2.05 4.053 0l4.949 4.95L13.951 0l2.051 2.049L11.053 7l4.949 4.949z"></path></symbol><symbol id="icon-info" viewBox="5.502 0 7 14"><path d="M8.165 5.923c.146-.384-.04-.577-.209-.577-.783 0-1.798 1.856-2.177 1.856-.15 0-.277-.147-.277-.279 0-.383.933-1.282 1.203-1.557.846-.811 1.949-1.434 3.174-1.434.908 0 1.879.555 1.125 2.63l-1.526 4.167c-.126.32-.358.854-.358 1.198 0 .148.088.298.255.298.632 0 1.796-1.818 2.091-1.818.105 0 .252.134.252.321C11.717 11.345 9.245 14 7.11 14c-.763 0-1.289-.361-1.289-1.176 0-1.027.717-2.778.865-3.145l1.479-3.756zm1.144-4.211C9.309.771 10.111 0 11.045 0c.845 0 1.457.577 1.457 1.456 0 .983-.804 1.709-1.757 1.709-.866-.001-1.436-.578-1.436-1.453z"></path></symbol><symbol id="icon-success" viewBox="0 0 16 14"><path d="M15.592 0c-4.867 2.984-8.4 6.751-9.987 8.641L1.717 5.595 0 6.979l6.717 6.832c1.157-2.961 4.817-8.748 9.287-12.86L15.592 0z"></path></symbol><symbol id="icon-warning" viewBox="7.002 0 2.789 14"><path d="M7.002 11.211h2.789V14H7.002v-2.789zM7.002 0v3.436l.741 5.326h1.326l.723-5.326V0h-2.79z"></path></symbol><symbol id="icon-facebook-bordered" viewBox="463.812 263.868 32 32"><path d="M479.812 263.868c-8.837 0-16 7.163-16 16s7.163 16 16 16 16-7.163 16-16-7.163-16-16-16zm0 30c-7.732 0-14-6.269-14-14s6.268-14 14-14 14 6.269 14 14-6.267 14-14 14z"></path><path d="M483.025 280.48l.32-2.477h-2.453v-1.582c0-.715.199-1.207 1.227-1.207h1.311v-2.213a17.753 17.753 0 0 0-1.907-.098c-1.894 0-3.186 1.154-3.186 3.271V278h-2.142v2.477h2.142v6.354h2.557v-6.354l2.131.003z"></path></symbol><symbol id="icon-facebook" viewBox="0 0 24 24"><circle fill="#4D67A4" cx="12" cy="12" r="12"></circle><path fill="#FFF" d="M8.396 10.143h2.137V8.165s-.092-1.292.892-2.274c.979-.979 2.298-.887 4.177-.724v2.21h-1.387s-.586-.013-.861.303c-.271.315-.24.764-.24.875v1.588h2.41l-.311 2.458h-2.116v6.336h-2.56v-6.345H8.396v-2.449z"></path></symbol><symbol id="icon-linkedin" viewBox="0 0 24 24"><circle fill="#0178B5" cx="12" cy="12" r="12"></circle><g fill="#FFF"><circle cx="8.022" cy="8.043" r="1.256"></circle><path d="M6.929 10.246h2.17v6.967h-2.17zm3.533 6.967h2.157v-3.594s-.078-1.627 1.219-1.627c1.301 0 1.211 1.231 1.211 1.635v3.586h2.183v-3.775s.129-2.745-1.472-3.191c-1.605-.445-2.661.071-3.219.985v-.985h-2.079v6.966z"></path></g></symbol><symbol id="icon-reddit" viewBox="0 0 24 24"><circle fill="#BCBCBC" cx="12" cy="12" r="12"></circle><path fill="#FFF" d="M4.661 9.741c.941 0 1.703.761 1.703 1.704 0 .938-.762 1.705-1.703 1.705s-1.704-.767-1.704-1.705c0-.943.763-1.704 1.704-1.704zm13.844 0a1.704 1.704 0 1 1 .001 3.409 1.704 1.704 0 0 1-.001-3.409z"></path><path fill="#FFF" d="M11.736 8.732c4.285 0 7.762 2.283 7.762 5.104 0 2.812-3.477 5.1-7.762 5.1-4.288 0-7.762-2.285-7.762-5.1.001-2.82 3.474-5.104 7.762-5.104z"></path><path fill="#010101" d="M11.736 19.262c-4.461 0-8.088-2.437-8.088-5.426 0-2.994 3.626-5.431 8.088-5.431 4.457 0 8.087 2.437 8.087 5.431 0 2.989-3.63 5.426-8.087 5.426zm0-10.205c-4.104 0-7.438 2.145-7.438 4.779 0 2.633 3.334 4.773 7.438 4.773 4.098 0 7.437-2.142 7.437-4.773 0-2.636-3.339-4.779-7.437-4.779z"></path><circle fill="#FFF" cx="18.014" cy="5.765" r="1.385"></circle><path fill="#010101" d="M18.014 7.456a1.694 1.694 0 0 1-1.695-1.691c0-.934.764-1.694 1.695-1.694s1.693.762 1.693 1.694c0 .934-.762 1.691-1.693 1.691zm0-2.734c-.574 0-1.043.469-1.043 1.043s.469 1.041 1.043 1.041a1.042 1.042 0 0 0 0-2.084zM3.818 13.275a2.068 2.068 0 0 1-1.068-1.811 2.074 2.074 0 0 1 3.538-1.466l-.458.462a1.412 1.412 0 0 0-1.007-.42 1.424 1.424 0 0 0-.688 2.669l-.317.566zm15.866 0l-.316-.566a1.43 1.43 0 0 0 .73-1.245c0-.784-.635-1.424-1.42-1.424-.381 0-.736.149-1.008.42l-.459-.462a2.074 2.074 0 1 1 2.473 3.277z"></path><path fill="#F04B23" d="M9.165 11.604c.704 0 1.272.564 1.272 1.27a1.273 1.273 0 1 1-1.272-1.27zm5.321 0a1.274 1.274 0 1 1-1.275 1.27c0-.704.57-1.27 1.275-1.27z"></path><path fill="#010101" d="M11.79 8.915a.327.327 0 0 1-.307-.44l1.472-4.181 3.616.862c.177.044.283.22.238.395a.32.32 0 0 1-.391.239l-3.045-.729-1.275 3.632a.334.334 0 0 1-.308.222zm-.069 8.376c-2.076 0-2.908-.939-2.943-.982a.332.332 0 0 1 .035-.461.334.334 0 0 1 .459.033c.019.021.713.758 2.449.758 1.766 0 2.539-.763 2.549-.767a.324.324 0 1 1 .469.446c-.04.043-.954.973-3.018.973z"></path></symbol><symbol id="icon-twitter-bordered" viewBox="463.812 263.868 32 32"><path d="M486.416 276.191a5.622 5.622 0 0 1-1.554.429 2.718 2.718 0 0 0 1.19-1.502 5.456 5.456 0 0 1-1.72.657 2.71 2.71 0 0 0-1.979-.854 2.711 2.711 0 0 0-2.642 3.326 7.681 7.681 0 0 1-5.586-2.831 2.714 2.714 0 0 0 .839 3.618 2.748 2.748 0 0 1-1.227-.339v.031a2.71 2.71 0 0 0 2.174 2.656 2.735 2.735 0 0 1-1.229.049 2.726 2.726 0 0 0 2.531 1.883 5.442 5.442 0 0 1-4.01 1.123 7.672 7.672 0 0 0 4.155 1.215c4.983 0 7.71-4.129 7.71-7.711 0-.115-.004-.232-.006-.351a5.41 5.41 0 0 0 1.354-1.399z"></path><path d="M479.812 263.868c-8.837 0-16 7.163-16 16s7.163 16 16 16 16-7.163 16-16-7.163-16-16-16zm0 30c-7.732 0-14-6.269-14-14s6.268-14 14-14 14 6.269 14 14-6.267 14-14 14z"></path></symbol><symbol id="icon-twitter" viewBox="0 0 24 24"><circle fill="#26A7DF" cx="12" cy="12" r="12"></circle><path fill="#FFF" d="M5.903 6.768s2.176 2.883 5.953 3.082c0 0-.51-1.702 1.058-3.004 1.568-1.305 3.508-.2 3.879.292 0 0 1.104-.172 1.875-.691 0 0-.252.995-1.18 1.594 0 0 1.086-.146 1.596-.439 0 0-.533.853-1.447 1.503 0 0 .381 3.595-2.709 6.57-3.09 2.971-8.176 2.352-10.012.797 0 0 2.495.277 4.249-1.219 0 0-2.074-.037-2.671-1.973 0 0 1.036.08 1.215-.119 0 0-2.232-.539-2.254-2.873 0 0 .68.339 1.297.359 0-.001-2.153-1.557-.849-3.879z"></path></symbol><symbol id="icon-weibo-bordered" viewBox="463.812 263.868 32 32"><path d="M479.812 263.868c-8.838 0-16 7.163-16 16s7.162 16 16 16c8.837 0 16-7.163 16-16s-7.163-16-16-16zm0 30c-7.732 0-14-6.269-14-14s6.268-14 14-14c7.731 0 14 6.269 14 14s-6.267 14-14 14z"></path><path d="M478.552 285.348c-2.616.261-4.876-.926-5.044-2.649-.167-1.722 1.814-3.33 4.433-3.588 2.609-.263 4.871.926 5.041 2.647.165 1.721-1.818 3.331-4.43 3.59m5.23-5.718c-.226-.065-.374-.109-.259-.403.25-.639.276-1.188.005-1.581-.515-.734-1.915-.693-3.521-.021 0 0-.508.224-.378-.181.247-.798.209-1.468-.178-1.852-.87-.878-3.194.032-5.183 2.027-1.489 1.494-2.357 3.082-2.357 4.453 0 2.619 3.354 4.213 6.631 4.213 4.297 0 7.154-2.504 7.154-4.493.001-1.198-1.007-1.881-1.914-2.162"></path><path fill="#020202" d="M486.637 274.833a4.176 4.176 0 0 0-3.982-1.291.608.608 0 0 0-.465.72.604.604 0 0 0 .72.466 2.968 2.968 0 0 1 2.827.92 3 3 0 0 1 .625 2.918.602.602 0 0 0 .39.762.603.603 0 0 0 .763-.391v-.001a4.218 4.218 0 0 0-.878-4.103"></path><path fill="#020202" d="M485.041 276.276a2.037 2.037 0 0 0-1.938-.63.518.518 0 0 0-.396.621.517.517 0 0 0 .617.398c.336-.071.702.03.947.307s.312.649.207.979a.52.52 0 0 0 .336.654.523.523 0 0 0 .657-.336 2.038 2.038 0 0 0-.43-1.993m-6.347 5.951c-.09.156-.293.233-.451.166-.151-.062-.204-.235-.115-.389.093-.155.284-.229.44-.168.157.056.214.235.126.391m-.832 1.074c-.253.405-.795.58-1.202.396-.403-.186-.521-.655-.27-1.051.248-.39.771-.566 1.176-.393.413.17.543.636.296 1.048m.95-2.864c-1.244-.326-2.65.294-3.19 1.396-.553 1.119-.021 2.369 1.236 2.775 1.303.42 2.84-.225 3.374-1.436.526-1.183-.132-2.402-1.42-2.735"></path></symbol><symbol id="icon-weibo" viewBox="4 4 24 24"><circle fill="#C8E8F9" cx="16" cy="16" r="12"></circle><path fill="#FFF" d="M9.098 18.194c0 1.981 2.574 3.593 5.757 3.593 3.178 0 5.756-1.611 5.756-3.593 0-1.989-2.578-3.601-5.756-3.601-3.183.001-5.757 1.612-5.757 3.601"></path><path fill="#DF0A21" d="M14.991 21.496c-2.817.278-5.244-.996-5.428-2.85-.183-1.855 1.95-3.586 4.767-3.862 2.813-.278 5.243.996 5.428 2.849.18 1.855-1.957 3.584-4.767 3.863m5.628-6.155c-.24-.067-.404-.116-.279-.432.27-.688.299-1.278.004-1.699-.551-.791-2.062-.748-3.789-.022 0-.001-.548.241-.406-.191.266-.859.224-1.577-.191-1.993-.937-.942-3.437.037-5.577 2.18-1.602 1.61-2.533 3.317-2.533 4.789 0 2.817 3.606 4.533 7.13 4.533 4.626 0 7.701-2.696 7.701-4.834.001-1.292-1.085-2.027-2.06-2.331"></path><path fill="#F4992C" d="M23.689 10.182a4.492 4.492 0 0 0-4.283-1.39.651.651 0 1 0 .269 1.275 3.2 3.2 0 0 1 3.045.99 3.224 3.224 0 0 1 .672 3.138.647.647 0 0 0 .418.823.647.647 0 0 0 .816-.421v-.003a4.51 4.51 0 0 0-.937-4.412"></path><path fill="#F4992C" d="M21.973 11.736a2.183 2.183 0 0 0-2.086-.679.56.56 0 0 0-.428.667.554.554 0 0 0 .662.43v.001c.361-.075.754.034 1.018.33.266.299.34.698.227 1.054a.563.563 0 0 0 .359.706.563.563 0 0 0 .709-.362 2.2 2.2 0 0 0-.461-2.147"></path><path fill="#13110C" d="M15.143 18.139c-.1.17-.315.251-.483.179-.168-.07-.222-.255-.128-.42.097-.167.311-.248.479-.182.165.062.228.252.132.423m-.896 1.155c-.271.434-.854.625-1.295.425-.433-.195-.561-.707-.285-1.128.269-.426.832-.609 1.265-.427.443.183.58.688.315 1.13m1.024-3.084c-1.34-.351-2.852.32-3.434 1.506-.597 1.206-.021 2.546 1.33 2.984 1.402.453 3.055-.243 3.629-1.543.569-1.275-.142-2.587-1.525-2.947"></path></symbol></svg>
</div>


        

        

    <div class="u-vh-full">
        <div class="u-visually-hidden" aria-hidden="true">
    
    <!--?xml version="1.0" encoding="UTF-8"?--><svg xmlns="http://www.w3.org/2000/svg"><defs><filter id="a" filterUnits="userSpaceOnUse" x="0" y="0" width="60.643" height="31.01"><fecolormatrix values="1 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 1 0"></fecolormatrix></filter></defs><symbol id="icon-arrow-down" viewBox="0 0 16 8"><path d="M8 6l6-6h2L8 8 0 0h2l6 6"></path></symbol><symbol id="icon-arrow-left-bullet" viewBox="0 0 8 16"><path d="M3 8l5 5v3L0 8l8-8v3L3 8z"></path></symbol><symbol id="icon-arrow-right-bullet" viewBox="0 0 8 16"><path d="M5 8L0 3V0l8 8-8 8v-3l5-5z"></path></symbol><symbol id="icon-arrow-right" viewBox="0 0 32 64"><path d="M0 0v64l32-32z"></path></symbol><symbol id="icon-close" viewBox="0 0 23.998 23.999"><path d="M20.998 0l-8.999 9L3 0 0 3l8.999 9L0 20.999l3 3L11.999 15l8.999 8.999 3-3L14.999 12l8.999-9z"></path></symbol><symbol id="icon-community" viewBox="0 0 60.643 31.01"><mask maskUnits="userSpaceOnUse" x="0" y="0" width="60.643" height="31.01" id="b"><path fill="#FFF" d="M0 31.01h60.643V0H0z" filter="url(#a)"></path></mask><path mask="url(#b)" d="M60.643 25.497L50.005 14.86a7.925 7.925 0 1 0-7.669 0l-6.847 6.847-1.334-1.334a7.925 7.925 0 0 0-3.834-14.861 7.925 7.925 0 0 0-3.834 14.861l-1.334 1.334-6.847-6.847a7.925 7.925 0 1 0-7.669 0L0 25.497h21.362L15.85 31.01h28.943l-5.514-5.513h21.364z"></path></symbol><symbol id="icon-download" viewBox="-301 390 9 14"><path d="M-301 395.6l4.5 5.1 4.5-5.1h-3V390h-3v5.6h-3zm0 6.5h9v1.9h-9z"></path></symbol><symbol id="icon-ext-link" viewBox="0 0 16 16"><path fill="#676767" d="M12.9 16H3.1C1.4 16 0 14.6 0 12.9V3.2C0 1.4 1.4 0 3.1 0h3.7v1H3.1C2 1 1 2 1 3.2v9.7C1 14 2 15 3.1 15h9.7c1.2 0 2.1-1 2.1-2.1V8.7h1v4.2c.1 1.7-1.3 3.1-3 3.1z"></path><path fill="#676767" d="M12.8 2.5l.7.7-9 8.9-.7-.7 9-8.9z"></path><path fill="#676767" d="M9.7 0L16 6.2V0z"></path></symbol><symbol id="icon-info-bordered" viewBox="470.812 270.868 18 18"><path d="M479.812 270.868c-4.972 0-9 4.029-9 9s4.028 9 9 9c4.971 0 9-4.029 9-9s-4.029-9-9-9zm0 16.875a7.875 7.875 0 0 1-7.875-7.875 7.875 7.875 0 1 1 15.75 0 7.875 7.875 0 0 1-7.875 7.875z"></path><path d="M479.284 279.586c.089-.238-.024-.361-.13-.361-.489 0-1.123 1.16-1.359 1.16-.093 0-.173-.092-.173-.174 0-.238.581-.801.751-.971.526-.506 1.217-.895 1.979-.895.567 0 1.174.346.703 1.639l-.952 2.604c-.079.199-.224.531-.224.746 0 .092.054.186.159.186.395 0 1.121-1.135 1.304-1.135.067 0 .158.086.158.199 0 .385-1.542 2.043-2.874 2.043-.477 0-.804-.225-.804-.732 0-.641.447-1.734.538-1.963l.924-2.346zm.727-3.41c0-.586.498-1.066 1.082-1.066.527 0 .91.357.91.906 0 .615-.501 1.068-1.098 1.068-.541-.002-.894-.361-.894-.908z"></path></symbol><symbol id="icon-opr" viewBox="0 0 18 18"><path d="M.9 2.5c-.2-.2-.4-.3-.6-.2-.3.2-.3.4-.3.6 0 .8 0 1.7.1 2.5 0 .5.3.7.7.6.9 0 1.7-.2 2.5-.3h.2c.5-.2.6-.5.3-.8-.2-.2-.4-.4-.6-.5-.1-.1-.2-.2-.4-.3.5-.6 1-1.1 1.7-1.5 4.6-3 10.9-.5 12 4.9.9 4.4-2.2 8.6-6.7 9.2-3.7.5-7.2-1.6-8.4-5.1 0-.1-.1-.2-.1-.3-.1-.3-.5-.5-.8-.4-.3.1-.5.4-.4.8.4 1.3 1.1 2.4 2 3.4 1.4 1.5 3.1 2.4 5.1 2.8 3 .5 5.6-.2 7.8-2.2 2.1-1.9 3-4.2 3-6.9-.1-4-3.1-7.6-7.1-8.4-3.3-.9-6.2 0-8.6 2.3-.2.1-.3.3-.5.5-.3-.2-.6-.5-.9-.7z"></path><path d="M13 4.7c-2.6 1.6-4.5 3.6-5.3 4.6L5.6 7.7l-.9.7L8.3 12c.6-1.6 2.6-4.6 4.9-6.8l-.2-.5z"></path></symbol><symbol id="icon-quotation" viewBox="0 0.333 55.932 35.619"><path d="M43.738.333L31.545 12.207v23.745h24.387V12.207H43.738zm-31.545 0L0 12.207v23.745h24.386V12.207H12.193z"></path></symbol><symbol id="icon-remove" viewBox="-296 388 18 18"><path d="M-291.7 396.1h9v2h-9z"></path><path d="M-287 405.5c-4.7 0-8.5-3.8-8.5-8.5s3.8-8.5 8.5-8.5 8.5 3.8 8.5 8.5-3.8 8.5-8.5 8.5zm0-16c-4.1 0-7.5 3.4-7.5 7.5s3.4 7.5 7.5 7.5 7.5-3.4 7.5-7.5-3.4-7.5-7.5-7.5z"></path></symbol><symbol id="icon-ribbon" viewBox="0 0 36 72"><path d="M0 0v72h.053L18 54.054 35.945 72H36V0H0zm25.8 11.4l3.001 3L15 28.2l-7.801-7.8 3-3L15 22.201 25.8 11.4z"></path></symbol><symbol id="icon-rss" viewBox="0 0 24 24"><g fill="#F26822"><ellipse cx="3.305" cy="20.702" rx="3.306" ry="3.298"></ellipse><path d="M15.978 24h-4.684c0-6.224-5.057-11.27-11.294-11.27V8.058c8.824 0 15.978 7.137 15.978 15.942z"></path><path d="M19.2 23.95C19.2 13.366 10.604 4.79 0 4.79V0c13.255 0 24 10.723 24 23.95h-4.8z"></path></g></symbol><symbol id="icon-search" viewBox="0 0 16 16"><path d="M6.6 10.6c2.2 0 4-1.7 4-4s-1.7-4-4-4-4 1.7-4 4 1.8 4 4 4zm4 1.4l1.5-1.5 3.7 3.6c.4.5.4 1.2 0 1.6-.4.4-1.1.4-1.6 0L10.6 12zm-4 1.2C2.9 13.2 0 10.3 0 6.6S2.9 0 6.6 0s6.6 2.9 6.6 6.6-2.9 6.6-6.6 6.6z"></path></symbol><symbol id="icon-speech-bubble" viewBox="0 0 40 54"><path d="M0 0v40h6.667v13.334L20 40h20V0z"></path></symbol><symbol id="icon-error" viewBox="2.002 0 14 14"><path d="M16.002 11.949L13.951 14 9.002 9.051 4.053 14l-2.051-2.051L6.951 7 2.002 2.05 4.053 0l4.949 4.95L13.951 0l2.051 2.049L11.053 7l4.949 4.949z"></path></symbol><symbol id="icon-info" viewBox="5.502 0 7 14"><path d="M8.165 5.923c.146-.384-.04-.577-.209-.577-.783 0-1.798 1.856-2.177 1.856-.15 0-.277-.147-.277-.279 0-.383.933-1.282 1.203-1.557.846-.811 1.949-1.434 3.174-1.434.908 0 1.879.555 1.125 2.63l-1.526 4.167c-.126.32-.358.854-.358 1.198 0 .148.088.298.255.298.632 0 1.796-1.818 2.091-1.818.105 0 .252.134.252.321C11.717 11.345 9.245 14 7.11 14c-.763 0-1.289-.361-1.289-1.176 0-1.027.717-2.778.865-3.145l1.479-3.756zm1.144-4.211C9.309.771 10.111 0 11.045 0c.845 0 1.457.577 1.457 1.456 0 .983-.804 1.709-1.757 1.709-.866-.001-1.436-.578-1.436-1.453z"></path></symbol><symbol id="icon-success" viewBox="0 0 16 14"><path d="M15.592 0c-4.867 2.984-8.4 6.751-9.987 8.641L1.717 5.595 0 6.979l6.717 6.832c1.157-2.961 4.817-8.748 9.287-12.86L15.592 0z"></path></symbol><symbol id="icon-warning" viewBox="7.002 0 2.789 14"><path d="M7.002 11.211h2.789V14H7.002v-2.789zM7.002 0v3.436l.741 5.326h1.326l.723-5.326V0h-2.79z"></path></symbol><symbol id="icon-facebook-bordered" viewBox="463.812 263.868 32 32"><path d="M479.812 263.868c-8.837 0-16 7.163-16 16s7.163 16 16 16 16-7.163 16-16-7.163-16-16-16zm0 30c-7.732 0-14-6.269-14-14s6.268-14 14-14 14 6.269 14 14-6.267 14-14 14z"></path><path d="M483.025 280.48l.32-2.477h-2.453v-1.582c0-.715.199-1.207 1.227-1.207h1.311v-2.213a17.753 17.753 0 0 0-1.907-.098c-1.894 0-3.186 1.154-3.186 3.271V278h-2.142v2.477h2.142v6.354h2.557v-6.354l2.131.003z"></path></symbol><symbol id="icon-facebook" viewBox="0 0 24 24"><circle fill="#4D67A4" cx="12" cy="12" r="12"></circle><path fill="#FFF" d="M8.396 10.143h2.137V8.165s-.092-1.292.892-2.274c.979-.979 2.298-.887 4.177-.724v2.21h-1.387s-.586-.013-.861.303c-.271.315-.24.764-.24.875v1.588h2.41l-.311 2.458h-2.116v6.336h-2.56v-6.345H8.396v-2.449z"></path></symbol><symbol id="icon-linkedin" viewBox="0 0 24 24"><circle fill="#0178B5" cx="12" cy="12" r="12"></circle><g fill="#FFF"><circle cx="8.022" cy="8.043" r="1.256"></circle><path d="M6.929 10.246h2.17v6.967h-2.17zm3.533 6.967h2.157v-3.594s-.078-1.627 1.219-1.627c1.301 0 1.211 1.231 1.211 1.635v3.586h2.183v-3.775s.129-2.745-1.472-3.191c-1.605-.445-2.661.071-3.219.985v-.985h-2.079v6.966z"></path></g></symbol><symbol id="icon-reddit" viewBox="0 0 24 24"><circle fill="#BCBCBC" cx="12" cy="12" r="12"></circle><path fill="#FFF" d="M4.661 9.741c.941 0 1.703.761 1.703 1.704 0 .938-.762 1.705-1.703 1.705s-1.704-.767-1.704-1.705c0-.943.763-1.704 1.704-1.704zm13.844 0a1.704 1.704 0 1 1 .001 3.409 1.704 1.704 0 0 1-.001-3.409z"></path><path fill="#FFF" d="M11.736 8.732c4.285 0 7.762 2.283 7.762 5.104 0 2.812-3.477 5.1-7.762 5.1-4.288 0-7.762-2.285-7.762-5.1.001-2.82 3.474-5.104 7.762-5.104z"></path><path fill="#010101" d="M11.736 19.262c-4.461 0-8.088-2.437-8.088-5.426 0-2.994 3.626-5.431 8.088-5.431 4.457 0 8.087 2.437 8.087 5.431 0 2.989-3.63 5.426-8.087 5.426zm0-10.205c-4.104 0-7.438 2.145-7.438 4.779 0 2.633 3.334 4.773 7.438 4.773 4.098 0 7.437-2.142 7.437-4.773 0-2.636-3.339-4.779-7.437-4.779z"></path><circle fill="#FFF" cx="18.014" cy="5.765" r="1.385"></circle><path fill="#010101" d="M18.014 7.456a1.694 1.694 0 0 1-1.695-1.691c0-.934.764-1.694 1.695-1.694s1.693.762 1.693 1.694c0 .934-.762 1.691-1.693 1.691zm0-2.734c-.574 0-1.043.469-1.043 1.043s.469 1.041 1.043 1.041a1.042 1.042 0 0 0 0-2.084zM3.818 13.275a2.068 2.068 0 0 1-1.068-1.811 2.074 2.074 0 0 1 3.538-1.466l-.458.462a1.412 1.412 0 0 0-1.007-.42 1.424 1.424 0 0 0-.688 2.669l-.317.566zm15.866 0l-.316-.566a1.43 1.43 0 0 0 .73-1.245c0-.784-.635-1.424-1.42-1.424-.381 0-.736.149-1.008.42l-.459-.462a2.074 2.074 0 1 1 2.473 3.277z"></path><path fill="#F04B23" d="M9.165 11.604c.704 0 1.272.564 1.272 1.27a1.273 1.273 0 1 1-1.272-1.27zm5.321 0a1.274 1.274 0 1 1-1.275 1.27c0-.704.57-1.27 1.275-1.27z"></path><path fill="#010101" d="M11.79 8.915a.327.327 0 0 1-.307-.44l1.472-4.181 3.616.862c.177.044.283.22.238.395a.32.32 0 0 1-.391.239l-3.045-.729-1.275 3.632a.334.334 0 0 1-.308.222zm-.069 8.376c-2.076 0-2.908-.939-2.943-.982a.332.332 0 0 1 .035-.461.334.334 0 0 1 .459.033c.019.021.713.758 2.449.758 1.766 0 2.539-.763 2.549-.767a.324.324 0 1 1 .469.446c-.04.043-.954.973-3.018.973z"></path></symbol><symbol id="icon-twitter-bordered" viewBox="463.812 263.868 32 32"><path d="M486.416 276.191a5.622 5.622 0 0 1-1.554.429 2.718 2.718 0 0 0 1.19-1.502 5.456 5.456 0 0 1-1.72.657 2.71 2.71 0 0 0-1.979-.854 2.711 2.711 0 0 0-2.642 3.326 7.681 7.681 0 0 1-5.586-2.831 2.714 2.714 0 0 0 .839 3.618 2.748 2.748 0 0 1-1.227-.339v.031a2.71 2.71 0 0 0 2.174 2.656 2.735 2.735 0 0 1-1.229.049 2.726 2.726 0 0 0 2.531 1.883 5.442 5.442 0 0 1-4.01 1.123 7.672 7.672 0 0 0 4.155 1.215c4.983 0 7.71-4.129 7.71-7.711 0-.115-.004-.232-.006-.351a5.41 5.41 0 0 0 1.354-1.399z"></path><path d="M479.812 263.868c-8.837 0-16 7.163-16 16s7.163 16 16 16 16-7.163 16-16-7.163-16-16-16zm0 30c-7.732 0-14-6.269-14-14s6.268-14 14-14 14 6.269 14 14-6.267 14-14 14z"></path></symbol><symbol id="icon-twitter" viewBox="0 0 24 24"><circle fill="#26A7DF" cx="12" cy="12" r="12"></circle><path fill="#FFF" d="M5.903 6.768s2.176 2.883 5.953 3.082c0 0-.51-1.702 1.058-3.004 1.568-1.305 3.508-.2 3.879.292 0 0 1.104-.172 1.875-.691 0 0-.252.995-1.18 1.594 0 0 1.086-.146 1.596-.439 0 0-.533.853-1.447 1.503 0 0 .381 3.595-2.709 6.57-3.09 2.971-8.176 2.352-10.012.797 0 0 2.495.277 4.249-1.219 0 0-2.074-.037-2.671-1.973 0 0 1.036.08 1.215-.119 0 0-2.232-.539-2.254-2.873 0 0 .68.339 1.297.359 0-.001-2.153-1.557-.849-3.879z"></path></symbol><symbol id="icon-weibo-bordered" viewBox="463.812 263.868 32 32"><path d="M479.812 263.868c-8.838 0-16 7.163-16 16s7.162 16 16 16c8.837 0 16-7.163 16-16s-7.163-16-16-16zm0 30c-7.732 0-14-6.269-14-14s6.268-14 14-14c7.731 0 14 6.269 14 14s-6.267 14-14 14z"></path><path d="M478.552 285.348c-2.616.261-4.876-.926-5.044-2.649-.167-1.722 1.814-3.33 4.433-3.588 2.609-.263 4.871.926 5.041 2.647.165 1.721-1.818 3.331-4.43 3.59m5.23-5.718c-.226-.065-.374-.109-.259-.403.25-.639.276-1.188.005-1.581-.515-.734-1.915-.693-3.521-.021 0 0-.508.224-.378-.181.247-.798.209-1.468-.178-1.852-.87-.878-3.194.032-5.183 2.027-1.489 1.494-2.357 3.082-2.357 4.453 0 2.619 3.354 4.213 6.631 4.213 4.297 0 7.154-2.504 7.154-4.493.001-1.198-1.007-1.881-1.914-2.162"></path><path fill="#020202" d="M486.637 274.833a4.176 4.176 0 0 0-3.982-1.291.608.608 0 0 0-.465.72.604.604 0 0 0 .72.466 2.968 2.968 0 0 1 2.827.92 3 3 0 0 1 .625 2.918.602.602 0 0 0 .39.762.603.603 0 0 0 .763-.391v-.001a4.218 4.218 0 0 0-.878-4.103"></path><path fill="#020202" d="M485.041 276.276a2.037 2.037 0 0 0-1.938-.63.518.518 0 0 0-.396.621.517.517 0 0 0 .617.398c.336-.071.702.03.947.307s.312.649.207.979a.52.52 0 0 0 .336.654.523.523 0 0 0 .657-.336 2.038 2.038 0 0 0-.43-1.993m-6.347 5.951c-.09.156-.293.233-.451.166-.151-.062-.204-.235-.115-.389.093-.155.284-.229.44-.168.157.056.214.235.126.391m-.832 1.074c-.253.405-.795.58-1.202.396-.403-.186-.521-.655-.27-1.051.248-.39.771-.566 1.176-.393.413.17.543.636.296 1.048m.95-2.864c-1.244-.326-2.65.294-3.19 1.396-.553 1.119-.021 2.369 1.236 2.775 1.303.42 2.84-.225 3.374-1.436.526-1.183-.132-2.402-1.42-2.735"></path></symbol><symbol id="icon-weibo" viewBox="4 4 24 24"><circle fill="#C8E8F9" cx="16" cy="16" r="12"></circle><path fill="#FFF" d="M9.098 18.194c0 1.981 2.574 3.593 5.757 3.593 3.178 0 5.756-1.611 5.756-3.593 0-1.989-2.578-3.601-5.756-3.601-3.183.001-5.757 1.612-5.757 3.601"></path><path fill="#DF0A21" d="M14.991 21.496c-2.817.278-5.244-.996-5.428-2.85-.183-1.855 1.95-3.586 4.767-3.862 2.813-.278 5.243.996 5.428 2.849.18 1.855-1.957 3.584-4.767 3.863m5.628-6.155c-.24-.067-.404-.116-.279-.432.27-.688.299-1.278.004-1.699-.551-.791-2.062-.748-3.789-.022 0-.001-.548.241-.406-.191.266-.859.224-1.577-.191-1.993-.937-.942-3.437.037-5.577 2.18-1.602 1.61-2.533 3.317-2.533 4.789 0 2.817 3.606 4.533 7.13 4.533 4.626 0 7.701-2.696 7.701-4.834.001-1.292-1.085-2.027-2.06-2.331"></path><path fill="#F4992C" d="M23.689 10.182a4.492 4.492 0 0 0-4.283-1.39.651.651 0 1 0 .269 1.275 3.2 3.2 0 0 1 3.045.99 3.224 3.224 0 0 1 .672 3.138.647.647 0 0 0 .418.823.647.647 0 0 0 .816-.421v-.003a4.51 4.51 0 0 0-.937-4.412"></path><path fill="#F4992C" d="M21.973 11.736a2.183 2.183 0 0 0-2.086-.679.56.56 0 0 0-.428.667.554.554 0 0 0 .662.43v.001c.361-.075.754.034 1.018.33.266.299.34.698.227 1.054a.563.563 0 0 0 .359.706.563.563 0 0 0 .709-.362 2.2 2.2 0 0 0-.461-2.147"></path><path fill="#13110C" d="M15.143 18.139c-.1.17-.315.251-.483.179-.168-.07-.222-.255-.128-.42.097-.167.311-.248.479-.182.165.062.228.252.132.423m-.896 1.155c-.271.434-.854.625-1.295.425-.433-.195-.561-.707-.285-1.128.269-.426.832-.609 1.265-.427.443.183.58.688.315 1.13m1.024-3.084c-1.34-.351-2.852.32-3.434 1.506-.597 1.206-.021 2.546 1.33 2.984 1.402.453 3.055-.243 3.629-1.543.569-1.275-.142-2.587-1.525-2.947"></path></symbol></svg>
</div>

        <a class="u-visually-hidden u-visually-hidden-focus" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#main-content">
    <span class="c-banner">Skip to content</span>
</a>

        
            
    <div class="adsbox c-ad c-ad--LB1">
        <div class="c-ad__inner">
            <p class="c-ad__label">Advertisement</p>
            <div id="div-gpt-ad-LB1" data-gpt-unitpath="/270604982/bmc/bmcbioinformatics/articles" data-gpt-sizes="728x90,970x90" data-gpt-targeting="pos=LB1;doi=10.1186/1471-2105-9-559;kwrd=Correlation Network,Module Membership,Module Eigengene,Brown Module,Hierarchical Cluster Dendrogram;pmc=L15001,B12050,I23050,L17004,M14018,M14018;" data-google-query-id="CKS1k63G9OECFUN7YgodNJgMMw">
                
            <div id="google_ads_iframe_/270604982/bmc/bmcbioinformatics/articles_0__container__" style="border: 0pt none; display: inline-block; width: 970px; height: 90px;"><iframe frameborder="0" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/container.html" id="google_ads_iframe_/270604982/bmc/bmcbioinformatics/articles_0" title="3rd party ad content" name="" scrolling="no" marginwidth="0" marginheight="0" width="970" height="90" data-is-safeframe="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-top-navigation-by-user-activation" data-google-container-id="1" style="border: 0px; vertical-align: bottom;" data-load-complete="true"></iframe></div></div>
        </div>
    </div>

        
         
    
        <div id="membership-message-loader-desktop" class="placeholder" data-placeholder="/placeholder/v1/membership/message"></div>
    
    
        <header class="c-popup-search" id="top" data-test="publisher-header">
    <nav class="c-navbar">
        <div class="c-navbar__container">
            
            <div class="basic-white-logo c-navbar__logo u-margin-right-xxl" itemscope="" itemtype="http://schema.org/Organization" data-test="navbar-logo-header">
                <link itemprop="parentOrganization" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#parentOrganization">
                <div class="c-logo">
    <a href="https://www.biomedcentral.com/" itemprop="url">
        <img alt="BMC" itemprop="logo" width="76" height="18" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/logo-bmc-white-series-1da6525f4e.svg">
        <div class="c-logo__strapline">
            <img alt="Part of Springer Nature" width="173" height="16" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/logo-bmc-white-strapline-sn-e7b1d6e67f.svg">
        </div>
    </a>
</div>

            </div>
            <button type="button" class="c-navbar__link c-dropdown js-publisher-menu-button u-hidden-at-lg" id="publisher-header-menu-button" data-toggle="collapse" data-test="publisher-header-menu-toggle" data-toggle-linked="publisher-header" data-target="publisher-header-search publisher-header-menu" aria-controls="publisher-header-menu publisher-header-search" aria-expanded="false">
                <span class="c-dropdown__toggle">Menu</span>
            </button>
            <div class="c-navbar__content c-collapse" id="publisher-header-menu">
                
    <ul class="c-navbar__nav" data-test="publisher-navigation">
        
        <li class="c-navbar__item">
            <a class="c-navbar__link" href="https://www.biomedcentral.com/journals" data-track="click" data-track-action="click-nav-link" data-track-category="publisher-header" data-track-label="Explore journals">
                Explore journals
            </a>
        </li>
        
        <li class="c-navbar__item">
            <a class="c-navbar__link" href="https://www.biomedcentral.com/getpublished" data-track="click" data-track-action="click-nav-link" data-track-category="publisher-header" data-track-label="Get published">
                Get published
            </a>
        </li>
        
        <li class="c-navbar__item">
            <a class="c-navbar__link" href="https://www.biomedcentral.com/about" data-track="click" data-track-action="click-nav-link" data-track-category="publisher-header" data-track-label="About BMC">
                About BMC
            </a>
        </li>
        
    </ul>


                <ul class="c-navbar__nav c-navbar__nav--right">
                    <li class="c-navbar__item">
                        <button type="button" class="c-navbar__link js-publisher-search-button u-hidden-at-lt-lg" data-toggle="collapse" data-test="header-search-button" data-target="publisher-header-search publisher-header-menu publisher-header-menu-button" aria-controls="publisher-header-search publisher-header-menu" aria-expanded="false" data-track="click" data-track-action="click-nav-link" data-track-category="publisher-header" data-track-label="search">
                            <span class="c-flex c-flex--align-center">
                                Search
                                <svg class="c-icon u-margin-left-xs" width="14" height="14" aria-hidden="true">
                                    <use xlink:href="#icon-search"></use>
                                </svg>
                            </span>
                        </button>
                    </li>
                    <li class="c-navbar__item">
                        <a data-replace="true" data-placeholder="/placeholder/v1/username" class="c-navbar__link" href="https://www.biomedcentral.com/login" data-test="login-link">
                            <span class="u-js-shown" data-track="click" data-track-action="click-nav-link" data-track-category="publisher-header" data-track-label="Login">Login</span>
                            <span class="u-js-hidden" data-track="click" data-track-action="click-nav-link" data-track-category="publisher-header" data-track-label="My Account">My Account</span>
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
    <div class="c-popup-search__content c-collapse js-publisher-search-bar" id="publisher-header-search">
        <div class="u-container">
            <div class="c-popup-search__container">
                <div class="ctx-search">
    <form role="search" class="c-form-field" method="GET" action="https://www.biomedcentral.com/search" data-track="submit" data-track-category="Search and Results" data-track-action="Submit search" data-dynamic-track-label="" data-track-label="" data-test="global-search" novalidate="">
        <label for="publisherSearch" class="c-form-field__label">Search all BMC articles</label>
        <div class="c-flex">
            <input id="publisherSearch" class="c-form-field__input js-publisher-search-input" autocomplete="off" role="textbox" data-test="search-input" name="query" type="text" value="">
            <div>
                <button class="c-button" type="submit" data-test="search-submit-button">
    <span class="u-visually-hidden">Search</span>
    <svg class="c-icon" width="16" height="16" aria-hidden="true">
        <use xlink:href="#icon-search"></use>
    </svg>
</button>


            </div>
        </div>
        <input type="hidden" name="searchType" value="publisherSearch">
    </form>
</div>

            </div>
        </div>
    </div>
</header>

    

        
            <header class="c-journal-header c-journal-header--bmc-bioinformatics ctx-journal-header">
                <div class="u-container">
                    <div class="c-journal-header__inner ">
                        
                        <div class="c-journal-title" id="journalTitle">
                            <a href="https://bmcbioinformatics.biomedcentral.com/">
    
        
    
    <span class="c-journal-title__text ">BMC Bioinformatics</span>
    
</a>
                        </div>
                        
                    </div>
                </div>
                <div class="c-navbar">
                    <div class="c-navbar__container">
                        
                            <button type="button" class="c-navbar__link c-dropdown u-hidden-at-lg" data-toggle="collapse" data-test="journal-header-menu" data-toggle-linked="journal-header" data-target="journal-menu" aria-controls="journal-menu" aria-expanded="false">
                                    <span class="c-dropdown__toggle">Menu</span>
                            </button>
                            <div class="c-navbar__content c-collapse" id="journal-menu">
                                <nav class="c-navbar__nav">
                                    <ul class="c-navbar__nav c-navbar__nav--journal" role="menu" data-test="site-navigation">
                                        
                                            <li class="c-navbar__item" role="menuitem">
                                                <a class="c-navbar__link" href="https://bmcbioinformatics.biomedcentral.com/">Home</a>
                                            </li>
                                        
                                            <li class="c-navbar__item" role="menuitem">
                                                <a class="c-navbar__link" href="https://bmcbioinformatics.biomedcentral.com/about">About</a>
                                            </li>
                                        
                                            <li class="c-navbar__item" role="menuitem">
                                                <a class="c-navbar__link c-navbar__link--is-shown" href="https://bmcbioinformatics.biomedcentral.com/articles">Articles</a>
                                            </li>
                                        
                                            <li class="c-navbar__item" role="menuitem">
                                                <a class="c-navbar__link" href="https://bmcbioinformatics.biomedcentral.com/submission-guidelines">Submission Guidelines</a>
                                            </li>
                                        
                                    </ul>
                                </nav>
                            </div>
                        
                    </div>
                </div>
                <div class="c-journal-header__identity c-journal-header__identity--default">
                    
                </div>
            </header>
            
        

        <div class="u-container u-margin-top-xl u-margin-bottom-xl" id="main-content">
            <div class="c-page-layout c-page-layout--fulltext" data-component="sticky-container">
                <div class="c-page-layout__nav u-hidden-at-lt-xl">
                    <nav xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="ArticleNavigation" aria-labelledby="toc-menu-title" data-component="sticky"><h3 id="toc-menu-title" class="u-visually-hidden" data-test="toc-title">Table of Contents</h3><ol class="c-nav c-nav--stacked" role="menu" id="tocMenu" aria-expanded="true" data-article-toc=""><li role="menuitem" class="c-nav__item"><a class="c-nav__link" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Abs1" data-track-category="Table of Contents" data-track="click" data-track-action="Link Clicked" data-track-label="Abstract" data-test="abstract-link">Abstract</a></li><li role="menuitem" class="c-nav__item"><a class="c-nav__link" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Sec1" data-track-category="Table of Contents" data-track="click" data-track-action="Link Clicked" data-track-label="Background">Background</a></li><li role="menuitem" class="c-nav__item"><a class="c-nav__link" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Sec3" data-track-category="Table of Contents" data-track="click" data-track-action="Link Clicked" data-track-label="Results">Results</a></li><li role="menuitem" class="c-nav__item"><a class="c-nav__link" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Sec19" data-track-category="Table of Contents" data-track="click" data-track-action="Link Clicked" data-track-label="Discussion">Discussion</a></li><li role="menuitem" class="c-nav__item"><a class="c-nav__link" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Sec20" data-track-category="Table of Contents" data-track="click" data-track-action="Link Clicked" data-track-label="Conclusion">Conclusion</a></li><li role="menuitem" class="c-nav__item"><a class="c-nav__link" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Sec21" data-track-category="Table of Contents" data-track="click" data-track-action="Link Clicked" data-track-label="Availability and requirements">Availability and requirements</a></li><li role="menuitem" class="c-nav__item"><a class="c-nav__link" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Sec22" data-track-category="Table of Contents" data-track="click" data-track-action="Link Clicked" data-track-label="Acknowledgements">Acknowledgements</a></li><li role="menuitem" class="c-nav__item"><a class="c-nav__link" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Decs" data-test="declarations-link" data-track-category="Table of Contents" data-track="click" data-track-action="Link Clicked" data-track-label="Declarations">
                  Declarations
                </a></li><li role="menuitem" class="c-nav__item"><a class="c-nav__link" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Bib1" data-track-category="Table of Contents" data-track="click" data-track-action="Link Clicked" data-track-label="References">References</a></li></ol></nav>
                </div>

                <main class="c-page-layout__main">
                    <article itemscope="" itemtype="http://schema.org/ScholarlyArticle">
                        
                        
    <div class="FulltextWrapper">
        <ul class="c-article-identifiers u-reset-list">
            <li class="c-article-identifiers__item" data-test="article-category">Software</li>
            <li class="c-article-identifiers__item" data-test="open-access">
                <span class="u-text-open-access">Open Access</span>
            </li>
            
            
        </ul>
    </div>

                        <div class="FulltextWrapper"><div xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="MainTitleSection"><h1 xmlns="http://www.w3.org/1999/xhtml" class="ArticleTitle" itemprop="headline" lang="en">WGCNA: an R package for weighted correlation network analysis</h1></div><div xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="AuthorNames" data-author-affiliations="true"><ul class="u-listReset"><li class="Author hasAffil" data-jumpto="Aff" data-test="author-name"><span xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="AuthorName" itemprop="author" itemscope="" itemtype="http://schema.org/Person">Peter&nbsp;Langfelder</span><sup><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Aff1" class="AffiliationID">1</a></sup> and </li><li class="Author hasAffil" data-jumpto="Aff" data-test="author-name"><span xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="AuthorName" itemprop="author" itemscope="" itemtype="http://schema.org/Person">Steve&nbsp;Horvath</span><sup><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Aff2" class="AffiliationID">2</a></sup><a href="mailto:shorvath@mednet.ucla.edu" class="EmailAuthor" title="Email author"><span class="ContactIcon" aria-hidden="true"></span><span class="u-srOnly">Email author</span></a></li></ul></div><div xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="ArticleContextInformation"><div class="ContextInformation"><div class="ContextInformationJournalTitles"><span class="ArticleCitation"><span itemscope="" itemtype="http://schema.org/Periodical" itemid="#periodical"><span xmlns="http://www.w3.org/1999/xhtml" class="JournalTitle">BMC Bioinformatics</span></span><span class="ArticleCitation_Year">2008</span><span class="ArticleCitation_Volume"><strong itemprop="isPartOf" itemtype="http://schema.org/PublicationVolume"><link itemprop="isPartOf" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#periodical">9</strong>:559</span></span></div><p class="ArticleDOI"><a itemprop="sameAs" class="u-text-inherit" href="https://doi.org/10.1186/1471-2105-9-559">https://doi.org/10.1186/1471-2105-9-559</a></p><p class="Copyright">© &nbsp;Langfelder and Horvath; licensee BioMed Central Ltd.&nbsp;2008</p></div></div><ul class="ArticleHistory u-reset-list"><li class="History HistoryReceived"><strong>Received: </strong><span>24&nbsp;July&nbsp;2008</span></li><li class="History HistoryAccepted"><strong>Accepted: </strong><span>29&nbsp;December&nbsp;2008</span></li><li class="History HistoryOnlineDate"><strong>Published: </strong><span itemprop="datePublished">29&nbsp;December&nbsp;2008</span></li></ul></div>
                        
                        
                        
    

                        <div class="FulltextWrapper"><section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="Abstract Section1 RenderAsSection1 c-section" id="Abs1" lang="en"><h2 class="c-section__heading"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Abstract" data-test="abstract-heading" data-disable-click-event-default="">Abstract</a></h2><div class="c-section__content" id="Abstract"><div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="AbstractSection" id="ASec1">
                <h3 xmlns="" class="Heading">Background</h3>
                <p class="Para">Correlation networks are increasingly being used in bioinformatics applications. For example, weighted gene co-expression network analysis is a systems biology method for describing the correlation patterns among genes across microarray samples. Weighted correlation network analysis (WGCNA) can be used for finding clusters (modules) of highly correlated genes, for summarizing such clusters using the module eigengene or an intramodular hub gene, for relating modules to one another and to external sample traits (using eigengene network methodology), and for calculating module membership measures. Correlation networks facilitate network based gene screening methods that can be used to identify candidate biomarkers or therapeutic targets. These methods have been successfully applied in various biological contexts, e.g. cancer, mouse genetics, yeast genetics, and analysis of brain imaging data. While parts of the correlation network methodology have been described in separate publications, there is a need to provide a user-friendly, comprehensive, and consistent software implementation and an accompanying tutorial.</p>
              </div><div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="AbstractSection" id="ASec2">
                <h3 xmlns="" class="Heading">Results</h3>
                <p class="Para">The WGCNA R software package is a comprehensive collection of R functions for performing various aspects of weighted correlation network analysis. The package includes functions for network construction, module detection, gene selection, calculations of topological properties, data simulation, visualization, and interfacing with external software. Along with the R package we also present R software tutorials. While the methods development was motivated by gene expression data, the underlying data mining approach can be applied to a variety of different settings.</p>
              </div><div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="AbstractSection" id="ASec3">
                <h3 xmlns="" class="Heading">Conclusion</h3>
                <p class="Para">The WGCNA package provides R functions for weighted correlation network analysis, e.g. co-expression network analysis of gene expression data. The R package along with its source code and additional material are freely available at <span class="ExternalRef"><a href="http://www.genetics.ucla.edu/labs/horvath/CoexpressionNetwork/Rpackages/WGCNA" target="_blank" class="ext-link"><span class="RefSource">http://www.genetics.ucla.edu/labs/horvath/CoexpressionNetwork/Rpackages/WGCNA</span></a></span>.</p>
              </div></div></section><section xmlns:func="http://oscar.fig.bmc.com" class="KeywordGroup Section1 RenderAsSection1 c-section" lang="en"><h2 xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="Heading js-ToggleCollapseSection c-section__heading" data-sticky-update="true"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Keywords" data-toggle="collapse" data-toggle-disable-clickoff="" data-target="Keywords">Keywords</a></h2><div class="js-CollapseSection CollapseSection c-section__content" data-test="collapse-section" id="Keywords"><ul class="c-keywords"><li class="c-keywords__item" itemprop="about">Correlation Network</li><li class="c-keywords__item" itemprop="about">Module Membership</li><li class="c-keywords__item" itemprop="about">Module Eigengene</li><li class="c-keywords__item" itemprop="about">Brown Module</li><li class="c-keywords__item" itemprop="about">Hierarchical Cluster Dendrogram</li></ul></div></section>
            <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="Sec1" class="Section1 RenderAsSection1 c-section"><h2 class="Heading js-ToggleCollapseSection c-section__heading" data-sticky-update="true"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Background" data-toggle="collapse" data-toggle-disable-clickoff="" data-target="Background">Background</a></h2><div class="js-CollapseSection CollapseSection c-section__content" id="Background" data-test="collapse-section"><div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Correlation networks are increasingly being used in biology to analyze large, high-dimensional data sets. Correlation networks are constructed on the basis of correlations between quantitative measurements that can be described by an <em xmlns="" class="EmphasisTypeItalic">n</em> × <em xmlns="" class="EmphasisTypeItalic">m</em> matrix <em xmlns="" class="EmphasisTypeItalic">X</em> = [<em xmlns="" class="EmphasisTypeItalic">x</em><sub>
                  <em xmlns="" class="EmphasisTypeItalic">il</em>
                </sub>] where the row indices correspond to network nodes (<em xmlns="" class="EmphasisTypeItalic">i</em> = 1, . . ., <em xmlns="" class="EmphasisTypeItalic">n</em>) and the column indices (<em xmlns="" class="EmphasisTypeItalic">l</em> = 1, . . ., <em xmlns="" class="EmphasisTypeItalic">m</em>) correspond to sample measurements:<div id="Equ1" class="Equation EquationMathjax"><div class="EquationContent"><span class="MathJax_Preview" style=""></span><span class="MathJax" id="MathJax-Element-1-Frame" tabindex="0" data-mathml="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot;&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mi&gt;X&lt;/mi&gt;&lt;mo&gt;=&lt;/mo&gt;&lt;mo stretchy=&quot;false&quot;&gt;[&lt;/mo&gt;&lt;msub&gt;&lt;mi&gt;x&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mi&gt;j&lt;/mi&gt;&lt;/mrow&gt;&lt;/msub&gt;&lt;mo stretchy=&quot;false&quot;&gt;]&lt;/mo&gt;&lt;mo&gt;=&lt;/mo&gt;&lt;mrow&gt;&lt;mo&gt;(&lt;/mo&gt;&lt;mrow&gt;&lt;mtable&gt;&lt;mtr&gt;&lt;mtd&gt;&lt;mrow&gt;&lt;msub&gt;&lt;mi&gt;x&lt;/mi&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;/msub&gt;&lt;/mrow&gt;&lt;/mtd&gt;&lt;/mtr&gt;&lt;mtr&gt;&lt;mtd&gt;&lt;mrow&gt;&lt;msub&gt;&lt;mi&gt;x&lt;/mi&gt;&lt;mn&gt;2&lt;/mn&gt;&lt;/msub&gt;&lt;/mrow&gt;&lt;/mtd&gt;&lt;/mtr&gt;&lt;mtr&gt;&lt;mtd&gt;&lt;mo&gt;&amp;#x22EF;&lt;/mo&gt;&lt;/mtd&gt;&lt;/mtr&gt;&lt;mtr&gt;&lt;mtd&gt;&lt;mrow&gt;&lt;msub&gt;&lt;mi&gt;x&lt;/mi&gt;&lt;mi&gt;n&lt;/mi&gt;&lt;/msub&gt;&lt;/mrow&gt;&lt;/mtd&gt;&lt;/mtr&gt;&lt;/mtable&gt;&lt;/mrow&gt;&lt;mo&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/mrow&gt;&lt;annotation encoding=&quot;MathType-MTEF&quot;&gt;          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemiwaGLaeyypa0Jaei4waSLaemiEaG3aaSbaaSqaaiabdMgaPjabdQgaQbqabaGccqGGDbqxcqGH9aqpdaqadaqaauaabeqaeeaaaaqaaiabdIha4naaBaaaleaacqaIXaqmaeqaaaGcbaGaemiEaG3aaSbaaSqaaiabikdaYaqabaaakeaacqWIVlctaeaacqWG4baEdaWgaaWcbaGaemOBa4gabeaaaaaakiaawIcacaGLPaaaaaa@422B@&lt;/annotation&gt;&lt;/semantics&gt;&lt;/mrow&gt;&lt;/semantics&gt;&lt;/math&gt;" role="presentation" style="position: relative;"><nobr aria-hidden="true"><span class="math" id="MathJax-Span-1" style="width: 9.436em; display: inline-block;"><span style="display: inline-block; position: relative; width: 8.482em; height: 0px; font-size: 111%;"><span style="position: absolute; clip: rect(2.016em, 1008.16em, 7.634em, -999.997em); top: -5.085em; left: 0em;"><span class="mrow" id="MathJax-Span-2"><span class="semantics" id="MathJax-Span-3"><span class="mrow" id="MathJax-Span-4"><span class="semantics" id="MathJax-Span-5"><span class="mrow" id="MathJax-Span-6"><span class="mi" id="MathJax-Span-7" style="font-family: MathJax_Math-italic;">X<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span><span class="mo" id="MathJax-Span-8" style="font-family: MathJax_Main; padding-left: 0.268em;">=</span><span class="mo" id="MathJax-Span-9" style="font-family: MathJax_Main; padding-left: 0.268em;">[</span><span class="msub" id="MathJax-Span-10"><span style="display: inline-block; position: relative; width: 1.169em; height: 0px;"><span style="position: absolute; clip: rect(3.394em, 1000.53em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-11" style="font-family: MathJax_Math-italic;">x</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.813em; left: 0.586em;"><span class="mrow" id="MathJax-Span-12"><span class="mi" id="MathJax-Span-13" style="font-size: 70.7%; font-family: MathJax_Math-italic;">i</span><span class="mi" id="MathJax-Span-14" style="font-size: 70.7%; font-family: MathJax_Math-italic;">j</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mo" id="MathJax-Span-15" style="font-family: MathJax_Main;">]</span><span class="mo" id="MathJax-Span-16" style="font-family: MathJax_Main; padding-left: 0.268em;">=</span><span class="mrow" id="MathJax-Span-17" style="padding-left: 0.268em;"><span class="mo" id="MathJax-Span-18" style="vertical-align: 2.917em;"><span style="display: inline-block; position: relative; width: 0.904em; height: 0px;"><span style="position: absolute; font-family: MathJax_Size4; top: -2.806em; left: 0em;">⎛<span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; font-family: MathJax_Size4; top: 0.692em; left: 0em;">⎝<span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="font-family: MathJax_Size4; position: absolute; top: -1.587em; left: 0em;">⎜<span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="font-family: MathJax_Size4; position: absolute; top: -1.004em; left: 0em;">⎜<span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="font-family: MathJax_Size4; position: absolute; top: -0.474em; left: 0em;">⎜<span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mrow" id="MathJax-Span-19"><span class="mtable" id="MathJax-Span-20"><span style="display: inline-block; position: relative; width: 1.169em; height: 0px; margin-right: 0.162em; margin-left: 0.162em;"><span style="position: absolute; clip: rect(3.182em, 1001.12em, 8.376em, -999.997em); top: -5.88em; left: 0em;"><span style="display: inline-block; position: relative; width: 1.169em; height: 0px;"><span style="position: absolute; clip: rect(3.394em, 1001.01em, 4.295em, -999.997em); top: -6.092em; left: 50%; margin-left: -0.474em;"><span class="mtd" id="MathJax-Span-21"><span class="mrow" id="MathJax-Span-22"><span class="mrow" id="MathJax-Span-23"><span class="msub" id="MathJax-Span-24"><span style="display: inline-block; position: relative; width: 1.01em; height: 0px;"><span style="position: absolute; clip: rect(3.394em, 1000.53em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-25" style="font-family: MathJax_Math-italic;">x</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.813em; left: 0.586em;"><span class="mn" id="MathJax-Span-26" style="font-size: 70.7%; font-family: MathJax_Main;">1</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.394em, 1001.01em, 4.295em, -999.997em); top: -4.661em; left: 50%; margin-left: -0.474em;"><span class="mtd" id="MathJax-Span-27"><span class="mrow" id="MathJax-Span-28"><span class="mrow" id="MathJax-Span-29"><span class="msub" id="MathJax-Span-30"><span style="display: inline-block; position: relative; width: 1.01em; height: 0px;"><span style="position: absolute; clip: rect(3.394em, 1000.53em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-31" style="font-family: MathJax_Math-italic;">x</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.813em; left: 0.586em;"><span class="mn" id="MathJax-Span-32" style="font-size: 70.7%; font-family: MathJax_Main;">2</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.5em, 1001.12em, 3.924em, -999.997em); top: -3.23em; left: 50%; margin-left: -0.58em;"><span class="mtd" id="MathJax-Span-33"><span class="mrow" id="MathJax-Span-34"><span class="mo" id="MathJax-Span-35" style="font-family: MathJax_Main;">⋯</span></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.394em, 1001.06em, 4.295em, -999.997em); top: -1.799em; left: 50%; margin-left: -0.527em;"><span class="mtd" id="MathJax-Span-36"><span class="mrow" id="MathJax-Span-37"><span class="mrow" id="MathJax-Span-38"><span class="msub" id="MathJax-Span-39"><span style="display: inline-block; position: relative; width: 1.063em; height: 0px;"><span style="position: absolute; clip: rect(3.394em, 1000.53em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-40" style="font-family: MathJax_Math-italic;">x</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.813em; left: 0.586em;"><span class="mi" id="MathJax-Span-41" style="font-size: 70.7%; font-family: MathJax_Math-italic;">n</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span><span style="display: inline-block; width: 0px; height: 5.885em;"></span></span></span></span></span><span class="mo" id="MathJax-Span-42" style="vertical-align: 2.917em;"><span style="display: inline-block; position: relative; width: 0.904em; height: 0px;"><span style="position: absolute; font-family: MathJax_Size4; top: -2.806em; left: 0em;">⎞<span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; font-family: MathJax_Size4; top: 0.692em; left: 0em;">⎠<span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="font-family: MathJax_Size4; position: absolute; top: -1.587em; left: 0em;">⎟<span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="font-family: MathJax_Size4; position: absolute; top: -1.004em; left: 0em;">⎟<span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="font-family: MathJax_Size4; position: absolute; top: -0.474em; left: 0em;">⎟<span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 5.09em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -2.703em; border-left: 0px solid; width: 0px; height: 6.003em;"></span></span></nobr><span class="MJX_Assistive_MathML" role="presentation"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><semantics><mrow><mi>X</mi><mo>=</mo><mo stretchy="false">[</mo><msub><mi>x</mi><mrow><mi>i</mi><mi>j</mi></mrow></msub><mo stretchy="false">]</mo><mo>=</mo><mrow><mo>(</mo><mrow><mtable><mtr><mtd><mrow><msub><mi>x</mi><mn>1</mn></msub></mrow></mtd></mtr><mtr><mtd><mrow><msub><mi>x</mi><mn>2</mn></msub></mrow></mtd></mtr><mtr><mtd><mo>⋯</mo></mtd></mtr><mtr><mtd><mrow><msub><mi>x</mi><mi>n</mi></msub></mrow></mtd></mtr></mtable></mrow><mo>)</mo></mrow></mrow><annotation encoding="MathType-MTEF">          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemiwaGLaeyypa0Jaei4waSLaemiEaG3aaSbaaSqaaiabdMgaPjabdQgaQbqabaGccqGGDbqxcqGH9aqpdaqadaqaauaabeqaeeaaaaqaaiabdIha4naaBaaaleaacqaIXaqmaeqaaaGcbaGaemiEaG3aaSbaaSqaaiabikdaYaqabaaakeaacqWIVlctaeaacqWG4baEdaWgaaWcbaGaemOBa4gabeaaaaaakiaawIcacaGLPaaaaaa@422B@</annotation></semantics></mrow></semantics></math></span></span><script type="math/mml" id="MathJax-Element-1"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow>
                      <semantics>
                        <mrow>
                          <mi>X</mi>
                          <mo>=</mo>
                          <mo stretchy="false">[</mo>
                          <msub>
                            <mi>x</mi>
                            <mrow>
                              <mi>i</mi>
                              <mi>j</mi>
                            </mrow>
                          </msub>
                          <mo stretchy="false">]</mo>
                          <mo>=</mo>
                          <mrow>
                            <mo>(</mo>
                            <mrow>
                              <mtable>
                                <mtr>
                                  <mtd>
                                    <mrow>
                                      <msub>
                                        <mi>x</mi>
                                        <mn>1</mn>
                                      </msub>
                                    </mrow>
                                  </mtd>
                                </mtr>
                                <mtr>
                                  <mtd>
                                    <mrow>
                                      <msub>
                                        <mi>x</mi>
                                        <mn>2</mn>
                                      </msub>
                                    </mrow>
                                  </mtd>
                                </mtr>
                                <mtr>
                                  <mtd>
                                    <mo>⋯</mo>
                                  </mtd>
                                </mtr>
                                <mtr>
                                  <mtd>
                                    <mrow>
                                      <msub>
                                        <mi>x</mi>
                                        <mi>n</mi>
                                      </msub>
                                    </mrow>
                                  </mtd>
                                </mtr>
                              </mtable>
                            </mrow>
                            <mo>)</mo>
                          </mrow>
                        </mrow>
                        <annotation encoding="MathType-MTEF">MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemiwaGLaeyypa0Jaei4waSLaemiEaG3aaSbaaSqaaiabdMgaPjabdQgaQbqabaGccqGGDbqxcqGH9aqpdaqadaqaauaabeqaeeaaaaqaaiabdIha4naaBaaaleaacqaIXaqmaeqaaaGcbaGaemiEaG3aaSbaaSqaaiabikdaYaqabaaakeaacqWIVlctaeaacqWG4baEdaWgaaWcbaGaemOBa4gabeaaaaaakiaawIcacaGLPaaaaaa@422B@</annotation>
                      </semantics>
                    </mrow></semantics></math></script></div> <div class="EquationNumber">(1)</div></div></div><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">We refer to the <em xmlns="" class="EmphasisTypeItalic">i</em>-th row <em xmlns="" class="EmphasisTypeItalic">x</em><sub>
                  <em xmlns="" class="EmphasisTypeItalic">i</em>
                </sub>as the <em xmlns="" class="EmphasisTypeItalic">i</em>-th <em xmlns="" class="EmphasisTypeItalic">node profile</em> across <em xmlns="" class="EmphasisTypeItalic">m</em> sample measurements.</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Sometimes a quantitative measure (referred to as <em xmlns="" class="EmphasisTypeItalic">sample trait</em>) is provided for the columns of <em xmlns="" class="EmphasisTypeItalic">X</em>. For example, <em xmlns="" class="EmphasisTypeItalic">T</em> = (<em xmlns="" class="EmphasisTypeItalic">T</em><sub>1</sub>, . . ., <em xmlns="" class="EmphasisTypeItalic">T</em><sub>
                  <em xmlns="" class="EmphasisTypeItalic">m</em>
                </sub>) could measure survival time or it could be a binary indicator variable (disease status). Abstractly speaking, we define a sample trait <em xmlns="" class="EmphasisTypeItalic">T</em> as a vector with <em xmlns="" class="EmphasisTypeItalic">m</em> components that correspond to the columns of the data matrix <em xmlns="" class="EmphasisTypeItalic">X</em>. A sample trait can be used to define a node significance measure. For example, a <em xmlns="" class="EmphasisTypeItalic">trait-based node significance measure</em> can be defined as the absolute value of the correlation between the <em xmlns="" class="EmphasisTypeItalic">i</em>-th node profile <em xmlns="" class="EmphasisTypeItalic">x</em><sub>
                  <em xmlns="" class="EmphasisTypeItalic">i</em>
                </sub>and the sample trait</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para"><em xmlns="" class="EmphasisTypeItalic">T</em>:<em xmlns="" class="EmphasisTypeItalic">GS</em><sub>
                  <em xmlns="" class="EmphasisTypeItalic">i</em>
                </sub>= |<em xmlns="" class="EmphasisTypeItalic">cor</em>(<em xmlns="" class="EmphasisTypeItalic">x</em><sub>
                  <em xmlns="" class="EmphasisTypeItalic">i</em>
                </sub>, <em xmlns="" class="EmphasisTypeItalic">T</em>)|.(2)</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Alternatively, a correlation test p-value [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR1" data-toggle="citation">1</a></span>] or a regression-based p-value for assessing the statistical significance between <em xmlns="" class="EmphasisTypeItalic">x</em><sub>
                  <em xmlns="" class="EmphasisTypeItalic">i</em>
                </sub>and the sample trait <em xmlns="" class="EmphasisTypeItalic">T</em> can be used to define a p-value based node significance measure, for example by defining</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para"><em xmlns="" class="EmphasisTypeItalic">GS</em><sub>
                  <em xmlns="" class="EmphasisTypeItalic">i</em>
                </sub>= -log <em xmlns="" class="EmphasisTypeItalic">p</em><sub>
                  <em xmlns="" class="EmphasisTypeItalic">i</em>
                </sub>.(3)</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">The rationale behind correlation network methodology is to use network language to describe the pairwise relationships (correlations) between the rows of <em xmlns="" class="EmphasisTypeItalic">X</em> (Equation 1). Although other statistical techniques exist for analyzing correlation matrices, network language is particularly intuitive to biologists and allows for simple social network analogies. Correlation networks can be used to address many analysis goals including the following. First, correlation networks can be used to find clusters (modules) of interconnected nodes. Thus, a network module is a set of rows of <em xmlns="" class="EmphasisTypeItalic">X</em> (Equation 1) which are closely connected according to a suitably defined measure of interconnectedness.</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">A second analysis goal is to summarize the node profiles of a given module by a representative, e.g. a highly connected hub node, which is centrally located in the module. Focusing the analysis on module or their representatives amounts to a network-based data reduction method. Relating modules instead of nodes to a sample trait can alleviate the multiple testing problem.</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">A third analysis goal is to identify 'significant' modules. Toward this end, a node significance measure can be used to identify modules with high average node significance (referred to as module significance).</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">A fourth analysis goal is to annotate all network nodes with respect to how close they are to the identified modules. This can be accomplished by defining a fuzzy measure of module memberships that generalizes the binary module membership indicator to a quantitative measure. Fuzzy measures of module membership can be used to identify nodes that lie intermediate between and close to two or more modules.</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">A fifth analysis goal is to define the network neighborhood of a given seed set of nodes. Intuitively speaking, a neighborhood is composed of nodes that are highly connected to a given set of nodes. Thus, neighborhood analysis facilitates a guilt-by-association screening strategy for finding nodes that interact with a given set of interesting nodes.</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">A sixth analysis goal is to screen for nodes based on node screening criteria which can be based on a node significance measure, on module membership information, on network topological properties (e.g. high connectivity), etc.</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">A seventh analysis goal is to contrast one network with another network. This differential network analysis can be used to identify changes in connectivity patterns or module structure between different conditions. An eighth analysis goal is to find shared modules between two or more networks (consensus module analysis). Since by definition consensus modules are building blocks in multiple networks, they may represent fundamental structural properties of the network.</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">The above incomplete enumeration of analysis goals shows that correlation networks can be used as a data exploratory technique (similar to cluster analysis, factor analysis, or other dimensional reduction techniques) and as a screening method. For example, correlation networks can be used to screen for modules and intramodular hubs that relate to a sample trait. Correlation networks allow one to generate testable hypotheses that should be validated in independent data or in designed validation experiments.</p><section id="Sec2" class="Section2 RenderAsSection2">
                <h3 class="Heading">Gene Co-Expression Networks</h3>
                <div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">In the following, we focus on gene co-expression networks which represent a major application of correlation network methodology. Co-expression networks have been found useful for describing the pairwise relationships among gene transcripts [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR2" data-toggle="citation">2</a></span><span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR3" data-toggle="citation">3</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR4" data-toggle="citation">4</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR5" data-toggle="citation">5</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR6" data-toggle="citation">6</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR7" data-toggle="citation">7</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR8" data-toggle="citation">8</a></span>, </span><span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR9" data-toggle="citation">9</a></span>]. In co-expression networks, we refer to nodes as 'genes', to the node profile <em xmlns="" class="EmphasisTypeItalic">x</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">i</em>
                  </sub>as the gene expression profile, and to the node significance measure <em xmlns="" class="EmphasisTypeItalic">GS</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">i</em>
                  </sub>as the gene significance measure. A glossary of important network-related terms can be found in Table <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Tab1">1</a></span>. Here we introduce an R software package that summarizes and extends our earlier work on weighted gene co-expression network analysis (WGCNA) [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR5" data-toggle="citation">5</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR10" data-toggle="citation">10</a></span><span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR11" data-toggle="citation">11</a></span>, </span><span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR12" data-toggle="citation">12</a></span>]. WGCNA has been used to analyze gene expression data from brain cancer [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR10" data-toggle="citation">10</a></span>], yeast cell cycle [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR13" data-toggle="citation">13</a></span>], mouse genetics [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR14" data-toggle="citation">14</a></span><span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR15" data-toggle="citation">15</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR16" data-toggle="citation">16</a></span>, </span><span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR17" data-toggle="citation">17</a></span>], primate brain tissue [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR18" data-toggle="citation">18</a></span><span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR19" data-toggle="citation">19</a></span>, </span><span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR20" data-toggle="citation">20</a></span>], diabetes [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR21" data-toggle="citation">21</a></span>], chronic fatigue patients [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR22" data-toggle="citation">22</a></span>] and plants [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR23" data-toggle="citation">23</a></span>]. While these publications have made R software code available in various forms, there is a need for a comprehensive R package that summarizes and standardizes methods and functions. To address this need, we introduce the WGCNA R package which also includes enhanced and novel functions for co-expression network analysis.<figure xmlns="" id="Tab1" class="FigureTable"><figcaption class="Caption" lang="en"><div xmlns="http://www.w3.org/1999/xhtml" class="CaptionContent"><span class="CaptionNumber">Table 1</span>
                        <p xmlns="" class="SimplePara">Glossary of WGCNA Terminology.</p>
                      </div></figcaption><div class="Table"><table xmlns="http://www.w3.org/1999/xhtml" border="1"><colgroup><col></colgroup><colgroup><col></colgroup><thead><tr class="header"><th><p xmlns="" class="SimplePara">Term</p></th><th><p xmlns="" class="SimplePara">Definition</p></th></tr></thead><tbody><tr class="noclass"><td><p xmlns="" class="SimplePara">Co-expression network</p></td><td><p xmlns="" class="SimplePara">We define co-expression networks as undirected, weighted gene networks. The nodes of such a network correspond to gene expression profiles, and edges between genes are determined by the pairwise correlations between gene expressions. By raising the absolute value of the correlation to a power <em class="EmphasisTypeItalic">β</em> ≥ 1 (soft thresholding), the weighted gene co-expression network construction emphasizes high correlations at the expense of low correlations. Specifically, <em class="EmphasisTypeItalic">a</em><sub xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">ij</em>
                              </sub>= |cor(<em class="EmphasisTypeItalic">x</em><sub xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">i</em>
                              </sub>, <em class="EmphasisTypeItalic">x</em><sub xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">j</em>
                              </sub>)|<sup xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">β</em>
                              </sup>represents the adjacency of an unsigned network. Optionally, the user can also specify a signed co-expression network where the adjacency is defined as <em class="EmphasisTypeItalic">a</em><sub xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">ij</em>
                              </sub>= |(1 + cor(<em class="EmphasisTypeItalic">x</em><sub xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">i</em>
                              </sub>, <em class="EmphasisTypeItalic">x</em><sub xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">j</em>
                              </sub>))<em class="EmphasisTypeItalic">/</em> 2|<sup xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">β</em>
                              </sup>.</p></td></tr><tr class="noclass"><td><p xmlns="" class="SimplePara">Module</p></td><td><p xmlns="" class="SimplePara">Modules are clusters of highly interconnected genes. In an unsigned co-expression network, modules correspond to clusters of genes with high absolute correlations. In a signed network, modules correspond to positively correlated genes.</p></td></tr><tr class="noclass"><td><p xmlns="" class="SimplePara">Connectivity</p></td><td><p xmlns="" class="SimplePara">For each gene, the connectivity (also known as degree) is defined as the sum of connection strengths with the other network genes: <em class="EmphasisTypeItalic">k</em><sub xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">i</em>
                              </sub>= ∑<sub xmlns="http://www.w3.org/1999/xhtml"><em xmlns="" class="EmphasisTypeItalic">u</em>≠<em xmlns="" class="EmphasisTypeItalic">i</em></sub><em class="EmphasisTypeItalic">a</em><sub xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">ui</em>
                              </sub>. In co-expression networks, the connectivity measures how correlated a gene is with all other network genes.</p></td></tr><tr class="noclass"><td><p xmlns="" class="SimplePara">Intramodular connectivity <em class="EmphasisTypeItalic">k</em><sub xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">IM</em>
                              </sub></p></td><td><p xmlns="" class="SimplePara">Intramodular connectivity measures how connected, or co-expressed, a given gene is with respect to the genes of a particular module. The intramodular connectivity may be interpreted as a measure of module membership.</p></td></tr><tr class="noclass"><td><p xmlns="" class="SimplePara">Module eigengene <em class="EmphasisTypeItalic">E</em></p></td><td><p xmlns="" class="SimplePara">The module eigengene <em class="EmphasisTypeItalic">E</em> is defined as the first principal component of a given module. It can be considered a representative of the gene expression profiles in a module.</p></td></tr><tr class="noclass"><td><p xmlns="" class="SimplePara">Eigengene significance</p></td><td><p xmlns="" class="SimplePara">When a microarray sample trait <em class="EmphasisTypeItalic">y</em> is available (e.g. case control status or body weight), one can correlate the module eigengenes with this outcome. The correlation coefficient is referred to as eigengene significance.</p></td></tr><tr class="noclass"><td><p xmlns="" class="SimplePara">Module Membership, also known as eigengene-based connectivity <em class="EmphasisTypeItalic">k</em><sub xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">ME</em>
                              </sub></p></td><td><p xmlns="" class="SimplePara">For each gene, we define a "fuzzy" measure of module membership by correlating its gene expression profile with the module eigengene of a given module. For example, <em class="EmphasisTypeItalic">MM</em><sup xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">blue</em>
                              </sup>(<em class="EmphasisTypeItalic">i</em>) = <span xmlns="http://www.w3.org/1999/xhtml" class="InlineEquation" id="IEq1"><span class="MathJax_Preview" style=""></span><span class="MathJax" id="MathJax-Element-2-Frame" tabindex="0" data-mathml="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot;&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;msubsup&gt;&lt;mi&gt;K&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;c&lt;/mi&gt;&lt;mi&gt;o&lt;/mi&gt;&lt;mi&gt;r&lt;/mi&gt;&lt;mo&gt;,&lt;/mo&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/mrow&gt;&lt;mrow&gt;&lt;mi&gt;b&lt;/mi&gt;&lt;mi&gt;l&lt;/mi&gt;&lt;mi&gt;u&lt;/mi&gt;&lt;mi&gt;e&lt;/mi&gt;&lt;/mrow&gt;&lt;/msubsup&gt;&lt;/mrow&gt;&lt;annotation encoding=&quot;MathType-MTEF&quot;&gt;          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabdkgaIjabdYgaSjabdwha1jabdwgaLbaaaaa@38F3@&lt;/annotation&gt;&lt;/semantics&gt;&lt;/mrow&gt;&lt;/semantics&gt;&lt;/math&gt;" role="presentation" style="position: relative;"><nobr aria-hidden="true"><span class="math" id="MathJax-Span-43" style="width: 2.577em; display: inline-block;"><span style="display: inline-block; position: relative; width: 2.32em; height: 0px; font-size: 111%;"><span style="position: absolute; clip: rect(1.097em, 1002.32em, 2.77em, -999.997em); top: -2.12em; left: 0em;"><span class="mrow" id="MathJax-Span-44"><span class="semantics" id="MathJax-Span-45"><span class="mrow" id="MathJax-Span-46"><span class="semantics" id="MathJax-Span-47"><span class="mrow" id="MathJax-Span-48"><span class="msubsup" id="MathJax-Span-49"><span style="display: inline-block; position: relative; width: 2.32em; height: 0px;"><span style="position: absolute; clip: rect(3.092em, 1000.9em, 4.186em, -999.997em); top: -3.986em; left: 0em;"><span class="mi" id="MathJax-Span-50" style="font-family: MathJax_Math-italic;">K<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.068em;"></span></span><span style="display: inline-block; width: 0px; height: 3.993em;"></span></span><span style="position: absolute; clip: rect(3.285em, 1001.36em, 4.186em, -999.997em); top: -4.308em; left: 0.968em;"><span class="mrow" id="MathJax-Span-51"><span class="mi" id="MathJax-Span-52" style="font-size: 70.7%; font-family: MathJax_Math-italic;">b</span><span class="mi" id="MathJax-Span-53" style="font-size: 70.7%; font-family: MathJax_Math-italic;">l</span><span class="mi" id="MathJax-Span-54" style="font-size: 70.7%; font-family: MathJax_Math-italic;">u</span><span class="mi" id="MathJax-Span-55" style="font-size: 70.7%; font-family: MathJax_Math-italic;">e</span></span><span style="display: inline-block; width: 0px; height: 3.993em;"></span></span><span style="position: absolute; clip: rect(3.349em, 1001.48em, 4.315em, -999.997em); top: -3.665em; left: 0.84em;"><span class="mrow" id="MathJax-Span-56"><span class="mi" id="MathJax-Span-57" style="font-size: 70.7%; font-family: MathJax_Math-italic;">c</span><span class="mi" id="MathJax-Span-58" style="font-size: 70.7%; font-family: MathJax_Math-italic;">o</span><span class="mi" id="MathJax-Span-59" style="font-size: 70.7%; font-family: MathJax_Math-italic;">r</span><span class="mo" id="MathJax-Span-60" style="font-size: 70.7%; font-family: MathJax_Main;">,</span><span class="mi" id="MathJax-Span-61" style="font-size: 70.7%; font-family: MathJax_Math-italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.993em;"></span></span></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 2.127em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -0.568em; border-left: 0px solid; width: 0px; height: 1.575em;"></span></span></nobr><span class="MJX_Assistive_MathML" role="presentation"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><semantics><mrow><msubsup><mi>K</mi><mrow><mi>c</mi><mi>o</mi><mi>r</mi><mo>,</mo><mi>i</mi></mrow><mrow><mi>b</mi><mi>l</mi><mi>u</mi><mi>e</mi></mrow></msubsup></mrow><annotation encoding="MathType-MTEF">          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabdkgaIjabdYgaSjabdwha1jabdwgaLbaaaaa@38F3@</annotation></semantics></mrow></semantics></math></span></span><script type="math/mml" id="MathJax-Element-2"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow>
                                    <semantics>
                                      <mrow>
                                        <msubsup>
                                          <mi>K</mi>
                                          <mrow>
                                            <mi>c</mi>
                                            <mi>o</mi>
                                            <mi>r</mi>
                                            <mo>,</mo>
                                            <mi>i</mi>
                                          </mrow>
                                          <mrow>
                                            <mi>b</mi>
                                            <mi>l</mi>
                                            <mi>u</mi>
                                            <mi>e</mi>
                                          </mrow>
                                        </msubsup>
                                      </mrow>
                                      <annotation encoding="MathType-MTEF">MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabdkgaIjabdYgaSjabdwha1jabdwgaLbaaaaa@38F3@</annotation>
                                    </semantics>
                                  </mrow></semantics></math></script></span> = cor(<em class="EmphasisTypeItalic">x</em><sub xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">i</em>
                              </sub>, <em class="EmphasisTypeItalic">E</em><sup xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">blue</em>
                              </sup>) measures how correlated gene <em class="EmphasisTypeItalic">i</em> is to the blue module eigengene. <em class="EmphasisTypeItalic">MM</em><sup xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">blue</em>
                              </sup>(<em class="EmphasisTypeItalic">i</em>) measures the membership of the <em class="EmphasisTypeItalic">i</em>-th gene with respect to the blue module. If <em class="EmphasisTypeItalic">MM</em><sup xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">blue</em>
                              </sup>(<em class="EmphasisTypeItalic">i</em>) is close to 0, the <em class="EmphasisTypeItalic">i</em>-th gene is not part of the blue module. On the other hand, if <em class="EmphasisTypeItalic">MM</em><sup xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">blue</em>
                              </sup>(<em class="EmphasisTypeItalic">i</em>) is close to 1 or -1, it is highly connected to the blue module genes. The sign of module membership encodes whether the gene has a positive or a negative relationship with the blue module eigengene. The module membership measure can be defined for all input genes (irrespective of their original module membership). It turns out that the module membership measure is highly related to the intramodular connectivity <em class="EmphasisTypeItalic">k</em><sub xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">IM</em>
                              </sub>. Highly connected intramodular hub genes tend to have high module membership values to the respective module.</p></td></tr><tr class="noclass"><td><p xmlns="" class="SimplePara">Hub gene</p></td><td><p xmlns="" class="SimplePara">This loosely defined term is used as an abbreviation of "highly connected gene." By definition, genes inside co-expression modules tend to have high connectivity.</p></td></tr><tr class="noclass"><td><p xmlns="" class="SimplePara">Gene significance <em class="EmphasisTypeItalic">GS</em></p></td><td><p xmlns="" class="SimplePara">To incorporate external information into the co-expression network, we make use of gene significance measures. Abstractly speaking, the higher the absolute value of <em class="EmphasisTypeItalic">GS</em><sub xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">i</em>
                              </sub>, the more biologically significant is the <em class="EmphasisTypeItalic">i</em>-th gene. For example, <em class="EmphasisTypeItalic">GS</em><sub xmlns="http://www.w3.org/1999/xhtml">
                                <em xmlns="" class="EmphasisTypeItalic">i</em>
                              </sub>could encode pathway membership (e.g. 1 if the gene is a known apoptosis gene and 0 otherwise), knockout essentiality, or the correlation with an external microarray sample trait. A gene significance measure could also be defined by minus log of a p-value. The only requirement is that gene significance of 0 indicates that the gene is not significant with regard to the biological question of interest. The gene significance can take on positive or negative values.</p></td></tr><tr class="noclass"><td><p xmlns="" class="SimplePara">Module significance</p></td><td><p xmlns="" class="SimplePara">Module significance is determined as the average absolute gene significance measure for all genes in a given module. When gene significance is defined as the correlation of gene expression profiles with an external trait <em class="EmphasisTypeItalic">y</em>, this measure tends to be highly related to the correlation between the module eigengene and <em class="EmphasisTypeItalic">y</em>.</p></td></tr></tbody></table></div></figure></div>
              </section></div></section>
            <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="Sec3" class="Section1 RenderAsSection1 c-section"><h2 class="Heading js-ToggleCollapseSection c-section__heading" data-sticky-update="true"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Results" data-toggle="collapse" data-toggle-disable-clickoff="" data-target="Results">Results</a></h2><div class="js-CollapseSection CollapseSection c-section__content" id="Results" data-test="collapse-section"><div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Figure <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig1">1</a></span> provides an overview of typical analysis steps and the rationale behind them. To determine whether a co-expression module is biologically meaningful, one can use functional enrichment and gene ontology information.<figure xmlns="" class="Figure" id="Fig1"><div xmlns="http://www.w3.org/1999/xhtml" class="MediaObject">
                    <a target="_blank" rel="noopener" href="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_Article_2544_Fig1_HTML.jpg"><img class="c-lazy-image u-js-shown" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/12859_2008_Article_2544_Fig1_HTML.jpg" alt="Figure 1" data-lazy-load="true" data-src="https://media.springernature.com/lw785/springer-static/image/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_Article_2544_Fig1_HTML.jpg"><noscript><img src="https://media.springernature.com/lw785/springer-static/image/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_Article_2544_Fig1_HTML.jpg" alt="Figure 1"></noscript></a>
                  </div><figcaption class="Caption" lang="en"><div xmlns="http://www.w3.org/1999/xhtml" class="CaptionContent"><span class="CaptionNumber">Figure 1</span>
                      <p xmlns="" class="SimplePara"><strong class="EmphasisTypeBold">Overview of WGCNA methodology</strong>. This flowchart presents a brief overview of the main steps of Weighted Gene Co-expression Network Analysis.</p>
                    </div></figcaption></figure></div><section id="Sec4" class="Section2 RenderAsSection2">
                <h3 class="Heading">Overview of functions included in the WGCNA package</h3>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">The WGCNA package contains a comprehensive set of functions for performing a correlation network analysis of large, high-dimensional data sets. Functions in the WGCNA package can be divided into the following categories: 1. network construction; 2. module detection; 3. module and gene selection; 4. calculations of topological properties; 5. data simulation; 6. visualization; 7. interfacing with external software packages. An exhaustive list of implemented functions together with detailed descriptions is provided in the R package manual posted on our web site. Here we briefly outline the main functionality of the package and highlight new contributions.</p>
              </section><section id="Sec5" class="Section2 RenderAsSection2">
                <h3 class="Heading">Category 1: Functions for network construction</h3>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">A network is fully specified by its <em xmlns="" class="EmphasisTypeItalic">adjacency matrix a</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">ij</em>
                  </sub>, a symmetric <em xmlns="" class="EmphasisTypeItalic">n</em> × <em xmlns="" class="EmphasisTypeItalic">n</em> matrix with entries in [0, 1] whose component <em xmlns="" class="EmphasisTypeItalic">a</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">ij</em>
                  </sub>encodes the network connection strength between nodes <em xmlns="" class="EmphasisTypeItalic">i</em> and <em xmlns="" class="EmphasisTypeItalic">j</em>. To calculate the adjacency matrix, an intermediate quantity called the <em xmlns="" class="EmphasisTypeItalic">co-expression similarity s</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">ij</em>
                  </sub>is first defined. The default method defines the co-expression similarity <em xmlns="" class="EmphasisTypeItalic">s</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">ij</em>
                  </sub>as the absolute value of the correlation coefficient between the profiles of nodes <em xmlns="" class="EmphasisTypeItalic">i</em> and <em xmlns="" class="EmphasisTypeItalic">j</em>:<em xmlns="" class="EmphasisTypeItalic">s</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">ij</em>
                  </sub>= |<em xmlns="" class="EmphasisTypeItalic">cor</em>(<em xmlns="" class="EmphasisTypeItalic">x</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">i</em>
                  </sub>, <em xmlns="" class="EmphasisTypeItalic">x</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">j</em>
                  </sub>)|.</p>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">The WGCNA package also implements alternative co-expression measures, e.g. more robust measures of correlation (the biweight midcorrelation [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR24" data-toggle="citation">24</a></span>] or the Spearman correlation). A signed co-expression measure can be defined to keep track of the sign of the co-expression information. For convenience, we define the co-expression similarity measure such that it takes on values in [0, 1].</p>
                <div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Using a thresholding procedure, the co-expression similarity is transformed into the adjacency. An unweighted network adjacency <em xmlns="" class="EmphasisTypeItalic">a</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">ij</em>
                  </sub>between gene expression profiles <em xmlns="" class="EmphasisTypeItalic">x</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">i</em>
                  </sub>and <em xmlns="" class="EmphasisTypeItalic">x</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">j</em>
                  </sub>can be defined by hard thresholding the co-expression similarity <em xmlns="" class="EmphasisTypeItalic">s</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">ij</em>
                  </sub>as<div id="Equ2" class="Equation EquationMathjax"><div class="EquationContent"><span class="MathJax_Preview" style=""></span><span class="MathJax" id="MathJax-Element-3-Frame" tabindex="0" data-mathml="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot;&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;msub&gt;&lt;mi&gt;a&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mi&gt;j&lt;/mi&gt;&lt;/mrow&gt;&lt;/msub&gt;&lt;mo&gt;=&lt;/mo&gt;&lt;mrow&gt;&lt;mo&gt;{&lt;/mo&gt;&lt;mrow&gt;&lt;mtable columnalign=&quot;left&quot;&gt;&lt;mtr columnalign=&quot;left&quot;&gt;&lt;mtd columnalign=&quot;left&quot;&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;/mtd&gt;&lt;mtd columnalign=&quot;left&quot;&gt;&lt;mrow&gt;&lt;mtext&gt;if&amp;#xA0;&lt;/mtext&gt;&lt;msub&gt;&lt;mi&gt;s&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mi&gt;j&lt;/mi&gt;&lt;/mrow&gt;&lt;/msub&gt;&lt;mo&gt;&amp;#x2265;&lt;/mo&gt;&lt;mi&gt;&amp;#x3C4;&lt;/mi&gt;&lt;mo&gt;;&lt;/mo&gt;&lt;/mrow&gt;&lt;/mtd&gt;&lt;/mtr&gt;&lt;mtr columnalign=&quot;left&quot;&gt;&lt;mtd columnalign=&quot;left&quot;&gt;&lt;mn&gt;0&lt;/mn&gt;&lt;/mtd&gt;&lt;mtd columnalign=&quot;left&quot;&gt;&lt;mrow&gt;&lt;mtext&gt;otherwise&lt;/mtext&gt;&lt;mo&gt;,&lt;/mo&gt;&lt;/mrow&gt;&lt;/mtd&gt;&lt;/mtr&gt;&lt;/mtable&gt;&lt;/mrow&gt;&lt;/mrow&gt;&lt;/mrow&gt;&lt;annotation encoding=&quot;MathType-MTEF&quot;&gt;          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemyyae2aaSbaaSqaaiabdMgaPjabdQgaQbqabaGccqGH9aqpdaGabaqaauaabaqaciaaaeaacqaIXaqmaeaacqqGPbqAcqqGMbGzcqqGGaaicqWGZbWCdaWgaaWcbaGaemyAaKMaemOAaOgabeaakiabgwMiZkabes8a0jabcUda7aqaaiabicdaWaqaaiabb+gaVjabbsha0jabbIgaOjabbwgaLjabbkhaYjabbEha3jabbMgaPjabbohaZjabbwgaLjabcYcaSaaaaiaawUhaaaaa@4E15@&lt;/annotation&gt;&lt;/semantics&gt;&lt;/mrow&gt;&lt;/semantics&gt;&lt;/math&gt;" role="presentation" style="position: relative;"><nobr aria-hidden="true"><span class="math" id="MathJax-Span-62" style="width: 10.231em; display: inline-block;"><span style="display: inline-block; position: relative; width: 9.224em; height: 0px; font-size: 111%;"><span style="position: absolute; clip: rect(1.698em, 1009.17em, 4.56em, -999.997em); top: -3.389em; left: 0em;"><span class="mrow" id="MathJax-Span-63"><span class="semantics" id="MathJax-Span-64"><span class="mrow" id="MathJax-Span-65"><span class="semantics" id="MathJax-Span-66"><span class="mrow" id="MathJax-Span-67"><span class="msub" id="MathJax-Span-68"><span style="display: inline-block; position: relative; width: 1.116em; height: 0px;"><span style="position: absolute; clip: rect(3.394em, 1000.53em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-69" style="font-family: MathJax_Math-italic;">a</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.813em; left: 0.533em;"><span class="mrow" id="MathJax-Span-70"><span class="mi" id="MathJax-Span-71" style="font-size: 70.7%; font-family: MathJax_Math-italic;">i</span><span class="mi" id="MathJax-Span-72" style="font-size: 70.7%; font-family: MathJax_Math-italic;">j</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mo" id="MathJax-Span-73" style="font-family: MathJax_Main; padding-left: 0.268em;">=</span><span class="mrow" id="MathJax-Span-74" style="padding-left: 0.268em;"><span class="mo" id="MathJax-Span-75" style="vertical-align: 0em;"><span style="font-family: MathJax_Size3;">{</span></span><span class="mrow" id="MathJax-Span-76"><span class="mtable" id="MathJax-Span-77"><span style="display: inline-block; position: relative; width: 5.726em; height: 0px; margin-right: 0.162em; margin-left: 0.162em;"><span style="position: absolute; clip: rect(2.44em, 1000.48em, 4.984em, -999.997em); top: -3.972em; left: 0em;"><span style="display: inline-block; position: relative; width: 0.48em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1000.43em, 4.136em, -999.997em); top: -4.661em; left: 0em;"><span class="mtd" id="MathJax-Span-78"><span class="mrow" id="MathJax-Span-79"><span class="mn" id="MathJax-Span-80" style="font-family: MathJax_Main;">1</span></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.129em, 1000.48em, 4.136em, -999.997em); top: -3.177em; left: 0em;"><span class="mtd" id="MathJax-Span-93"><span class="mrow" id="MathJax-Span-94"><span class="mn" id="MathJax-Span-95" style="font-family: MathJax_Main;">0</span></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(2.387em, 1004.35em, 5.143em, -999.997em); top: -3.972em; left: 1.328em;"><span style="display: inline-block; position: relative; width: 4.401em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1003.92em, 4.454em, -999.997em); top: -4.661em; left: 0em;"><span class="mtd" id="MathJax-Span-81"><span class="mrow" id="MathJax-Span-82"><span class="mrow" id="MathJax-Span-83"><span class="mtext" id="MathJax-Span-84" style="font-family: MathJax_Main;">if&nbsp;</span><span class="msub" id="MathJax-Span-85"><span style="display: inline-block; position: relative; width: 1.063em; height: 0px;"><span style="position: absolute; clip: rect(3.394em, 1000.43em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-86" style="font-family: MathJax_Math-italic;">s</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.813em; left: 0.48em;"><span class="mrow" id="MathJax-Span-87"><span class="mi" id="MathJax-Span-88" style="font-size: 70.7%; font-family: MathJax_Math-italic;">i</span><span class="mi" id="MathJax-Span-89" style="font-size: 70.7%; font-family: MathJax_Math-italic;">j</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mo" id="MathJax-Span-90" style="font-family: MathJax_Main; padding-left: 0.268em;">≥</span><span class="mi" id="MathJax-Span-91" style="font-family: MathJax_Math-italic; padding-left: 0.268em;">τ<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.109em;"></span></span><span class="mo" id="MathJax-Span-92" style="font-family: MathJax_Main;">;</span></span></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.129em, 1004.35em, 4.348em, -999.997em); top: -3.177em; left: 0em;"><span class="mtd" id="MathJax-Span-96"><span class="mrow" id="MathJax-Span-97"><span class="mrow" id="MathJax-Span-98"><span class="mtext" id="MathJax-Span-99" style="font-family: MathJax_Main;">otherwise</span><span class="mo" id="MathJax-Span-100" style="font-family: MathJax_Main;">,</span></span></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.394em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -1.174em; border-left: 0px solid; width: 0px; height: 2.944em;"></span></span></nobr><span class="MJX_Assistive_MathML" role="presentation"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><semantics><mrow><msub><mi>a</mi><mrow><mi>i</mi><mi>j</mi></mrow></msub><mo>=</mo><mrow><mo>{</mo><mrow><mtable columnalign="left"><mtr columnalign="left"><mtd columnalign="left"><mn>1</mn></mtd><mtd columnalign="left"><mrow><mtext>if&nbsp;</mtext><msub><mi>s</mi><mrow><mi>i</mi><mi>j</mi></mrow></msub><mo>≥</mo><mi>τ</mi><mo>;</mo></mrow></mtd></mtr><mtr columnalign="left"><mtd columnalign="left"><mn>0</mn></mtd><mtd columnalign="left"><mrow><mtext>otherwise</mtext><mo>,</mo></mrow></mtd></mtr></mtable></mrow></mrow></mrow><annotation encoding="MathType-MTEF">          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemyyae2aaSbaaSqaaiabdMgaPjabdQgaQbqabaGccqGH9aqpdaGabaqaauaabaqaciaaaeaacqaIXaqmaeaacqqGPbqAcqqGMbGzcqqGGaaicqWGZbWCdaWgaaWcbaGaemyAaKMaemOAaOgabeaakiabgwMiZkabes8a0jabcUda7aqaaiabicdaWaqaaiabb+gaVjabbsha0jabbIgaOjabbwgaLjabbkhaYjabbEha3jabbMgaPjabbohaZjabbwgaLjabcYcaSaaaaiaawUhaaaaa@4E15@</annotation></semantics></mrow></semantics></math></span></span><script type="math/mml" id="MathJax-Element-3"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow>
                        <semantics>
                          <mrow>
                            <msub>
                              <mi>a</mi>
                              <mrow>
                                <mi>i</mi>
                                <mi>j</mi>
                              </mrow>
                            </msub>
                            <mo>=</mo>
                            <mrow>
                              <mo>{</mo>
                              <mrow>
                                <mtable columnalign="left">
                                  <mtr columnalign="left">
                                    <mtd columnalign="left">
                                      <mn>1</mn>
                                    </mtd>
                                    <mtd columnalign="left">
                                      <mrow>
                                        <mtext>if </mtext>
                                        <msub>
                                          <mi>s</mi>
                                          <mrow>
                                            <mi>i</mi>
                                            <mi>j</mi>
                                          </mrow>
                                        </msub>
                                        <mo>≥</mo>
                                        <mi>τ</mi>
                                        <mo>;</mo>
                                      </mrow>
                                    </mtd>
                                  </mtr>
                                  <mtr columnalign="left">
                                    <mtd columnalign="left">
                                      <mn>0</mn>
                                    </mtd>
                                    <mtd columnalign="left">
                                      <mrow>
                                        <mtext>otherwise</mtext>
                                        <mo>,</mo>
                                      </mrow>
                                    </mtd>
                                  </mtr>
                                </mtable>
                              </mrow>
                            </mrow>
                          </mrow>
                          <annotation encoding="MathType-MTEF">MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemyyae2aaSbaaSqaaiabdMgaPjabdQgaQbqabaGccqGH9aqpdaGabaqaauaabaqaciaaaeaacqaIXaqmaeaacqqGPbqAcqqGMbGzcqqGGaaicqWGZbWCdaWgaaWcbaGaemyAaKMaemOAaOgabeaakiabgwMiZkabes8a0jabcUda7aqaaiabicdaWaqaaiabb+gaVjabbsha0jabbIgaOjabbwgaLjabbkhaYjabbEha3jabbMgaPjabbohaZjabbwgaLjabcYcaSaaaaiaawUhaaaaa@4E15@</annotation>
                        </semantics>
                      </mrow></semantics></math></script></div> <div class="EquationNumber">(4)</div></div></div>
                <div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">where <em xmlns="" class="EmphasisTypeItalic">τ</em> is the "hard" threshold parameter. Thus, two genes are linked (<em xmlns="" class="EmphasisTypeItalic">a</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">ij</em>
                  </sub>= 1) if the absolute correlation between their expression profiles exceeds the (hard) threshold <em xmlns="" class="EmphasisTypeItalic">τ</em>. The hard-thresholding procedure is implemented in the function signumAdjacencyFunction. While unweighted networks are widely used, they do not reflect the continuous nature of the underlying co-expression information and may thus lead to an information loss. In contrast, weighted networks allow the adjacency to take on continuous values between 0 and 1. A weighed network adjacency can be defined by raising the co-expression similarity to a power [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR5" data-toggle="citation">5</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR10" data-toggle="citation">10</a></span>]:<div id="Equ3" class="Equation EquationMathjax"><div class="EquationContent"><span class="MathJax_Preview" style=""></span><span class="MathJax" id="MathJax-Element-4-Frame" tabindex="0" data-mathml="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot;&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;msub&gt;&lt;mi&gt;a&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mi&gt;j&lt;/mi&gt;&lt;/mrow&gt;&lt;/msub&gt;&lt;mo&gt;=&lt;/mo&gt;&lt;msubsup&gt;&lt;mi&gt;s&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mi&gt;j&lt;/mi&gt;&lt;/mrow&gt;&lt;mi&gt;&amp;#x3B2;&lt;/mi&gt;&lt;/msubsup&gt;&lt;mo&gt;,&lt;/mo&gt;&lt;/mrow&gt;&lt;annotation encoding=&quot;MathType-MTEF&quot;&gt;          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemyyae2aaSbaaSqaaiabdMgaPjabdQgaQbqabaGccqGH9aqpcqWGZbWCdaqhaaWcbaGaemyAaKMaemOAaOgabaGaeqOSdigaaOGaeiilaWcaaa@3841@&lt;/annotation&gt;&lt;/semantics&gt;&lt;/mrow&gt;&lt;/semantics&gt;&lt;/math&gt;" role="presentation" style="position: relative;"><nobr aria-hidden="true"><span class="math" id="MathJax-Span-101" style="width: 4.189em; display: inline-block;"><span style="display: inline-block; position: relative; width: 3.765em; height: 0px; font-size: 111%;"><span style="position: absolute; clip: rect(1.063em, 1003.71em, 2.758em, -999.997em); top: -2.17em; left: 0em;"><span class="mrow" id="MathJax-Span-102"><span class="semantics" id="MathJax-Span-103"><span class="mrow" id="MathJax-Span-104"><span class="semantics" id="MathJax-Span-105"><span class="mrow" id="MathJax-Span-106"><span class="msub" id="MathJax-Span-107"><span style="display: inline-block; position: relative; width: 1.116em; height: 0px;"><span style="position: absolute; clip: rect(3.394em, 1000.53em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-108" style="font-family: MathJax_Math-italic;">a</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.813em; left: 0.533em;"><span class="mrow" id="MathJax-Span-109"><span class="mi" id="MathJax-Span-110" style="font-size: 70.7%; font-family: MathJax_Math-italic;">i</span><span class="mi" id="MathJax-Span-111" style="font-size: 70.7%; font-family: MathJax_Math-italic;">j</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mo" id="MathJax-Span-112" style="font-family: MathJax_Main; padding-left: 0.268em;">=</span><span class="msubsup" id="MathJax-Span-113" style="padding-left: 0.268em;"><span style="display: inline-block; position: relative; width: 1.063em; height: 0px;"><span style="position: absolute; clip: rect(3.394em, 1000.43em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-114" style="font-family: MathJax_Math-italic;">s</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.341em, 1000.48em, 4.295em, -999.997em); top: -4.449em; left: 0.48em;"><span class="mi" id="MathJax-Span-115" style="font-size: 70.7%; font-family: MathJax_Math-italic;">β<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.341em, 1000.64em, 4.295em, -999.997em); top: -3.654em; left: 0.48em;"><span class="mrow" id="MathJax-Span-116"><span class="mi" id="MathJax-Span-117" style="font-size: 70.7%; font-family: MathJax_Math-italic;">i</span><span class="mi" id="MathJax-Span-118" style="font-size: 70.7%; font-family: MathJax_Math-italic;">j</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mo" id="MathJax-Span-119" style="font-family: MathJax_Main;">,</span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 2.175em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -0.526em; border-left: 0px solid; width: 0px; height: 1.709em;"></span></span></nobr><span class="MJX_Assistive_MathML" role="presentation"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><semantics><mrow><msub><mi>a</mi><mrow><mi>i</mi><mi>j</mi></mrow></msub><mo>=</mo><msubsup><mi>s</mi><mrow><mi>i</mi><mi>j</mi></mrow><mi>β</mi></msubsup><mo>,</mo></mrow><annotation encoding="MathType-MTEF">          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemyyae2aaSbaaSqaaiabdMgaPjabdQgaQbqabaGccqGH9aqpcqWGZbWCdaqhaaWcbaGaemyAaKMaemOAaOgabaGaeqOSdigaaOGaeiilaWcaaa@3841@</annotation></semantics></mrow></semantics></math></span></span><script type="math/mml" id="MathJax-Element-4"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow>
                        <semantics>
                          <mrow>
                            <msub>
                              <mi>a</mi>
                              <mrow>
                                <mi>i</mi>
                                <mi>j</mi>
                              </mrow>
                            </msub>
                            <mo>=</mo>
                            <msubsup>
                              <mi>s</mi>
                              <mrow>
                                <mi>i</mi>
                                <mi>j</mi>
                              </mrow>
                              <mi>β</mi>
                            </msubsup>
                            <mo>,</mo>
                          </mrow>
                          <annotation encoding="MathType-MTEF">MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemyyae2aaSbaaSqaaiabdMgaPjabdQgaQbqabaGccqGH9aqpcqWGZbWCdaqhaaWcbaGaemyAaKMaemOAaOgabaGaeqOSdigaaOGaeiilaWcaaa@3841@</annotation>
                        </semantics>
                      </mrow></semantics></math></script></div> <div class="EquationNumber">(5)</div></div></div>
                <div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">with <em xmlns="" class="EmphasisTypeItalic">β</em> ≥ 1. The function adjacency calculates the adjacency matrix from expression data. The adjacency in Equation <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Equ5">5</a></span> implies that the weighted adjacency <em xmlns="" class="EmphasisTypeItalic">a</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">ij</em>
                  </sub>between two genes is proportional to their similarity on a logarithmic scale, <em xmlns="" class="EmphasisTypeItalic">log</em>(<em xmlns="" class="EmphasisTypeItalic">a</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">ij</em>
                  </sub>) = <em xmlns="" class="EmphasisTypeItalic">β</em> × <em xmlns="" class="EmphasisTypeItalic">log</em>(<em xmlns="" class="EmphasisTypeItalic">s</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">ij</em>
                  </sub>). Adjacency functions for both weighted and unweighted networks require the user to choose threshold parameters, for example by applying the approximate scale-free topology criterion [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR5" data-toggle="citation">5</a></span>]. The package provides functions pickSoftThreshold, pickHardThreshold that assist in choosing the parameters, as well as the function scaleFreePlot for evaluating whether the network exhibits a scale free topology. Figure <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig2">2A</a></span> shows a plot identifying scale free topology in simulated expression data.<figure xmlns="" class="Figure" id="Fig2"><div xmlns="http://www.w3.org/1999/xhtml" class="MediaObject">
                      <a target="_blank" rel="noopener" href="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_Article_2544_Fig2_HTML.jpg"><img class="c-lazy-image u-js-shown" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/12859_2008_Article_2544_Fig2_HTML.jpg" alt="Figure 2" data-lazy-load="true" data-src="https://media.springernature.com/lw785/springer-static/image/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_Article_2544_Fig2_HTML.jpg"><noscript><img src="https://media.springernature.com/lw785/springer-static/image/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_Article_2544_Fig2_HTML.jpg" alt="Figure 2"></noscript></a>
                    </div><figcaption class="Caption" lang="en"><div xmlns="http://www.w3.org/1999/xhtml" class="CaptionContent"><span class="CaptionNumber">Figure 2</span>
                        <p xmlns="" class="SimplePara"><strong class="EmphasisTypeBold">Network visualization plots</strong>. A. Log-log plot of whole-network connectivity distribution. The <em class="EmphasisTypeItalic">x</em>-axis shows the logarithm of whole network connectivity, <em class="EmphasisTypeItalic">y</em>-axis the logarithm of the corresponding frequency distribution. On this plot the distribution approximately follows a straight line, which is referred to as approximately scale-free topology. B. Results of classical multidimensional scaling. Modules tend to form separate 'fingers' in this plot. Intramodular hub genes are located at the finger tips. C. Network heatmap plot. Branches in the hierarchical clustering dendrograms correspond to modules. Color-coded module membership is displayed in the color bars below and to the right of the dendrograms. In the heatmap, high co-expression interconnectedness is indicated by progressively more saturated yellow and red colors. Modules correspond to blocks of highly interconnected genes. Genes with high intramodular connectivity are located at the tip of the module branches since they display the highest interconnectedness with the rest of the genes in the module.</p>
                      </div></figcaption></figure></div>
              </section><section id="Sec6" class="Section2 RenderAsSection2">
                <h3 class="Heading">Category 2: Functions for module detection</h3>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Once the network has been constructed, module detection is often a logical next step. Modules are defined as clusters of densely interconnected genes. Several measures of network interconnectedness are described in [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR25" data-toggle="citation">25</a></span>]. As default, we we use the topological overlap measure [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR5" data-toggle="citation">5</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR25" data-toggle="citation">25</a></span><span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR26" data-toggle="citation">26</a></span>, </span><span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR27" data-toggle="citation">27</a></span>] since it has worked well in several applications. WGCNA identifies gene modules using unsupervised clustering, i.e. without the use of a priori defined gene sets. The user has a choice of several module detection methods. The default method is hierarchical clustering using the standard R function hclust [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR28" data-toggle="citation">28</a></span>]; branches of the hierarchical clustering dendrogram correspond to modules and can be identified using one of a number of available branch cutting methods, for example the constant-height cut or two Dynamic Branch Cut methods [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR29" data-toggle="citation">29</a></span>].</p>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">In Figure <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig2">2C</a></span> we show a network heatmap plot (interconnectivity plot) of a gene network together with the corresponding hierarchical clustering dendrograms and the resulting modules. Figure <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig2">2B</a></span> provides an alternate visualization of the module structure via a multi-dimensional scaling plot (standard R function cmdscale).</p>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">One drawback of hierarchical clustering is that it can be difficult to determine how many (if any) clusters are present in the data set. Although the height and shape parameters of the Dynamic Tree Cut method provide improved exibility for branch cutting and module detection, it remains an open research question how to choose optimal cutting parameters or how to estimate the number of clusters in the data set [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR30" data-toggle="citation">30</a></span>]. While our default parameter values have worked well in several applications, in practice we recommend to carry out a cluster stability/robustness analysis. A co-expression module may reflect a true biological signal (e.g. a pathway) or it may reflect noise (e.g. a technical artifacts, tissue contamination, or a false positive). To test whether the identified modules are biologically meaningful, gene ontology information (functional enrichment analysis) can be used. Toward this end, we provide an R tutorial that describes how to interface the WGCNA package with relevant external software packages and databases.</p>
                <section id="Sec7" class="Section3 RenderAsSection3">
                  <h4 class="Heading">Summarizing the profiles of a module</h4>
                  <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Several options have been implemented for summarizing the gene expression profiles of a given module. For example, the function moduleEigengenes represents the module expressions of the <em xmlns="" class="EmphasisTypeItalic">q</em>-th module by the module eigengene <em xmlns="" class="EmphasisTypeItalic">E</em><sup>(<em xmlns="" class="EmphasisTypeItalic">q</em>)</sup>, defined as the first principal component of the expression matrix. The eigengene <em xmlns="" class="EmphasisTypeItalic">E</em> can be thought of as a weighted average expression profile. Eigengene calculation incorporates imputation of missing values implemented in the package impute [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR31" data-toggle="citation">31</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR32" data-toggle="citation">32</a></span>]. Alternatively, the user can use the intramodular connectivity measure to define the most highly connected intramodular hub gene as the module representative. One can show that intramodular hub genes are highly correlated with the module eigengene [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR11" data-toggle="citation">11</a></span>].</p>
                </section>
                <section id="Sec8" class="Section3 RenderAsSection3">
                  <h4 class="Heading">Fuzzy measure of module membership</h4>
                  <div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Hierarchical clustering and most other standard clustering methods such as Partitioning Around Medoids (PAM) [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR28" data-toggle="citation">28</a></span>] result in a binary module assignment, i.e. a node is either in or outside of a module. In some applications it may be advantageous to define a continuous, fuzzy measure of module membership for all nodes. Such measure is particularly useful to identify nodes that lie near the boundary of a module, or nodes that are intermediate between two or more modules. As explained in detail in [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR11" data-toggle="citation">11</a></span>], the module membership of node <em xmlns="" class="EmphasisTypeItalic">i</em> in module <em xmlns="" class="EmphasisTypeItalic">q</em> can be defined as<div id="Equ4" class="Equation EquationMathjax"><div class="EquationContent"><span class="MathJax_Preview" style=""></span><span class="MathJax" id="MathJax-Element-5-Frame" tabindex="0" data-mathml="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot;&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;msubsup&gt;&lt;mi&gt;K&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;c&lt;/mi&gt;&lt;mi&gt;o&lt;/mi&gt;&lt;mi&gt;r&lt;/mi&gt;&lt;mo&gt;,&lt;/mo&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/mrow&gt;&lt;mrow&gt;&lt;mo stretchy=&quot;false&quot;&gt;(&lt;/mo&gt;&lt;mi&gt;q&lt;/mi&gt;&lt;mo stretchy=&quot;false&quot;&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/msubsup&gt;&lt;mo&gt;:&lt;/mo&gt;&lt;mo&gt;=&lt;/mo&gt;&lt;mtext&gt;cor&lt;/mtext&gt;&lt;mo stretchy=&quot;false&quot;&gt;(&lt;/mo&gt;&lt;msub&gt;&lt;mi&gt;x&lt;/mi&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/msub&gt;&lt;mo&gt;,&lt;/mo&gt;&lt;msup&gt;&lt;mi&gt;E&lt;/mi&gt;&lt;mrow&gt;&lt;mo stretchy=&quot;false&quot;&gt;(&lt;/mo&gt;&lt;mi&gt;q&lt;/mi&gt;&lt;mo stretchy=&quot;false&quot;&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/msup&gt;&lt;mo stretchy=&quot;false&quot;&gt;)&lt;/mo&gt;&lt;mo&gt;,&lt;/mo&gt;&lt;/mrow&gt;&lt;annotation encoding=&quot;MathType-MTEF&quot;&gt;          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaakiabcQda6iabg2da9iabbogaJjabb+gaVjabbkhaYjabcIcaOiabdIha4naaBaaaleaacqWGPbqAaeqaaOGaeiilaWIaemyrau0aaWbaaSqabeaacqGGOaakcqWGXbqCcqGGPaqkaaGccqGGPaqkcqGGSaalaaa@47F6@&lt;/annotation&gt;&lt;/semantics&gt;&lt;/mrow&gt;&lt;/semantics&gt;&lt;/math&gt;" role="presentation" style="position: relative;"><nobr aria-hidden="true"><span class="math" id="MathJax-Span-120" style="width: 10.443em; display: inline-block;"><span style="display: inline-block; position: relative; width: 9.383em; height: 0px; font-size: 111%;"><span style="position: absolute; clip: rect(0.957em, 1009.33em, 2.758em, -999.997em); top: -2.17em; left: 0em;"><span class="mrow" id="MathJax-Span-121"><span class="semantics" id="MathJax-Span-122"><span class="mrow" id="MathJax-Span-123"><span class="semantics" id="MathJax-Span-124"><span class="mrow" id="MathJax-Span-125"><span class="msubsup" id="MathJax-Span-126"><span style="display: inline-block; position: relative; width: 2.334em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1000.9em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-127" style="font-family: MathJax_Math-italic;">K<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.056em;"></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.288em, 1000.96em, 4.295em, -999.997em); top: -4.502em; left: 0.957em;"><span class="mrow" id="MathJax-Span-128"><span class="mo" id="MathJax-Span-129" style="font-size: 70.7%; font-family: MathJax_Main;">(</span><span class="mi" id="MathJax-Span-130" style="font-size: 70.7%; font-family: MathJax_Math-italic;">q<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span><span class="mo" id="MathJax-Span-131" style="font-size: 70.7%; font-family: MathJax_Main;">)</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.341em, 1001.49em, 4.295em, -999.997em); top: -3.654em; left: 0.851em;"><span class="mrow" id="MathJax-Span-132"><span class="mi" id="MathJax-Span-133" style="font-size: 70.7%; font-family: MathJax_Math-italic;">c</span><span class="mi" id="MathJax-Span-134" style="font-size: 70.7%; font-family: MathJax_Math-italic;">o</span><span class="mi" id="MathJax-Span-135" style="font-size: 70.7%; font-family: MathJax_Math-italic;">r</span><span class="mo" id="MathJax-Span-136" style="font-size: 70.7%; font-family: MathJax_Main;">,</span><span class="mi" id="MathJax-Span-137" style="font-size: 70.7%; font-family: MathJax_Math-italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mo" id="MathJax-Span-138" style="font-family: MathJax_Main; padding-left: 0.268em;">:</span><span class="mo" id="MathJax-Span-139" style="font-family: MathJax_Main;">=</span><span class="mtext" id="MathJax-Span-140" style="font-family: MathJax_Main; padding-left: 0.268em;">cor</span><span class="mo" id="MathJax-Span-141" style="font-family: MathJax_Main;">(</span><span class="msub" id="MathJax-Span-142"><span style="display: inline-block; position: relative; width: 0.904em; height: 0px;"><span style="position: absolute; clip: rect(3.394em, 1000.53em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-143" style="font-family: MathJax_Math-italic;">x</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.813em; left: 0.586em;"><span class="mi" id="MathJax-Span-144" style="font-size: 70.7%; font-family: MathJax_Math-italic;">i</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mo" id="MathJax-Span-145" style="font-family: MathJax_Main;">,</span><span class="msup" id="MathJax-Span-146" style="padding-left: 0.162em;"><span style="display: inline-block; position: relative; width: 1.751em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1000.75em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-147" style="font-family: MathJax_Math-italic;">E<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -4.343em; left: 0.851em;"><span class="mrow" id="MathJax-Span-148"><span class="mo" id="MathJax-Span-149" style="font-size: 70.7%; font-family: MathJax_Main;">(</span><span class="mi" id="MathJax-Span-150" style="font-size: 70.7%; font-family: MathJax_Math-italic;">q<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span><span class="mo" id="MathJax-Span-151" style="font-size: 70.7%; font-family: MathJax_Main;">)</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mo" id="MathJax-Span-152" style="font-family: MathJax_Main;">)</span><span class="mo" id="MathJax-Span-153" style="font-family: MathJax_Main;">,</span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 2.175em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -0.526em; border-left: 0px solid; width: 0px; height: 1.768em;"></span></span></nobr><span class="MJX_Assistive_MathML" role="presentation"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><semantics><mrow><msubsup><mi>K</mi><mrow><mi>c</mi><mi>o</mi><mi>r</mi><mo>,</mo><mi>i</mi></mrow><mrow><mo stretchy="false">(</mo><mi>q</mi><mo stretchy="false">)</mo></mrow></msubsup><mo>:</mo><mo>=</mo><mtext>cor</mtext><mo stretchy="false">(</mo><msub><mi>x</mi><mi>i</mi></msub><mo>,</mo><msup><mi>E</mi><mrow><mo stretchy="false">(</mo><mi>q</mi><mo stretchy="false">)</mo></mrow></msup><mo stretchy="false">)</mo><mo>,</mo></mrow><annotation encoding="MathType-MTEF">          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaakiabcQda6iabg2da9iabbogaJjabb+gaVjabbkhaYjabcIcaOiabdIha4naaBaaaleaacqWGPbqAaeqaaOGaeiilaWIaemyrau0aaWbaaSqabeaacqGGOaakcqWGXbqCcqGGPaqkaaGccqGGPaqkcqGGSaalaaa@47F6@</annotation></semantics></mrow></semantics></math></span></span><script type="math/mml" id="MathJax-Element-5"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow>
                          <semantics>
                            <mrow>
                              <msubsup>
                                <mi>K</mi>
                                <mrow>
                                  <mi>c</mi>
                                  <mi>o</mi>
                                  <mi>r</mi>
                                  <mo>,</mo>
                                  <mi>i</mi>
                                </mrow>
                                <mrow>
                                  <mo stretchy="false">(</mo>
                                  <mi>q</mi>
                                  <mo stretchy="false">)</mo>
                                </mrow>
                              </msubsup>
                              <mo>:</mo>
                              <mo>=</mo>
                              <mtext>cor</mtext>
                              <mo stretchy="false">(</mo>
                              <msub>
                                <mi>x</mi>
                                <mi>i</mi>
                              </msub>
                              <mo>,</mo>
                              <msup>
                                <mi>E</mi>
                                <mrow>
                                  <mo stretchy="false">(</mo>
                                  <mi>q</mi>
                                  <mo stretchy="false">)</mo>
                                </mrow>
                              </msup>
                              <mo stretchy="false">)</mo>
                              <mo>,</mo>
                            </mrow>
                            <annotation encoding="MathType-MTEF">MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaakiabcQda6iabg2da9iabbogaJjabb+gaVjabbkhaYjabcIcaOiabdIha4naaBaaaleaacqWGPbqAaeqaaOGaeiilaWIaemyrau0aaWbaaSqabeaacqGGOaakcqWGXbqCcqGGPaqkaaGccqGGPaqkcqGGSaalaaa@47F6@</annotation>
                          </semantics>
                        </mrow></semantics></math></script></div> <div class="EquationNumber">(6)</div></div></div>
                  <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">where <em xmlns="" class="EmphasisTypeItalic">x</em><sub>
                      <em xmlns="" class="EmphasisTypeItalic">i</em>
                    </sub>is the profile of node <em xmlns="" class="EmphasisTypeItalic">i</em> and <em xmlns="" class="EmphasisTypeItalic">E</em><sup>(<em xmlns="" class="EmphasisTypeItalic">q</em>)</sup>is the module eigengene of module <em xmlns="" class="EmphasisTypeItalic">q</em>. The module membership measure <span class="InlineEquation" id="IEq2"><span class="MathJax_Preview" style=""></span><span class="MathJax" id="MathJax-Element-6-Frame" tabindex="0" data-mathml="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot;&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;msubsup&gt;&lt;mi&gt;K&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;c&lt;/mi&gt;&lt;mi&gt;o&lt;/mi&gt;&lt;mi&gt;r&lt;/mi&gt;&lt;mo&gt;,&lt;/mo&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/mrow&gt;&lt;mrow&gt;&lt;mo stretchy=&quot;false&quot;&gt;(&lt;/mo&gt;&lt;mi&gt;q&lt;/mi&gt;&lt;mo stretchy=&quot;false&quot;&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/msubsup&gt;&lt;/mrow&gt;&lt;annotation encoding=&quot;MathType-MTEF&quot;&gt;          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaaaaa@369C@&lt;/annotation&gt;&lt;/semantics&gt;&lt;/mrow&gt;&lt;/semantics&gt;&lt;/math&gt;" role="presentation" style="position: relative;"><nobr aria-hidden="true"><span class="math" id="MathJax-Span-154" style="width: 2.599em; display: inline-block;"><span style="display: inline-block; position: relative; width: 2.334em; height: 0px; font-size: 111%;"><span style="position: absolute; clip: rect(0.904em, 1002.33em, 2.705em, -999.997em); top: -2.117em; left: 0em;"><span class="mrow" id="MathJax-Span-155"><span class="semantics" id="MathJax-Span-156"><span class="mrow" id="MathJax-Span-157"><span class="semantics" id="MathJax-Span-158"><span class="mrow" id="MathJax-Span-159"><span class="msubsup" id="MathJax-Span-160"><span style="display: inline-block; position: relative; width: 2.334em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1000.9em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-161" style="font-family: MathJax_Math-italic;">K<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.056em;"></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.288em, 1000.96em, 4.295em, -999.997em); top: -4.502em; left: 0.957em;"><span class="mrow" id="MathJax-Span-162"><span class="mo" id="MathJax-Span-163" style="font-size: 70.7%; font-family: MathJax_Main;">(</span><span class="mi" id="MathJax-Span-164" style="font-size: 70.7%; font-family: MathJax_Math-italic;">q<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span><span class="mo" id="MathJax-Span-165" style="font-size: 70.7%; font-family: MathJax_Main;">)</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.341em, 1001.49em, 4.295em, -999.997em); top: -3.654em; left: 0.851em;"><span class="mrow" id="MathJax-Span-166"><span class="mi" id="MathJax-Span-167" style="font-size: 70.7%; font-family: MathJax_Math-italic;">c</span><span class="mi" id="MathJax-Span-168" style="font-size: 70.7%; font-family: MathJax_Math-italic;">o</span><span class="mi" id="MathJax-Span-169" style="font-size: 70.7%; font-family: MathJax_Math-italic;">r</span><span class="mo" id="MathJax-Span-170" style="font-size: 70.7%; font-family: MathJax_Main;">,</span><span class="mi" id="MathJax-Span-171" style="font-size: 70.7%; font-family: MathJax_Math-italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 2.122em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -0.526em; border-left: 0px solid; width: 0px; height: 1.768em;"></span></span></nobr><span class="MJX_Assistive_MathML" role="presentation"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><semantics><mrow><msubsup><mi>K</mi><mrow><mi>c</mi><mi>o</mi><mi>r</mi><mo>,</mo><mi>i</mi></mrow><mrow><mo stretchy="false">(</mo><mi>q</mi><mo stretchy="false">)</mo></mrow></msubsup></mrow><annotation encoding="MathType-MTEF">          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaaaaa@369C@</annotation></semantics></mrow></semantics></math></span></span><script type="math/mml" id="MathJax-Element-6"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow>
                          <semantics>
                            <mrow>
                              <msubsup>
                                <mi>K</mi>
                                <mrow>
                                  <mi>c</mi>
                                  <mi>o</mi>
                                  <mi>r</mi>
                                  <mo>,</mo>
                                  <mi>i</mi>
                                </mrow>
                                <mrow>
                                  <mo stretchy="false">(</mo>
                                  <mi>q</mi>
                                  <mo stretchy="false">)</mo>
                                </mrow>
                              </msubsup>
                            </mrow>
                            <annotation encoding="MathType-MTEF">MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaaaaa@369C@</annotation>
                          </semantics>
                        </mrow></semantics></math></script></span> lies in [-1, 1] and specifies how close node <em xmlns="" class="EmphasisTypeItalic">i</em> is to module <em xmlns="" class="EmphasisTypeItalic">q</em>, <em xmlns="" class="EmphasisTypeItalic">q</em> = 1, . . ., <em xmlns="" class="EmphasisTypeItalic">Q</em>. The larger |<span class="InlineEquation" id="IEq3"><span class="MathJax_Preview" style=""></span><span class="MathJax" id="MathJax-Element-7-Frame" tabindex="0" data-mathml="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot;&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;msubsup&gt;&lt;mi&gt;K&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;c&lt;/mi&gt;&lt;mi&gt;o&lt;/mi&gt;&lt;mi&gt;r&lt;/mi&gt;&lt;mo&gt;,&lt;/mo&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/mrow&gt;&lt;mrow&gt;&lt;mo stretchy=&quot;false&quot;&gt;(&lt;/mo&gt;&lt;mi&gt;q&lt;/mi&gt;&lt;mo stretchy=&quot;false&quot;&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/msubsup&gt;&lt;/mrow&gt;&lt;annotation encoding=&quot;MathType-MTEF&quot;&gt;          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaaaaa@369C@&lt;/annotation&gt;&lt;/semantics&gt;&lt;/mrow&gt;&lt;/semantics&gt;&lt;/math&gt;" role="presentation" style="position: relative;"><nobr aria-hidden="true"><span class="math" id="MathJax-Span-172" style="width: 2.599em; display: inline-block;"><span style="display: inline-block; position: relative; width: 2.334em; height: 0px; font-size: 111%;"><span style="position: absolute; clip: rect(0.904em, 1002.33em, 2.705em, -999.997em); top: -2.117em; left: 0em;"><span class="mrow" id="MathJax-Span-173"><span class="semantics" id="MathJax-Span-174"><span class="mrow" id="MathJax-Span-175"><span class="semantics" id="MathJax-Span-176"><span class="mrow" id="MathJax-Span-177"><span class="msubsup" id="MathJax-Span-178"><span style="display: inline-block; position: relative; width: 2.334em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1000.9em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-179" style="font-family: MathJax_Math-italic;">K<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.056em;"></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.288em, 1000.96em, 4.295em, -999.997em); top: -4.502em; left: 0.957em;"><span class="mrow" id="MathJax-Span-180"><span class="mo" id="MathJax-Span-181" style="font-size: 70.7%; font-family: MathJax_Main;">(</span><span class="mi" id="MathJax-Span-182" style="font-size: 70.7%; font-family: MathJax_Math-italic;">q<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span><span class="mo" id="MathJax-Span-183" style="font-size: 70.7%; font-family: MathJax_Main;">)</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.341em, 1001.49em, 4.295em, -999.997em); top: -3.654em; left: 0.851em;"><span class="mrow" id="MathJax-Span-184"><span class="mi" id="MathJax-Span-185" style="font-size: 70.7%; font-family: MathJax_Math-italic;">c</span><span class="mi" id="MathJax-Span-186" style="font-size: 70.7%; font-family: MathJax_Math-italic;">o</span><span class="mi" id="MathJax-Span-187" style="font-size: 70.7%; font-family: MathJax_Math-italic;">r</span><span class="mo" id="MathJax-Span-188" style="font-size: 70.7%; font-family: MathJax_Main;">,</span><span class="mi" id="MathJax-Span-189" style="font-size: 70.7%; font-family: MathJax_Math-italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 2.122em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -0.526em; border-left: 0px solid; width: 0px; height: 1.768em;"></span></span></nobr><span class="MJX_Assistive_MathML" role="presentation"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><semantics><mrow><msubsup><mi>K</mi><mrow><mi>c</mi><mi>o</mi><mi>r</mi><mo>,</mo><mi>i</mi></mrow><mrow><mo stretchy="false">(</mo><mi>q</mi><mo stretchy="false">)</mo></mrow></msubsup></mrow><annotation encoding="MathType-MTEF">          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaaaaa@369C@</annotation></semantics></mrow></semantics></math></span></span><script type="math/mml" id="MathJax-Element-7"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow>
                          <semantics>
                            <mrow>
                              <msubsup>
                                <mi>K</mi>
                                <mrow>
                                  <mi>c</mi>
                                  <mi>o</mi>
                                  <mi>r</mi>
                                  <mo>,</mo>
                                  <mi>i</mi>
                                </mrow>
                                <mrow>
                                  <mo stretchy="false">(</mo>
                                  <mi>q</mi>
                                  <mo stretchy="false">)</mo>
                                </mrow>
                              </msubsup>
                            </mrow>
                            <annotation encoding="MathType-MTEF">MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaaaaa@369C@</annotation>
                          </semantics>
                        </mrow></semantics></math></script></span>|, the more similar node <em xmlns="" class="EmphasisTypeItalic">i</em> is to the eigengene of the <em xmlns="" class="EmphasisTypeItalic">q</em>-th module. In some publications [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR14" data-toggle="citation">14</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR15" data-toggle="citation">15</a></span>], <span class="InlineEquation" id="IEq4"><span class="MathJax_Preview" style=""></span><span class="MathJax" id="MathJax-Element-8-Frame" tabindex="0" data-mathml="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot;&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;msubsup&gt;&lt;mi&gt;K&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;c&lt;/mi&gt;&lt;mi&gt;o&lt;/mi&gt;&lt;mi&gt;r&lt;/mi&gt;&lt;mo&gt;,&lt;/mo&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/mrow&gt;&lt;mrow&gt;&lt;mo stretchy=&quot;false&quot;&gt;(&lt;/mo&gt;&lt;mi&gt;q&lt;/mi&gt;&lt;mo stretchy=&quot;false&quot;&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/msubsup&gt;&lt;/mrow&gt;&lt;annotation encoding=&quot;MathType-MTEF&quot;&gt;          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaaaaa@369C@&lt;/annotation&gt;&lt;/semantics&gt;&lt;/mrow&gt;&lt;/semantics&gt;&lt;/math&gt;" role="presentation" style="position: relative;"><nobr aria-hidden="true"><span class="math" id="MathJax-Span-190" style="width: 2.599em; display: inline-block;"><span style="display: inline-block; position: relative; width: 2.334em; height: 0px; font-size: 111%;"><span style="position: absolute; clip: rect(0.904em, 1002.33em, 2.705em, -999.997em); top: -2.117em; left: 0em;"><span class="mrow" id="MathJax-Span-191"><span class="semantics" id="MathJax-Span-192"><span class="mrow" id="MathJax-Span-193"><span class="semantics" id="MathJax-Span-194"><span class="mrow" id="MathJax-Span-195"><span class="msubsup" id="MathJax-Span-196"><span style="display: inline-block; position: relative; width: 2.334em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1000.9em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-197" style="font-family: MathJax_Math-italic;">K<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.056em;"></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.288em, 1000.96em, 4.295em, -999.997em); top: -4.502em; left: 0.957em;"><span class="mrow" id="MathJax-Span-198"><span class="mo" id="MathJax-Span-199" style="font-size: 70.7%; font-family: MathJax_Main;">(</span><span class="mi" id="MathJax-Span-200" style="font-size: 70.7%; font-family: MathJax_Math-italic;">q<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span><span class="mo" id="MathJax-Span-201" style="font-size: 70.7%; font-family: MathJax_Main;">)</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.341em, 1001.49em, 4.295em, -999.997em); top: -3.654em; left: 0.851em;"><span class="mrow" id="MathJax-Span-202"><span class="mi" id="MathJax-Span-203" style="font-size: 70.7%; font-family: MathJax_Math-italic;">c</span><span class="mi" id="MathJax-Span-204" style="font-size: 70.7%; font-family: MathJax_Math-italic;">o</span><span class="mi" id="MathJax-Span-205" style="font-size: 70.7%; font-family: MathJax_Math-italic;">r</span><span class="mo" id="MathJax-Span-206" style="font-size: 70.7%; font-family: MathJax_Main;">,</span><span class="mi" id="MathJax-Span-207" style="font-size: 70.7%; font-family: MathJax_Math-italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 2.122em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -0.526em; border-left: 0px solid; width: 0px; height: 1.768em;"></span></span></nobr><span class="MJX_Assistive_MathML" role="presentation"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><semantics><mrow><msubsup><mi>K</mi><mrow><mi>c</mi><mi>o</mi><mi>r</mi><mo>,</mo><mi>i</mi></mrow><mrow><mo stretchy="false">(</mo><mi>q</mi><mo stretchy="false">)</mo></mrow></msubsup></mrow><annotation encoding="MathType-MTEF">          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaaaaa@369C@</annotation></semantics></mrow></semantics></math></span></span><script type="math/mml" id="MathJax-Element-8"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow>
                          <semantics>
                            <mrow>
                              <msubsup>
                                <mi>K</mi>
                                <mrow>
                                  <mi>c</mi>
                                  <mi>o</mi>
                                  <mi>r</mi>
                                  <mo>,</mo>
                                  <mi>i</mi>
                                </mrow>
                                <mrow>
                                  <mo stretchy="false">(</mo>
                                  <mi>q</mi>
                                  <mo stretchy="false">)</mo>
                                </mrow>
                              </msubsup>
                            </mrow>
                            <annotation encoding="MathType-MTEF">MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaaaaa@369C@</annotation>
                          </semantics>
                        </mrow></semantics></math></script></span> is referred to as signed module eigengene (ME) based connectivity measure <em xmlns="" class="EmphasisTypeItalic">K</em><sub>
                      <em xmlns="" class="EmphasisTypeItalic">ME</em>
                    </sub>. This is the reason why we named the corresponding R function signedKME.</p>
                </section>
                <section id="Sec9" class="Section3 RenderAsSection3">
                  <h4 class="Heading">Automatic block-wise module detection</h4>
                  <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Many microarray gene expression measurements report expression levels of tens of thousands of distinct genes (or probes). Building and analyzing a full network among such a large number of nodes can be computationally challenging because of memory size and processor speed limitations. The WGCNA package contains several improvements that address this challenge. The function blockwiseModules is designed to handle network construction and module detection in large data sets. The function first pre-clusters nodes into large clusters, referred to as blocks, using a variant of k-means clustering (function projectiveKMeans). Next, hierarchical clustering is applied to each block and modules are defined as branches of the resulting dendrogram. To synthesize the module detection results across blocks, an automatic module merging step (function mergeCloseModules) is performed that merges modules whose eigengenes are highly correlated. The time and memory savings of the block-wise approach are substantial: a standard, single-block network analysis of <em xmlns="" class="EmphasisTypeItalic">n</em> nodes requires <em xmlns="" class="EmphasisTypeItalic">O</em>(<em xmlns="" class="EmphasisTypeItalic">n</em><sup>2</sup>) memory and <em xmlns="" class="EmphasisTypeItalic">O</em>(<em xmlns="" class="EmphasisTypeItalic">n</em><sup>3</sup>) calculations, while the block-wise approach with block size <em xmlns="" class="EmphasisTypeItalic">n</em><sub>
                      <em xmlns="" class="EmphasisTypeItalic">b</em>
                    </sub>requires only <em xmlns="" class="EmphasisTypeItalic">O</em>(<span class="InlineEquation" id="IEq5"><span class="MathJax_Preview" style=""></span><span class="MathJax" id="MathJax-Element-9-Frame" tabindex="0" data-mathml="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot;&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;msubsup&gt;&lt;mi&gt;n&lt;/mi&gt;&lt;mi&gt;b&lt;/mi&gt;&lt;mn&gt;2&lt;/mn&gt;&lt;/msubsup&gt;&lt;/mrow&gt;&lt;annotation encoding=&quot;MathType-MTEF&quot;&gt;          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemOBa42aa0baaSqaaiabdkgaIbqaaiabikdaYaaaaaa@2FA6@&lt;/annotation&gt;&lt;/semantics&gt;&lt;/mrow&gt;&lt;/semantics&gt;&lt;/math&gt;" role="presentation" style="position: relative;"><nobr aria-hidden="true"><span class="math" id="MathJax-Span-208" style="width: 1.116em; display: inline-block;"><span style="display: inline-block; position: relative; width: 1.01em; height: 0px; font-size: 111%;"><span style="position: absolute; clip: rect(1.169em, 1001.01em, 2.599em, -999.997em); top: -2.117em; left: 0em;"><span class="mrow" id="MathJax-Span-209"><span class="semantics" id="MathJax-Span-210"><span class="mrow" id="MathJax-Span-211"><span class="semantics" id="MathJax-Span-212"><span class="mrow" id="MathJax-Span-213"><span class="msubsup" id="MathJax-Span-214"><span style="display: inline-block; position: relative; width: 1.01em; height: 0px;"><span style="position: absolute; clip: rect(3.394em, 1000.59em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-215" style="font-family: MathJax_Math-italic;">n</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.341em, 1000.43em, 4.136em, -999.997em); top: -4.29em; left: 0.586em;"><span class="mn" id="MathJax-Span-216" style="font-size: 70.7%; font-family: MathJax_Main;">2</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.341em, 1000.37em, 4.136em, -999.997em); top: -3.654em; left: 0.586em;"><span class="mi" id="MathJax-Span-217" style="font-size: 70.7%; font-family: MathJax_Math-italic;">b</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 2.122em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -0.409em; border-left: 0px solid; width: 0px; height: 1.415em;"></span></span></nobr><span class="MJX_Assistive_MathML" role="presentation"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><semantics><mrow><msubsup><mi>n</mi><mi>b</mi><mn>2</mn></msubsup></mrow><annotation encoding="MathType-MTEF">          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemOBa42aa0baaSqaaiabdkgaIbqaaiabikdaYaaaaaa@2FA6@</annotation></semantics></mrow></semantics></math></span></span><script type="math/mml" id="MathJax-Element-9"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow>
                          <semantics>
                            <mrow>
                              <msubsup>
                                <mi>n</mi>
                                <mi>b</mi>
                                <mn>2</mn>
                              </msubsup>
                            </mrow>
                            <annotation encoding="MathType-MTEF">MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemOBa42aa0baaSqaaiabdkgaIbqaaiabikdaYaaaaaa@2FA6@</annotation>
                          </semantics>
                        </mrow></semantics></math></script></span>) memory and <em xmlns="" class="EmphasisTypeItalic">O</em>(<em xmlns="" class="EmphasisTypeItalic">n</em><span class="InlineEquation" id="IEq6"><span class="MathJax_Preview" style=""></span><span class="MathJax" id="MathJax-Element-10-Frame" tabindex="0" data-mathml="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot;&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;msubsup&gt;&lt;mi&gt;n&lt;/mi&gt;&lt;mi&gt;b&lt;/mi&gt;&lt;mn&gt;2&lt;/mn&gt;&lt;/msubsup&gt;&lt;/mrow&gt;&lt;annotation encoding=&quot;MathType-MTEF&quot;&gt;          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemOBa42aa0baaSqaaiabdkgaIbqaaiabikdaYaaaaaa@2FA6@&lt;/annotation&gt;&lt;/semantics&gt;&lt;/mrow&gt;&lt;/semantics&gt;&lt;/math&gt;" role="presentation" style="position: relative;"><nobr aria-hidden="true"><span class="math" id="MathJax-Span-218" style="width: 1.116em; display: inline-block;"><span style="display: inline-block; position: relative; width: 1.01em; height: 0px; font-size: 111%;"><span style="position: absolute; clip: rect(1.169em, 1001.01em, 2.599em, -999.997em); top: -2.117em; left: 0em;"><span class="mrow" id="MathJax-Span-219"><span class="semantics" id="MathJax-Span-220"><span class="mrow" id="MathJax-Span-221"><span class="semantics" id="MathJax-Span-222"><span class="mrow" id="MathJax-Span-223"><span class="msubsup" id="MathJax-Span-224"><span style="display: inline-block; position: relative; width: 1.01em; height: 0px;"><span style="position: absolute; clip: rect(3.394em, 1000.59em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-225" style="font-family: MathJax_Math-italic;">n</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.341em, 1000.43em, 4.136em, -999.997em); top: -4.29em; left: 0.586em;"><span class="mn" id="MathJax-Span-226" style="font-size: 70.7%; font-family: MathJax_Main;">2</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.341em, 1000.37em, 4.136em, -999.997em); top: -3.654em; left: 0.586em;"><span class="mi" id="MathJax-Span-227" style="font-size: 70.7%; font-family: MathJax_Math-italic;">b</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 2.122em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -0.409em; border-left: 0px solid; width: 0px; height: 1.415em;"></span></span></nobr><span class="MJX_Assistive_MathML" role="presentation"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><semantics><mrow><msubsup><mi>n</mi><mi>b</mi><mn>2</mn></msubsup></mrow><annotation encoding="MathType-MTEF">          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemOBa42aa0baaSqaaiabdkgaIbqaaiabikdaYaaaaaa@2FA6@</annotation></semantics></mrow></semantics></math></span></span><script type="math/mml" id="MathJax-Element-10"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow>
                          <semantics>
                            <mrow>
                              <msubsup>
                                <mi>n</mi>
                                <mi>b</mi>
                                <mn>2</mn>
                              </msubsup>
                            </mrow>
                            <annotation encoding="MathType-MTEF">MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemOBa42aa0baaSqaaiabdkgaIbqaaiabikdaYaaaaaa@2FA6@</annotation>
                          </semantics>
                        </mrow></semantics></math></script></span>) calculations, making an analysis of say 50 000 genes in blocks of 7 000 feasible on a standard computer.</p>
                </section>
                <section id="Sec10" class="Section3 RenderAsSection3">
                  <h4 class="Heading">Consensus module detection</h4>
                  <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">When dealing with multiple adjacency matrices representing different networks, it can be interesting to find <em xmlns="" class="EmphasisTypeItalic">consensus modules</em>, defined as modules that are present in all or most networks [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR12" data-toggle="citation">12</a></span>]. Intuitively, two nodes should be connected in a consensus network only if all of the input networks agree on that connection. This naturally suggest to define the consensus network similarity between two nodes as the minimum of the input network similarities. In certain cases it may be useful to replace minimum by a suitable quantile (e.g. the first quartile) since the resulting measure may be more robust. Consensus module detection can be performed step-by-step for maximum control and exibility, or in one step using the function blockwiseConsensusModule that calculates consensus modules across given data sets in a block-wise manner analogous to the block-wise module detection in a single data set.</p>
                </section>
              </section><section id="Sec11" class="Section2 RenderAsSection2">
                <h3 class="Heading">Category 3: Functions for module and gene selection</h3>
                <div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Finding biologically or clinically significant modules and genes is a major goal of many co-expression analyses. The definition of biological or clinical significance depends on the research question under consideration. Abstractly speaking, we define a gene significance measure as a function <em xmlns="" class="EmphasisTypeItalic">GS</em> that assigns a non-negative number to each gene; the higher <em xmlns="" class="EmphasisTypeItalic">GS</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">i</em>
                  </sub>the more <em xmlns="" class="EmphasisTypeItalic">biologically</em> significant is gene <em xmlns="" class="EmphasisTypeItalic">i</em>. In functional enrichment analysis, a gene significance measure could indicate pathway membership. In gene knock-out experiments, gene significance could indicate knock-out essentiality. A microarray sample trait <em xmlns="" class="EmphasisTypeItalic">T</em> can be used to define a trait-based gene significance measure as the absolute correlation between the trait and the expression profiles, Equation <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Equ2">2</a></span>. A measure of module significance can be defined as average gene significance across the module genes (Figure <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig3">3A</a></span>). When dealing with a sample trait <em xmlns="" class="EmphasisTypeItalic">T</em>, a measure of statistical significance between the module eigengene <em xmlns="" class="EmphasisTypeItalic">E</em> and the trait <em xmlns="" class="EmphasisTypeItalic">T</em> can be defined, for example, using correlation (Equation 2) or a p-value (Equation 3) obtained from a univariate regression model between <em xmlns="" class="EmphasisTypeItalic">E</em> and <em xmlns="" class="EmphasisTypeItalic">T</em>. Modules with high trait significance may represent pathways associated with the sample trait. Genes with high module membership in modules related to traits (Figure <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig3">3B</a></span>) are natural candidates for further validation [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR10" data-toggle="citation">10</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR14" data-toggle="citation">14</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR15" data-toggle="citation">15</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR18" data-toggle="citation">18</a></span>].<figure xmlns="" class="Figure" id="Fig3"><div xmlns="http://www.w3.org/1999/xhtml" class="MediaObject">
                      <a target="_blank" rel="noopener" href="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_Article_2544_Fig3_HTML.jpg"><img class="c-lazy-image u-js-shown" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/12859_2008_Article_2544_Fig3_HTML.jpg" alt="Figure 3" data-lazy-load="true" data-src="https://media.springernature.com/lw785/springer-static/image/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_Article_2544_Fig3_HTML.jpg"><noscript><img src="https://media.springernature.com/lw785/springer-static/image/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_Article_2544_Fig3_HTML.jpg" alt="Figure 3"></noscript></a>
                    </div><figcaption class="Caption" lang="en"><div xmlns="http://www.w3.org/1999/xhtml" class="CaptionContent"><span class="CaptionNumber">Figure 3</span>
                        <p xmlns="" class="SimplePara"><strong class="EmphasisTypeBold">Module and eigengene network plots</strong>. A. Barplot of mean gene significance across modules. In this example we use a trait-based gene significance, Equation <span xmlns="http://www.w3.org/1999/xhtml" class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Equ2">2</a></span>. The higher the mean gene significance in a module, the more significantly related the module is to the clinical trait of interest. B. Scatterplot of gene significance (<em class="EmphasisTypeItalic">y</em>-axis) vs. module membership (<em class="EmphasisTypeItalic">x</em>-axis) in the most significant module (green module, see panel A). In modules related to a trait of interest, genes with high module membership often also have high gene significance. C. Hierarchical clustering dendrogram of module eigengenes (labeled by their colors) and the microarray sample trait <em class="EmphasisTypeItalic">y</em>. D. Heatmap plot of the adjacencies in the eigengene network including the trait <em class="EmphasisTypeItalic">y</em>. Each row and column in the heatmap corresponds to one module eigengene (labeled by color) or the trait (labeled by <em class="EmphasisTypeItalic">y</em>). In the heatmap, green color represents low adjacency (negative correlation), while red represents high adjacency (positive correlation).</p>
                      </div></figcaption></figure></div>
              </section><section id="Sec12" class="Section2 RenderAsSection2">
                <h3 class="Heading">Category 4: Functions for studying topological properties</h3>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Many topological properties of networks can be succinctly described using network concepts, also known as network statistics or indices [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR11" data-toggle="citation">11</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR33" data-toggle="citation">33</a></span>]. Network concepts include whole network connectivity (degree), intramodular connectivity, topological overlap, the clustering coefficient, density etc. Differential analysis of network concepts such as intramodular connectivity may reveal regulatory changes in gene expressions [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR15" data-toggle="citation">15</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR18" data-toggle="citation">18</a></span>]. The WGCNA package implements several functions, such as softConnectivity, intramodularConnectivity, TOMSimilarity, clusterCoef, networkConcepts, for computing these network concepts. Basic R functions can be used to create summary statistics of these concepts and for testing their differences across networks.</p>
                <section id="Sec13" class="Section3 RenderAsSection3">
                  <h4 class="Heading">Network concepts for measuring cluster structure</h4>
                  <div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Gene clustering trees and TOM plots that visualize interconnectivity patterns often suggest the presence of large modules. Network theory offers a wealth of intuitive network concepts for describing the pairwise relationships among genes that are depicted in cluster trees and heat maps [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR11" data-toggle="citation">11</a></span>]. To illustrate this point, we describe two network concepts in the following. By visual inspection of Figures <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig2">2C</a></span> and <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig4">4B</a></span>, genes appear to be highly interconnected, e.g. turquoise module genes form a reddish square in the TOM plot. This property of dense connections among the genes of module <em xmlns="" class="EmphasisTypeItalic">q</em> can be measured using the concept of module density, which is defined as the average adjacency of the module genes:<figure xmlns="" class="Figure" id="Fig4"><div xmlns="http://www.w3.org/1999/xhtml" class="MediaObject">
                        <a target="_blank" rel="noopener" href="https://media.springernature.com/full/springer-static/image/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_Article_2544_Fig4_HTML.jpg"><img class="c-lazy-image u-js-shown" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/12859_2008_Article_2544_Fig4_HTML.jpg" alt="Figure 4" data-lazy-load="true" data-src="https://media.springernature.com/lw785/springer-static/image/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_Article_2544_Fig4_HTML.jpg"><noscript><img src="https://media.springernature.com/lw785/springer-static/image/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_Article_2544_Fig4_HTML.jpg" alt="Figure 4"></noscript></a>
                      </div><figcaption class="Caption" lang="en"><div xmlns="http://www.w3.org/1999/xhtml" class="CaptionContent"><span class="CaptionNumber">Figure 4</span>
                          <p xmlns="" class="SimplePara"><strong class="EmphasisTypeBold">Example WGCNA analysis of liver expression data in female mice</strong>. A. Gene dendrogram obtained by average linkage hierarchical clustering. The color row underneath the dendrogram shows the module assignment determined by the Dynamic Tree Cut. B. Heatmap plot of topological overlap in the gene network. In the heatmap, each row and column corresponds to a gene, light color denotes low topological overlap, and progressively darker red denotes higher topological overlap. Darker squares along the diagonal correspond to modules. The gene dendrogram and module assignment are shown along the left and top. C. Hierarchical clustering of module eigengenes that summarize the modules found in the clustering analysis. Branches of the dendrogram (the meta-modules) group together eigengenes that are positively correlated. D. Heatmap plot of the adjacencies in the eigengene network including the trait weight. Each row and column in the heatmap corresponds to one module eigengene (labeled by color) or weight. In the heatmap, green color represents low adjacency (negative correlation), while red represents high adjacency (positive correlation). Squares of red color along the diagonal are the meta-modules. E. A scatterplot of gene significance for weight (GS, Equation <span xmlns="http://www.w3.org/1999/xhtml" class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Equ2">2</a></span>) versus module membership (MM, Equation <span xmlns="http://www.w3.org/1999/xhtml" class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Equ6">6</a></span>) in the brown module. GS and MM exhibit a very significant correlation, implying that hub genes of the brown module also tend to be highly correlated with weight. F. The network of the 30 most highly connected genes in the brown module. In this network we only display a connection of the corresponding topological overlap is above a threshold of 0.08.</p>
                        </div></figcaption></figure></div>
                  <div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">
                    <div id="Equ5" class="Equation EquationMathjax"><div class="EquationContent"><span class="MathJax_Preview" style=""></span><span class="MathJax" id="MathJax-Element-11-Frame" tabindex="0" data-mathml="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot;&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mi&gt;D&lt;/mi&gt;&lt;mi&gt;e&lt;/mi&gt;&lt;mi&gt;n&lt;/mi&gt;&lt;mi&gt;s&lt;/mi&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mi&gt;t&lt;/mi&gt;&lt;mi&gt;y&lt;/mi&gt;&lt;mo stretchy=&quot;false&quot;&gt;(&lt;/mo&gt;&lt;msup&gt;&lt;mi&gt;A&lt;/mi&gt;&lt;mrow&gt;&lt;mo stretchy=&quot;false&quot;&gt;(&lt;/mo&gt;&lt;mi&gt;q&lt;/mi&gt;&lt;mo stretchy=&quot;false&quot;&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/msup&gt;&lt;mo stretchy=&quot;false&quot;&gt;)&lt;/mo&gt;&lt;mo&gt;=&lt;/mo&gt;&lt;mfrac&gt;&lt;mrow&gt;&lt;mstyle displaystyle=&quot;true&quot;&gt;&lt;msub&gt;&lt;mo&gt;&amp;#x2211;&lt;/mo&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/msub&gt;&lt;mrow&gt;&lt;mstyle displaystyle=&quot;true&quot;&gt;&lt;msub&gt;&lt;mo&gt;&amp;#x2211;&lt;/mo&gt;&lt;mrow&gt;&lt;mi&gt;j&lt;/mi&gt;&lt;mo&gt;&amp;#x2260;&lt;/mo&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/mrow&gt;&lt;/msub&gt;&lt;mrow&gt;&lt;msubsup&gt;&lt;mi&gt;a&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mi&gt;j&lt;/mi&gt;&lt;/mrow&gt;&lt;mrow&gt;&lt;mo stretchy=&quot;false&quot;&gt;(&lt;/mo&gt;&lt;mi&gt;q&lt;/mi&gt;&lt;mo stretchy=&quot;false&quot;&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/msubsup&gt;&lt;/mrow&gt;&lt;/mstyle&gt;&lt;/mrow&gt;&lt;/mstyle&gt;&lt;/mrow&gt;&lt;mrow&gt;&lt;msup&gt;&lt;mi&gt;n&lt;/mi&gt;&lt;mrow&gt;&lt;mo stretchy=&quot;false&quot;&gt;(&lt;/mo&gt;&lt;mi&gt;q&lt;/mi&gt;&lt;mo stretchy=&quot;false&quot;&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/msup&gt;&lt;mo stretchy=&quot;false&quot;&gt;(&lt;/mo&gt;&lt;msup&gt;&lt;mi&gt;n&lt;/mi&gt;&lt;mrow&gt;&lt;mo stretchy=&quot;false&quot;&gt;(&lt;/mo&gt;&lt;mi&gt;q&lt;/mi&gt;&lt;mo stretchy=&quot;false&quot;&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/msup&gt;&lt;mo&gt;&amp;#x2212;&lt;/mo&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;mo stretchy=&quot;false&quot;&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/mfrac&gt;&lt;/mrow&gt;&lt;annotation encoding=&quot;MathType-MTEF&quot;&gt;          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemiraqKaemyzauMaemOBa4Maem4CamNaemyAaKMaemiDaqNaemyEaKNaeiikaGIaemyqae0aaWbaaSqabeaacqGGOaakcqWGXbqCcqGGPaqkaaGccqGGPaqkcqGH9aqpjuaGdaWcaaqaamaaqababaWaaabeaeaacqWGHbqydaqhaaqaaiabdMgaPjabdQgaQbqaaiabcIcaOiabdghaXjabcMcaPaaaaeaacqWGQbGAcqGHGjsUcqWGPbqAaeqacqGHris5aaqaaiabdMgaPbqabiabggHiLdaabaGaemOBa42aaWbaaeqabaGaeiikaGIaemyCaeNaeiykaKcaaiabcIcaOiabd6gaUnaaCaaabeqaaiabcIcaOiabdghaXjabcMcaPaaacqGHsislcqaIXaqmcqGGPaqkaaaaaa@5AE0@&lt;/annotation&gt;&lt;/semantics&gt;&lt;/mrow&gt;&lt;/semantics&gt;&lt;/math&gt;" role="presentation" style="position: relative;"><nobr aria-hidden="true"><span class="math" id="MathJax-Span-228" style="width: 13.463em; display: inline-block;"><span style="display: inline-block; position: relative; width: 12.138em; height: 0px; font-size: 111%;"><span style="position: absolute; clip: rect(0.427em, 1012.14em, 3.023em, -999.997em); top: -2.17em; left: 0em;"><span class="mrow" id="MathJax-Span-229"><span class="semantics" id="MathJax-Span-230"><span class="mrow" id="MathJax-Span-231"><span class="semantics" id="MathJax-Span-232"><span class="mrow" id="MathJax-Span-233"><span class="mi" id="MathJax-Span-234" style="font-family: MathJax_Math-italic;">D</span><span class="mi" id="MathJax-Span-235" style="font-family: MathJax_Math-italic;">e</span><span class="mi" id="MathJax-Span-236" style="font-family: MathJax_Math-italic;">n</span><span class="mi" id="MathJax-Span-237" style="font-family: MathJax_Math-italic;">s</span><span class="mi" id="MathJax-Span-238" style="font-family: MathJax_Math-italic;">i</span><span class="mi" id="MathJax-Span-239" style="font-family: MathJax_Math-italic;">t</span><span class="mi" id="MathJax-Span-240" style="font-family: MathJax_Math-italic;">y<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span><span class="mo" id="MathJax-Span-241" style="font-family: MathJax_Main;">(</span><span class="msup" id="MathJax-Span-242"><span style="display: inline-block; position: relative; width: 1.698em; height: 0px;"><span style="position: absolute; clip: rect(3.076em, 1000.75em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-243" style="font-family: MathJax_Math-italic;">A</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -4.343em; left: 0.745em;"><span class="mrow" id="MathJax-Span-244"><span class="mo" id="MathJax-Span-245" style="font-size: 70.7%; font-family: MathJax_Main;">(</span><span class="mi" id="MathJax-Span-246" style="font-size: 70.7%; font-family: MathJax_Math-italic;">q<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span><span class="mo" id="MathJax-Span-247" style="font-size: 70.7%; font-family: MathJax_Main;">)</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mo" id="MathJax-Span-248" style="font-family: MathJax_Main;">)</span><span class="mo" id="MathJax-Span-249" style="font-family: MathJax_Main; padding-left: 0.268em;">=</span><span class="mfrac" id="MathJax-Span-250" style="padding-left: 0.268em;"><span style="display: inline-block; position: relative; width: 4.56em; height: 0px; margin-right: 0.109em; margin-left: 0.109em;"><span style="position: absolute; clip: rect(3.076em, 1004.45em, 4.613em, -999.997em); top: -4.82em; left: 50%; margin-left: -2.223em;"><span class="mrow" id="MathJax-Span-251"><span class="mstyle" id="MathJax-Span-252"><span class="mrow" id="MathJax-Span-253"><span class="msub" id="MathJax-Span-254"><span style="display: inline-block; position: relative; width: 1.275em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1001.01em, 4.454em, -999.997em); top: -3.972em; left: 0em;"><span class="mo" id="MathJax-Span-255" style="font-size: 70.7%; font-family: MathJax_Size2; vertical-align: 0em;">∑</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.654em; left: 1.01em;"><span class="mi" id="MathJax-Span-256" style="font-size: 50%; font-family: MathJax_Math-italic;">i</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mrow" id="MathJax-Span-257" style="padding-left: 0.162em;"><span class="mstyle" id="MathJax-Span-258"><span class="mrow" id="MathJax-Span-259"><span class="msub" id="MathJax-Span-260"><span style="display: inline-block; position: relative; width: 1.857em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1001.01em, 4.454em, -999.997em); top: -3.972em; left: 0em;"><span class="mo" id="MathJax-Span-261" style="font-size: 70.7%; font-family: MathJax_Size2; vertical-align: 0em;">∑</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.654em; left: 1.01em;"><span class="mrow" id="MathJax-Span-262"><span class="mi" id="MathJax-Span-263" style="font-size: 50%; font-family: MathJax_Math-italic;">j</span><span class="mo" id="MathJax-Span-264" style="font-size: 50%; font-family: MathJax_Main;">≠</span><span class="mi" id="MathJax-Span-265" style="font-size: 50%; font-family: MathJax_Math-italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mrow" id="MathJax-Span-266" style="padding-left: 0.162em;"><span class="msubsup" id="MathJax-Span-267"><span style="display: inline-block; position: relative; width: 1.063em; height: 0px;"><span style="position: absolute; clip: rect(3.5em, 1000.37em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-268" style="font-size: 70.7%; font-family: MathJax_Math-italic;">a</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.447em, 1000.69em, 4.242em, -999.997em); top: -4.343em; left: 0.374em;"><span class="mrow" id="MathJax-Span-269"><span class="mo" id="MathJax-Span-270" style="font-size: 50%; font-family: MathJax_Main;">(</span><span class="mi" id="MathJax-Span-271" style="font-size: 50%; font-family: MathJax_Math-italic;">q<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span><span class="mo" id="MathJax-Span-272" style="font-size: 50%; font-family: MathJax_Main;">)</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.5em, 1000.43em, 4.242em, -999.997em); top: -3.76em; left: 0.374em;"><span class="mrow" id="MathJax-Span-273"><span class="mi" id="MathJax-Span-274" style="font-size: 50%; font-family: MathJax_Math-italic;">i</span><span class="mi" id="MathJax-Span-275" style="font-size: 50%; font-family: MathJax_Math-italic;">j</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.235em, 1003.61em, 4.295em, -999.997em); top: -3.495em; left: 50%; margin-left: -1.799em;"><span class="mrow" id="MathJax-Span-276"><span class="msup" id="MathJax-Span-277"><span style="display: inline-block; position: relative; width: 1.116em; height: 0px;"><span style="position: absolute; clip: rect(3.5em, 1000.43em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-278" style="font-size: 70.7%; font-family: MathJax_Math-italic;">n</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -4.184em; left: 0.427em;"><span class="mrow" id="MathJax-Span-279"><span class="mo" id="MathJax-Span-280" style="font-size: 50%; font-family: MathJax_Main;">(</span><span class="mi" id="MathJax-Span-281" style="font-size: 50%; font-family: MathJax_Math-italic;">q<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span><span class="mo" id="MathJax-Span-282" style="font-size: 50%; font-family: MathJax_Main;">)</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mo" id="MathJax-Span-283" style="font-size: 70.7%; font-family: MathJax_Main;">(</span><span class="msup" id="MathJax-Span-284"><span style="display: inline-block; position: relative; width: 1.116em; height: 0px;"><span style="position: absolute; clip: rect(3.5em, 1000.43em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-285" style="font-size: 70.7%; font-family: MathJax_Math-italic;">n</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -4.184em; left: 0.427em;"><span class="mrow" id="MathJax-Span-286"><span class="mo" id="MathJax-Span-287" style="font-size: 50%; font-family: MathJax_Main;">(</span><span class="mi" id="MathJax-Span-288" style="font-size: 50%; font-family: MathJax_Math-italic;">q<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span><span class="mo" id="MathJax-Span-289" style="font-size: 50%; font-family: MathJax_Main;">)</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mo" id="MathJax-Span-290" style="font-size: 70.7%; font-family: MathJax_Main;">−</span><span class="mn" id="MathJax-Span-291" style="font-size: 70.7%; font-family: MathJax_Main;">1</span><span class="mo" id="MathJax-Span-292" style="font-size: 70.7%; font-family: MathJax_Main;">)</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(0.851em, 1004.56em, 1.222em, -999.997em); top: -1.269em; left: 0em;"><span style="display: inline-block; overflow: hidden; vertical-align: 0em; border-top: 1.3px solid; width: 4.56em; height: 0px;"></span><span style="display: inline-block; width: 0px; height: 1.063em;"></span></span></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 2.175em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -0.821em; border-left: 0px solid; width: 0px; height: 2.65em;"></span></span></nobr><span class="MJX_Assistive_MathML" role="presentation"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><semantics><mrow><mi>D</mi><mi>e</mi><mi>n</mi><mi>s</mi><mi>i</mi><mi>t</mi><mi>y</mi><mo stretchy="false">(</mo><msup><mi>A</mi><mrow><mo stretchy="false">(</mo><mi>q</mi><mo stretchy="false">)</mo></mrow></msup><mo stretchy="false">)</mo><mo>=</mo><mfrac><mrow><mstyle displaystyle="true"><msub><mo>∑</mo><mi>i</mi></msub><mrow><mstyle displaystyle="true"><msub><mo>∑</mo><mrow><mi>j</mi><mo>≠</mo><mi>i</mi></mrow></msub><mrow><msubsup><mi>a</mi><mrow><mi>i</mi><mi>j</mi></mrow><mrow><mo stretchy="false">(</mo><mi>q</mi><mo stretchy="false">)</mo></mrow></msubsup></mrow></mstyle></mrow></mstyle></mrow><mrow><msup><mi>n</mi><mrow><mo stretchy="false">(</mo><mi>q</mi><mo stretchy="false">)</mo></mrow></msup><mo stretchy="false">(</mo><msup><mi>n</mi><mrow><mo stretchy="false">(</mo><mi>q</mi><mo stretchy="false">)</mo></mrow></msup><mo>−</mo><mn>1</mn><mo stretchy="false">)</mo></mrow></mfrac></mrow><annotation encoding="MathType-MTEF">          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemiraqKaemyzauMaemOBa4Maem4CamNaemyAaKMaemiDaqNaemyEaKNaeiikaGIaemyqae0aaWbaaSqabeaacqGGOaakcqWGXbqCcqGGPaqkaaGccqGGPaqkcqGH9aqpjuaGdaWcaaqaamaaqababaWaaabeaeaacqWGHbqydaqhaaqaaiabdMgaPjabdQgaQbqaaiabcIcaOiabdghaXjabcMcaPaaaaeaacqWGQbGAcqGHGjsUcqWGPbqAaeqacqGHris5aaqaaiabdMgaPbqabiabggHiLdaabaGaemOBa42aaWbaaeqabaGaeiikaGIaemyCaeNaeiykaKcaaiabcIcaOiabd6gaUnaaCaaabeqaaiabcIcaOiabdghaXjabcMcaPaaacqGHsislcqaIXaqmcqGGPaqkaaaaaa@5AE0@</annotation></semantics></mrow></semantics></math></span></span><script type="math/mml" id="MathJax-Element-11"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow>
                          <semantics>
                            <mrow>
                              <mi>D</mi>
                              <mi>e</mi>
                              <mi>n</mi>
                              <mi>s</mi>
                              <mi>i</mi>
                              <mi>t</mi>
                              <mi>y</mi>
                              <mo stretchy="false">(</mo>
                              <msup>
                                <mi>A</mi>
                                <mrow>
                                  <mo stretchy="false">(</mo>
                                  <mi>q</mi>
                                  <mo stretchy="false">)</mo>
                                </mrow>
                              </msup>
                              <mo stretchy="false">)</mo>
                              <mo>=</mo>
                              <mfrac>
                                <mrow>
                                  <mstyle displaystyle="true">
                                    <msub>
                                      <mo>∑</mo>
                                      <mi>i</mi>
                                    </msub>
                                    <mrow>
                                      <mstyle displaystyle="true">
                                        <msub>
                                          <mo>∑</mo>
                                          <mrow>
                                            <mi>j</mi>
                                            <mo>≠</mo>
                                            <mi>i</mi>
                                          </mrow>
                                        </msub>
                                        <mrow>
                                          <msubsup>
                                            <mi>a</mi>
                                            <mrow>
                                              <mi>i</mi>
                                              <mi>j</mi>
                                            </mrow>
                                            <mrow>
                                              <mo stretchy="false">(</mo>
                                              <mi>q</mi>
                                              <mo stretchy="false">)</mo>
                                            </mrow>
                                          </msubsup>
                                        </mrow>
                                      </mstyle>
                                    </mrow>
                                  </mstyle>
                                </mrow>
                                <mrow>
                                  <msup>
                                    <mi>n</mi>
                                    <mrow>
                                      <mo stretchy="false">(</mo>
                                      <mi>q</mi>
                                      <mo stretchy="false">)</mo>
                                    </mrow>
                                  </msup>
                                  <mo stretchy="false">(</mo>
                                  <msup>
                                    <mi>n</mi>
                                    <mrow>
                                      <mo stretchy="false">(</mo>
                                      <mi>q</mi>
                                      <mo stretchy="false">)</mo>
                                    </mrow>
                                  </msup>
                                  <mo>−</mo>
                                  <mn>1</mn>
                                  <mo stretchy="false">)</mo>
                                </mrow>
                              </mfrac>
                            </mrow>
                            <annotation encoding="MathType-MTEF">MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaemiraqKaemyzauMaemOBa4Maem4CamNaemyAaKMaemiDaqNaemyEaKNaeiikaGIaemyqae0aaWbaaSqabeaacqGGOaakcqWGXbqCcqGGPaqkaaGccqGGPaqkcqGH9aqpjuaGdaWcaaqaamaaqababaWaaabeaeaacqWGHbqydaqhaaqaaiabdMgaPjabdQgaQbqaaiabcIcaOiabdghaXjabcMcaPaaaaeaacqWGQbGAcqGHGjsUcqWGPbqAaeqacqGHris5aaqaaiabdMgaPbqabiabggHiLdaabaGaemOBa42aaWbaaeqabaGaeiikaGIaemyCaeNaeiykaKcaaiabcIcaOiabd6gaUnaaCaaabeqaaiabcIcaOiabdghaXjabcMcaPaaacqGHsislcqaIXaqmcqGGPaqkaaaaaa@5AE0@</annotation>
                          </semantics>
                        </mrow></semantics></math></script></div> <div class="EquationNumber">(7)</div></div>
                  </div>
                  <div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">where <em xmlns="" class="EmphasisTypeItalic">A</em><sup>(<em xmlns="" class="EmphasisTypeItalic">q</em>)</sup>denotes the <em xmlns="" class="EmphasisTypeItalic">n</em><sup>(<em xmlns="" class="EmphasisTypeItalic">q</em>)</sup>× <em xmlns="" class="EmphasisTypeItalic">n</em><sup>(<em xmlns="" class="EmphasisTypeItalic">q</em>)</sup>adjacency matrix corresponding to the sub-network formed by the genes of module <em xmlns="" class="EmphasisTypeItalic">q</em>. Another useful concept is the <em xmlns="" class="EmphasisTypeItalic">clustering coefficient</em> of gene <em xmlns="" class="EmphasisTypeItalic">i</em>, which is a measure of 'cliquishness' [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR34" data-toggle="citation">34</a></span>]. Specifically,<div id="Equ6" class="Equation EquationMathjax"><div class="EquationContent"><span class="MathJax_Preview" style=""></span><span class="MathJax" id="MathJax-Element-12-Frame" tabindex="0" data-mathml="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot;&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mi&gt;C&lt;/mi&gt;&lt;mi&gt;l&lt;/mi&gt;&lt;mi&gt;u&lt;/mi&gt;&lt;mi&gt;s&lt;/mi&gt;&lt;mi&gt;t&lt;/mi&gt;&lt;mi&gt;e&lt;/mi&gt;&lt;mi&gt;r&lt;/mi&gt;&lt;mi&gt;C&lt;/mi&gt;&lt;mi&gt;o&lt;/mi&gt;&lt;mi&gt;e&lt;/mi&gt;&lt;msub&gt;&lt;mi&gt;f&lt;/mi&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/msub&gt;&lt;mo&gt;=&lt;/mo&gt;&lt;mfrac&gt;&lt;mrow&gt;&lt;mstyle displaystyle=&quot;true&quot;&gt;&lt;msub&gt;&lt;mo&gt;&amp;#x2211;&lt;/mo&gt;&lt;mrow&gt;&lt;mi&gt;l&lt;/mi&gt;&lt;mo&gt;&amp;#x2260;&lt;/mo&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/mrow&gt;&lt;/msub&gt;&lt;mrow&gt;&lt;mstyle displaystyle=&quot;true&quot;&gt;&lt;msub&gt;&lt;mo&gt;&amp;#x2211;&lt;/mo&gt;&lt;mrow&gt;&lt;mi&gt;m&lt;/mi&gt;&lt;mo&gt;&amp;#x2260;&lt;/mo&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mo&gt;,&lt;/mo&gt;&lt;mi&gt;l&lt;/mi&gt;&lt;/mrow&gt;&lt;/msub&gt;&lt;mrow&gt;&lt;msub&gt;&lt;mi&gt;a&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mi&gt;l&lt;/mi&gt;&lt;/mrow&gt;&lt;/msub&gt;&lt;msub&gt;&lt;mi&gt;a&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;l&lt;/mi&gt;&lt;mi&gt;m&lt;/mi&gt;&lt;/mrow&gt;&lt;/msub&gt;&lt;msub&gt;&lt;mi&gt;a&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;m&lt;/mi&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/mrow&gt;&lt;/msub&gt;&lt;/mrow&gt;&lt;/mstyle&gt;&lt;/mrow&gt;&lt;/mstyle&gt;&lt;/mrow&gt;&lt;mrow&gt;&lt;mrow&gt;&lt;mo&gt;{&lt;/mo&gt;&lt;mrow&gt;&lt;msup&gt;&lt;mrow&gt;&lt;mrow&gt;&lt;mo&gt;(&lt;/mo&gt;&lt;mrow&gt;&lt;mstyle displaystyle=&quot;true&quot;&gt;&lt;msub&gt;&lt;mo&gt;&amp;#x2211;&lt;/mo&gt;&lt;mrow&gt;&lt;mi&gt;l&lt;/mi&gt;&lt;mo&gt;&amp;#x2260;&lt;/mo&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/mrow&gt;&lt;/msub&gt;&lt;mrow&gt;&lt;msub&gt;&lt;mi&gt;a&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mi&gt;l&lt;/mi&gt;&lt;/mrow&gt;&lt;/msub&gt;&lt;/mrow&gt;&lt;/mstyle&gt;&lt;/mrow&gt;&lt;mo&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/mrow&gt;&lt;mn&gt;2&lt;/mn&gt;&lt;/msup&gt;&lt;mo&gt;&amp;#x2212;&lt;/mo&gt;&lt;mstyle displaystyle=&quot;true&quot;&gt;&lt;msub&gt;&lt;mo&gt;&amp;#x2211;&lt;/mo&gt;&lt;mrow&gt;&lt;mi&gt;l&lt;/mi&gt;&lt;mo&gt;&amp;#x2260;&lt;/mo&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/mrow&gt;&lt;/msub&gt;&lt;mrow&gt;&lt;msup&gt;&lt;mrow&gt;&lt;mo stretchy=&quot;false&quot;&gt;(&lt;/mo&gt;&lt;msub&gt;&lt;mi&gt;a&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mi&gt;l&lt;/mi&gt;&lt;/mrow&gt;&lt;/msub&gt;&lt;mo stretchy=&quot;false&quot;&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;mn&gt;2&lt;/mn&gt;&lt;/msup&gt;&lt;/mrow&gt;&lt;/mstyle&gt;&lt;/mrow&gt;&lt;mo&gt;}&lt;/mo&gt;&lt;/mrow&gt;&lt;/mrow&gt;&lt;/mfrac&gt;&lt;mo&gt;.&lt;/mo&gt;&lt;/mrow&gt;&lt;annotation encoding=&quot;MathType-MTEF&quot;&gt;          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4qamKaemiBaWMaemyDauNaem4CamNaemiDaqNaemyzauMaemOCaiNaem4qamKaem4Ba8MaemyzauMaemOzay2aaSbaaSqaaiabdMgaPbqabaGccqGH9aqpjuaGdaWcaaqaamaaqababaWaaabeaeaacqWGHbqydaWgaaqaaiabdMgaPjabdYgaSbqabaGaemyyae2aaSbaaeaacqWGSbaBcqWGTbqBaeqaaiabdggaHnaaBaaabaGaemyBa0MaemyAaKgabeaaaeaacqWGTbqBcqGHGjsUcqWGPbqAcqGGSaalcqWGSbaBaeqacqGHris5aaqaaiabdYgaSjabgcMi5kabdMgaPbqabiabggHiLdaabaWaaiWaaeaadaqadaqaamaaqababaGaemyyae2aaSbaaeaacqWGPbqAcqWGSbaBaeqaaaqaaiabdYgaSjabgcMi5kabdMgaPbqabiabggHiLdaacaGLOaGaayzkaaWaaWbaaeqabaGaeGOmaidaaiabgkHiTmaaqababaGaeiikaGIaemyyae2aaSbaaeaacqWGPbqAcqWGSbaBaeqaaiabcMcaPmaaCaaabeqaaiabikdaYaaaaeaacqWGSbaBcqGHGjsUcqWGPbqAaeqacqGHris5aaGaay5Eaiaaw2haaaaacqGGUaGlaaa@77AB@&lt;/annotation&gt;&lt;/semantics&gt;&lt;/mrow&gt;&lt;/semantics&gt;&lt;/math&gt;" role="presentation" style="position: relative;"><nobr aria-hidden="true"><span class="math" id="MathJax-Span-293" style="width: 18.657em; display: inline-block;"><span style="display: inline-block; position: relative; width: 16.802em; height: 0px; font-size: 111%;"><span style="position: absolute; clip: rect(0.48em, 1016.75em, 3.924em, -999.997em); top: -2.17em; left: 0em;"><span class="mrow" id="MathJax-Span-294"><span class="semantics" id="MathJax-Span-295"><span class="mrow" id="MathJax-Span-296"><span class="semantics" id="MathJax-Span-297"><span class="mrow" id="MathJax-Span-298"><span class="mi" id="MathJax-Span-299" style="font-family: MathJax_Math-italic;">C<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.056em;"></span></span><span class="mi" id="MathJax-Span-300" style="font-family: MathJax_Math-italic;">l</span><span class="mi" id="MathJax-Span-301" style="font-family: MathJax_Math-italic;">u</span><span class="mi" id="MathJax-Span-302" style="font-family: MathJax_Math-italic;">s</span><span class="mi" id="MathJax-Span-303" style="font-family: MathJax_Math-italic;">t</span><span class="mi" id="MathJax-Span-304" style="font-family: MathJax_Math-italic;">e</span><span class="mi" id="MathJax-Span-305" style="font-family: MathJax_Math-italic;">r</span><span class="mi" id="MathJax-Span-306" style="font-family: MathJax_Math-italic;">C<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.056em;"></span></span><span class="mi" id="MathJax-Span-307" style="font-family: MathJax_Math-italic;">o</span><span class="mi" id="MathJax-Span-308" style="font-family: MathJax_Math-italic;">e</span><span class="msub" id="MathJax-Span-309"><span style="display: inline-block; position: relative; width: 0.798em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1000.53em, 4.348em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-310" style="font-family: MathJax_Math-italic;">f<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.056em;"></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.813em; left: 0.48em;"><span class="mi" id="MathJax-Span-311" style="font-size: 70.7%; font-family: MathJax_Math-italic;">i</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mo" id="MathJax-Span-312" style="font-family: MathJax_Main; padding-left: 0.268em;">=</span><span class="mfrac" id="MathJax-Span-313" style="padding-left: 0.268em;"><span style="display: inline-block; position: relative; width: 9.118em; height: 0px; margin-right: 0.109em; margin-left: 0.109em;"><span style="position: absolute; clip: rect(3.129em, 1007.26em, 4.613em, -999.997em); top: -4.82em; left: 50%; margin-left: -3.654em;"><span class="mrow" id="MathJax-Span-314"><span class="mstyle" id="MathJax-Span-315"><span class="mrow" id="MathJax-Span-316"><span class="msub" id="MathJax-Span-317"><span style="display: inline-block; position: relative; width: 1.804em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1001.01em, 4.454em, -999.997em); top: -3.972em; left: 0em;"><span class="mo" id="MathJax-Span-318" style="font-size: 70.7%; font-family: MathJax_Size2; vertical-align: 0em;">∑</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.654em; left: 1.01em;"><span class="mrow" id="MathJax-Span-319"><span class="mi" id="MathJax-Span-320" style="font-size: 50%; font-family: MathJax_Math-italic;">l</span><span class="mo" id="MathJax-Span-321" style="font-size: 50%; font-family: MathJax_Main;">≠</span><span class="mi" id="MathJax-Span-322" style="font-size: 50%; font-family: MathJax_Math-italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mrow" id="MathJax-Span-323" style="padding-left: 0.162em;"><span class="mstyle" id="MathJax-Span-324"><span class="mrow" id="MathJax-Span-325"><span class="msub" id="MathJax-Span-326"><span style="display: inline-block; position: relative; width: 2.387em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1001.01em, 4.454em, -999.997em); top: -3.972em; left: 0em;"><span class="mo" id="MathJax-Span-327" style="font-size: 70.7%; font-family: MathJax_Size2; vertical-align: 0em;">∑</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.654em; left: 1.01em;"><span class="mrow" id="MathJax-Span-328"><span class="mi" id="MathJax-Span-329" style="font-size: 50%; font-family: MathJax_Math-italic;">m</span><span class="mo" id="MathJax-Span-330" style="font-size: 50%; font-family: MathJax_Main;">≠</span><span class="mi" id="MathJax-Span-331" style="font-size: 50%; font-family: MathJax_Math-italic;">i</span><span class="mo" id="MathJax-Span-332" style="font-size: 50%; font-family: MathJax_Main;">,</span><span class="mi" id="MathJax-Span-333" style="font-size: 50%; font-family: MathJax_Math-italic;">l</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mrow" id="MathJax-Span-334" style="padding-left: 0.162em;"><span class="msub" id="MathJax-Span-335"><span style="display: inline-block; position: relative; width: 0.745em; height: 0px;"><span style="position: absolute; clip: rect(3.5em, 1000.37em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-336" style="font-size: 70.7%; font-family: MathJax_Math-italic;">a</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.866em; left: 0.374em;"><span class="mrow" id="MathJax-Span-337"><span class="mi" id="MathJax-Span-338" style="font-size: 50%; font-family: MathJax_Math-italic;">i</span><span class="mi" id="MathJax-Span-339" style="font-size: 50%; font-family: MathJax_Math-italic;">l</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="msub" id="MathJax-Span-340"><span style="display: inline-block; position: relative; width: 1.01em; height: 0px;"><span style="position: absolute; clip: rect(3.5em, 1000.37em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-341" style="font-size: 70.7%; font-family: MathJax_Math-italic;">a</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.866em; left: 0.374em;"><span class="mrow" id="MathJax-Span-342"><span class="mi" id="MathJax-Span-343" style="font-size: 50%; font-family: MathJax_Math-italic;">l</span><span class="mi" id="MathJax-Span-344" style="font-size: 50%; font-family: MathJax_Math-italic;">m</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="msub" id="MathJax-Span-345"><span style="display: inline-block; position: relative; width: 1.063em; height: 0px;"><span style="position: absolute; clip: rect(3.5em, 1000.37em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-346" style="font-size: 70.7%; font-family: MathJax_Math-italic;">a</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.866em; left: 0.374em;"><span class="mrow" id="MathJax-Span-347"><span class="mi" id="MathJax-Span-348" style="font-size: 50%; font-family: MathJax_Math-italic;">m</span><span class="mi" id="MathJax-Span-349" style="font-size: 50%; font-family: MathJax_Math-italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(2.811em, 1008.91em, 4.825em, -999.997em); top: -3.018em; left: 50%; margin-left: -4.502em;"><span class="mrow" id="MathJax-Span-350"><span class="mrow" id="MathJax-Span-351"><span class="mo" id="MathJax-Span-352" style="vertical-align: 0em;"><span><span style="font-size: 70.7%; font-family: MathJax_Size3;">{</span></span></span><span class="mrow" id="MathJax-Span-353"><span class="msup" id="MathJax-Span-354"><span style="display: inline-block; position: relative; width: 3.871em; height: 0px;"><span style="position: absolute; clip: rect(3.023em, 1003.39em, 4.613em, -999.997em); top: -3.972em; left: 0em;"><span class="mrow" id="MathJax-Span-355"><span class="mrow" id="MathJax-Span-356"><span class="mo" id="MathJax-Span-357" style="vertical-align: 0em;"><span><span style="font-size: 70.7%; font-family: MathJax_Size2;">(</span></span></span><span class="mrow" id="MathJax-Span-358"><span class="mstyle" id="MathJax-Span-359"><span class="mrow" id="MathJax-Span-360"><span class="msub" id="MathJax-Span-361"><span style="display: inline-block; position: relative; width: 1.804em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1001.01em, 4.454em, -999.997em); top: -3.972em; left: 0em;"><span class="mo" id="MathJax-Span-362" style="font-size: 70.7%; font-family: MathJax_Size2; vertical-align: 0em;">∑</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.654em; left: 1.01em;"><span class="mrow" id="MathJax-Span-363"><span class="mi" id="MathJax-Span-364" style="font-size: 50%; font-family: MathJax_Math-italic;">l</span><span class="mo" id="MathJax-Span-365" style="font-size: 50%; font-family: MathJax_Main;">≠</span><span class="mi" id="MathJax-Span-366" style="font-size: 50%; font-family: MathJax_Math-italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mrow" id="MathJax-Span-367" style="padding-left: 0.162em;"><span class="msub" id="MathJax-Span-368"><span style="display: inline-block; position: relative; width: 0.745em; height: 0px;"><span style="position: absolute; clip: rect(3.5em, 1000.37em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-369" style="font-size: 70.7%; font-family: MathJax_Math-italic;">a</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.866em; left: 0.374em;"><span class="mrow" id="MathJax-Span-370"><span class="mi" id="MathJax-Span-371" style="font-size: 50%; font-family: MathJax_Math-italic;">i</span><span class="mi" id="MathJax-Span-372" style="font-size: 50%; font-family: MathJax_Math-italic;">l</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span></span><span class="mo" id="MathJax-Span-373" style="vertical-align: 0em;"><span><span style="font-size: 70.7%; font-family: MathJax_Size2;">)</span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -4.608em; left: 3.553em;"><span class="mn" id="MathJax-Span-374" style="font-size: 50%; font-family: MathJax_Main;">2</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mo" id="MathJax-Span-375" style="font-size: 70.7%; font-family: MathJax_Main;">−</span><span class="mstyle" id="MathJax-Span-376"><span class="mrow" id="MathJax-Span-377"><span class="msub" id="MathJax-Span-378"><span style="display: inline-block; position: relative; width: 1.804em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1001.01em, 4.454em, -999.997em); top: -3.972em; left: 0em;"><span class="mo" id="MathJax-Span-379" style="font-size: 70.7%; font-family: MathJax_Size2; vertical-align: 0em;">∑</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.654em; left: 1.01em;"><span class="mrow" id="MathJax-Span-380"><span class="mi" id="MathJax-Span-381" style="font-size: 50%; font-family: MathJax_Math-italic;">l</span><span class="mo" id="MathJax-Span-382" style="font-size: 50%; font-family: MathJax_Main;">≠</span><span class="mi" id="MathJax-Span-383" style="font-size: 50%; font-family: MathJax_Math-italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mrow" id="MathJax-Span-384" style="padding-left: 0.162em;"><span class="msup" id="MathJax-Span-385"><span style="display: inline-block; position: relative; width: 1.592em; height: 0px;"><span style="position: absolute; clip: rect(3.288em, 1001.22em, 4.295em, -999.997em); top: -3.972em; left: 0em;"><span class="mrow" id="MathJax-Span-386"><span class="mo" id="MathJax-Span-387" style="font-size: 70.7%; font-family: MathJax_Main;">(</span><span class="msub" id="MathJax-Span-388"><span style="display: inline-block; position: relative; width: 0.745em; height: 0px;"><span style="position: absolute; clip: rect(3.5em, 1000.37em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-389" style="font-size: 70.7%; font-family: MathJax_Math-italic;">a</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -3.866em; left: 0.374em;"><span class="mrow" id="MathJax-Span-390"><span class="mi" id="MathJax-Span-391" style="font-size: 50%; font-family: MathJax_Math-italic;">i</span><span class="mi" id="MathJax-Span-392" style="font-size: 50%; font-family: MathJax_Math-italic;">l</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span><span class="mo" id="MathJax-Span-393" style="font-size: 70.7%; font-family: MathJax_Main;">)</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; top: -4.29em; left: 1.328em;"><span class="mn" id="MathJax-Span-394" style="font-size: 50%; font-family: MathJax_Main;">2</span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span></span><span class="mo" id="MathJax-Span-395" style="vertical-align: 0em;"><span><span style="font-size: 70.7%; font-family: MathJax_Size3;">}</span></span></span></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(0.851em, 1009.12em, 1.222em, -999.997em); top: -1.269em; left: 0em;"><span style="display: inline-block; overflow: hidden; vertical-align: 0em; border-top: 1.3px solid; width: 9.118em; height: 0px;"></span><span style="display: inline-block; width: 0px; height: 1.063em;"></span></span></span></span><span class="mo" id="MathJax-Span-396" style="font-family: MathJax_Main;">.</span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 2.175em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -1.821em; border-left: 0px solid; width: 0px; height: 3.591em;"></span></span></nobr><span class="MJX_Assistive_MathML" role="presentation"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><semantics><mrow><mi>C</mi><mi>l</mi><mi>u</mi><mi>s</mi><mi>t</mi><mi>e</mi><mi>r</mi><mi>C</mi><mi>o</mi><mi>e</mi><msub><mi>f</mi><mi>i</mi></msub><mo>=</mo><mfrac><mrow><mstyle displaystyle="true"><msub><mo>∑</mo><mrow><mi>l</mi><mo>≠</mo><mi>i</mi></mrow></msub><mrow><mstyle displaystyle="true"><msub><mo>∑</mo><mrow><mi>m</mi><mo>≠</mo><mi>i</mi><mo>,</mo><mi>l</mi></mrow></msub><mrow><msub><mi>a</mi><mrow><mi>i</mi><mi>l</mi></mrow></msub><msub><mi>a</mi><mrow><mi>l</mi><mi>m</mi></mrow></msub><msub><mi>a</mi><mrow><mi>m</mi><mi>i</mi></mrow></msub></mrow></mstyle></mrow></mstyle></mrow><mrow><mrow><mo>{</mo><mrow><msup><mrow><mrow><mo>(</mo><mrow><mstyle displaystyle="true"><msub><mo>∑</mo><mrow><mi>l</mi><mo>≠</mo><mi>i</mi></mrow></msub><mrow><msub><mi>a</mi><mrow><mi>i</mi><mi>l</mi></mrow></msub></mrow></mstyle></mrow><mo>)</mo></mrow></mrow><mn>2</mn></msup><mo>−</mo><mstyle displaystyle="true"><msub><mo>∑</mo><mrow><mi>l</mi><mo>≠</mo><mi>i</mi></mrow></msub><mrow><msup><mrow><mo stretchy="false">(</mo><msub><mi>a</mi><mrow><mi>i</mi><mi>l</mi></mrow></msub><mo stretchy="false">)</mo></mrow><mn>2</mn></msup></mrow></mstyle></mrow><mo>}</mo></mrow></mrow></mfrac><mo>.</mo></mrow><annotation encoding="MathType-MTEF">          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4qamKaemiBaWMaemyDauNaem4CamNaemiDaqNaemyzauMaemOCaiNaem4qamKaem4Ba8MaemyzauMaemOzay2aaSbaaSqaaiabdMgaPbqabaGccqGH9aqpjuaGdaWcaaqaamaaqababaWaaabeaeaacqWGHbqydaWgaaqaaiabdMgaPjabdYgaSbqabaGaemyyae2aaSbaaeaacqWGSbaBcqWGTbqBaeqaaiabdggaHnaaBaaabaGaemyBa0MaemyAaKgabeaaaeaacqWGTbqBcqGHGjsUcqWGPbqAcqGGSaalcqWGSbaBaeqacqGHris5aaqaaiabdYgaSjabgcMi5kabdMgaPbqabiabggHiLdaabaWaaiWaaeaadaqadaqaamaaqababaGaemyyae2aaSbaaeaacqWGPbqAcqWGSbaBaeqaaaqaaiabdYgaSjabgcMi5kabdMgaPbqabiabggHiLdaacaGLOaGaayzkaaWaaWbaaeqabaGaeGOmaidaaiabgkHiTmaaqababaGaeiikaGIaemyyae2aaSbaaeaacqWGPbqAcqWGSbaBaeqaaiabcMcaPmaaCaaabeqaaiabikdaYaaaaeaacqWGSbaBcqGHGjsUcqWGPbqAaeqacqGHris5aaGaay5Eaiaaw2haaaaacqGGUaGlaaa@77AB@</annotation></semantics></mrow></semantics></math></span></span><script type="math/mml" id="MathJax-Element-12"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow>
                          <semantics>
                            <mrow>
                              <mi>C</mi>
                              <mi>l</mi>
                              <mi>u</mi>
                              <mi>s</mi>
                              <mi>t</mi>
                              <mi>e</mi>
                              <mi>r</mi>
                              <mi>C</mi>
                              <mi>o</mi>
                              <mi>e</mi>
                              <msub>
                                <mi>f</mi>
                                <mi>i</mi>
                              </msub>
                              <mo>=</mo>
                              <mfrac>
                                <mrow>
                                  <mstyle displaystyle="true">
                                    <msub>
                                      <mo>∑</mo>
                                      <mrow>
                                        <mi>l</mi>
                                        <mo>≠</mo>
                                        <mi>i</mi>
                                      </mrow>
                                    </msub>
                                    <mrow>
                                      <mstyle displaystyle="true">
                                        <msub>
                                          <mo>∑</mo>
                                          <mrow>
                                            <mi>m</mi>
                                            <mo>≠</mo>
                                            <mi>i</mi>
                                            <mo>,</mo>
                                            <mi>l</mi>
                                          </mrow>
                                        </msub>
                                        <mrow>
                                          <msub>
                                            <mi>a</mi>
                                            <mrow>
                                              <mi>i</mi>
                                              <mi>l</mi>
                                            </mrow>
                                          </msub>
                                          <msub>
                                            <mi>a</mi>
                                            <mrow>
                                              <mi>l</mi>
                                              <mi>m</mi>
                                            </mrow>
                                          </msub>
                                          <msub>
                                            <mi>a</mi>
                                            <mrow>
                                              <mi>m</mi>
                                              <mi>i</mi>
                                            </mrow>
                                          </msub>
                                        </mrow>
                                      </mstyle>
                                    </mrow>
                                  </mstyle>
                                </mrow>
                                <mrow>
                                  <mrow>
                                    <mo>{</mo>
                                    <mrow>
                                      <msup>
                                        <mrow>
                                          <mrow>
                                            <mo>(</mo>
                                            <mrow>
                                              <mstyle displaystyle="true">
                                                <msub>
                                                  <mo>∑</mo>
                                                  <mrow>
                                                    <mi>l</mi>
                                                    <mo>≠</mo>
                                                    <mi>i</mi>
                                                  </mrow>
                                                </msub>
                                                <mrow>
                                                  <msub>
                                                    <mi>a</mi>
                                                    <mrow>
                                                      <mi>i</mi>
                                                      <mi>l</mi>
                                                    </mrow>
                                                  </msub>
                                                </mrow>
                                              </mstyle>
                                            </mrow>
                                            <mo>)</mo>
                                          </mrow>
                                        </mrow>
                                        <mn>2</mn>
                                      </msup>
                                      <mo>−</mo>
                                      <mstyle displaystyle="true">
                                        <msub>
                                          <mo>∑</mo>
                                          <mrow>
                                            <mi>l</mi>
                                            <mo>≠</mo>
                                            <mi>i</mi>
                                          </mrow>
                                        </msub>
                                        <mrow>
                                          <msup>
                                            <mrow>
                                              <mo stretchy="false">(</mo>
                                              <msub>
                                                <mi>a</mi>
                                                <mrow>
                                                  <mi>i</mi>
                                                  <mi>l</mi>
                                                </mrow>
                                              </msub>
                                              <mo stretchy="false">)</mo>
                                            </mrow>
                                            <mn>2</mn>
                                          </msup>
                                        </mrow>
                                      </mstyle>
                                    </mrow>
                                    <mo>}</mo>
                                  </mrow>
                                </mrow>
                              </mfrac>
                              <mo>.</mo>
                            </mrow>
                            <annotation encoding="MathType-MTEF">MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xI8qiVKYPFjYdHaVhbbf9v8qqaqFr0xc9vqFj0dXdbba91qpepeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4qamKaemiBaWMaemyDauNaem4CamNaemiDaqNaemyzauMaemOCaiNaem4qamKaem4Ba8MaemyzauMaemOzay2aaSbaaSqaaiabdMgaPbqabaGccqGH9aqpjuaGdaWcaaqaamaaqababaWaaabeaeaacqWGHbqydaWgaaqaaiabdMgaPjabdYgaSbqabaGaemyyae2aaSbaaeaacqWGSbaBcqWGTbqBaeqaaiabdggaHnaaBaaabaGaemyBa0MaemyAaKgabeaaaeaacqWGTbqBcqGHGjsUcqWGPbqAcqGGSaalcqWGSbaBaeqacqGHris5aaqaaiabdYgaSjabgcMi5kabdMgaPbqabiabggHiLdaabaWaaiWaaeaadaqadaqaamaaqababaGaemyyae2aaSbaaeaacqWGPbqAcqWGSbaBaeqaaaqaaiabdYgaSjabgcMi5kabdMgaPbqabiabggHiLdaacaGLOaGaayzkaaWaaWbaaeqabaGaeGOmaidaaiabgkHiTmaaqababaGaeiikaGIaemyyae2aaSbaaeaacqWGPbqAcqWGSbaBaeqaaiabcMcaPmaaCaaabeqaaiabikdaYaaaaeaacqWGSbaBcqGHGjsUcqWGPbqAaeqacqGHris5aaGaay5Eaiaaw2haaaaacqGGUaGlaaa@77AB@</annotation>
                          </semantics>
                        </mrow></semantics></math></script></div> <div class="EquationNumber">(8)</div></div></div>
                  <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">In unweighted networks, <em xmlns="" class="EmphasisTypeItalic">ClusterCoef</em><sub>
                      <em xmlns="" class="EmphasisTypeItalic">i</em>
                    </sub>equals 1 if and only if all neighbors of gene <em xmlns="" class="EmphasisTypeItalic">i</em> are also linked to each other. For weighted networks, 0 ≤ <em xmlns="" class="EmphasisTypeItalic">a</em><sub>
                      <em xmlns="" class="EmphasisTypeItalic">ij</em>
                    </sub>≤ 1 implies that 0 ≤ <em xmlns="" class="EmphasisTypeItalic">ClusterCoef</em><sub>
                      <em xmlns="" class="EmphasisTypeItalic">i</em>
                    </sub>≤ 1 [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR5" data-toggle="citation">5</a></span>]. The mean clustering coefficient has been used to measure the extent of module structure present in a network [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR26" data-toggle="citation">26</a></span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR34" data-toggle="citation">34</a></span>].</p>
                </section>
              </section><section id="Sec14" class="Section2 RenderAsSection2">
                <h3 class="Heading">Category 5: Functions for simulating microarray data with modular structure</h3>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Simple yet sufficiently realistic simulated data is often important for evaluation of novel data mining methods. The WGCNA package includes simulation functions simulateDatExpr, simulateMultiExpr, simulateDatExpr5Modules that result in expression data sets with a customizable modular (cluster) structure. The user can choose the modular structure by specifying a set of seed eigengenes, one for each module, around which each module is built. Module genes are simulated to exhibit progressively lower correlations with the seed which leads to genes with progressively lower intramodular connectivity. The user can specify module sizes and the number of background genes, i.e. genes outside of the modules. The seed eigengenes can be simulated to reflect dependence relationships between the modules (function simulateEigengeneNetwork).</p>
              </section><section id="Sec15" class="Section2 RenderAsSection2">
                <h3 class="Heading">Category 6: Visualization functions</h3>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Module structure and network connections in the expression data can be visualized in several different ways. For example, the co-expression module structure can be visualized by heatmap plots of gene-gene connectivity that can be produced using the function TOMplot. Examples are presented in Figures <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig2">2C</a></span> and <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig4">4B</a></span>. An alternative is a multi-dimensional scaling plot; an example is presented in Figure <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig2">2B</a></span>. Relationships among modules can be summarized by a hierarchical clustering dendrogram of their eigengenes, or by a heatmap plot of the corresponding eigengene network (function labeledHeatmap), illustrated in Figures <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig3">3C, D</a></span>, and <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig4">4C, D</a></span>. The package includes several additional functions designed to aid the user in visualizing input data and results. These functions rely on basic plotting functions provided in R and the packages sma [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR35" data-toggle="citation">35</a></span>] and fields [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR36" data-toggle="citation">36</a></span>].</p>
              </section><section id="Sec16" class="Section2 RenderAsSection2">
                <h3 class="Heading">Category 7: Functions for interfacing with other software packages</h3>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">To enhance the integration of WGCNA results with other network visualization packages and gene ontology analysis software, we have created several R functions and corresponding tutorials. For example, our R functions exportNetworkToVisANT and exportNetworkToCytoscape allow the user to export networks in a format suitable for VisANT [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR37" data-toggle="citation">37</a></span>] and Cytoscape [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR38" data-toggle="citation">38</a></span>], respectively.</p>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Our online R tutorials also show how to interface WGCNA results with gene ontology packages available directly in R, e.g. GOSim [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR39" data-toggle="citation">39</a></span>]. Many gene ontology based functional enrichment analysis software programs such as David [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR40" data-toggle="citation">40</a></span>], AmiGO [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR41" data-toggle="citation">41</a></span>], Webgestalt [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR42" data-toggle="citation">42</a></span>] simply take lists of gene identifiers as input. Ingenuity Pathway Analysis allows the user to input gene expression data or gene identifiers.</p>
              </section><section id="Sec17" class="Section2 RenderAsSection2">
                <h3 class="Heading">Mouse Data Application</h3>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">As an example of the type of analysis one can perform with WGCNA, we describe a network analysis of liver expression data from female mice. The data and biological findings of this analysis have been described in [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR14" data-toggle="citation">14</a></span>]. Briefly, mRNA levels in female mouse livers were measured by microarrays with over 23,000 probe sets. In addition to the expression data, multiple physiological and metabolic traits were measured. For computational reasons, the original analysis presented in [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR14" data-toggle="citation">14</a></span>] was restricted to 3600 most connected genes, and for simplicity we will work with the same set of genes (although we note that the presented package is capable of handling all genes as well). While we do use the same data, the module detection methods are slightly different and the results are similar but not the same. The code used to perform this analysis is part of the tutorials posted on our webpage.</p>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">The network and the 18 identified modules are depicted in Figures <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig4">4A, B</a></span>. To understand the physiologic significance of the modules, we correlated the 18 module eigengenes with physiological traits such as body weight, cholesterol level, insulin level. The full module-trait correlation table is presented in the accompanying tutorial.</p>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">In the following, we will only consider mouse body weight as sample trait. The module eigengenes of the following three modules were highly correlated with body weight: brown (409 genes, correlation with weight <em xmlns="" class="EmphasisTypeItalic">r</em> = 0.59, correlation p-value <em xmlns="" class="EmphasisTypeItalic">p</em> = 5 × 10<sup>-14</sup>), red (221 genes, <em xmlns="" class="EmphasisTypeItalic">r</em> = 0.51, <em xmlns="" class="EmphasisTypeItalic">p</em> = 3 × 10<sup>-10</sup>), and salmon (91 genes, <em xmlns="" class="EmphasisTypeItalic">r</em> = 0.43, <em xmlns="" class="EmphasisTypeItalic">p</em> = 2 × 10<sup>-7</sup>).</p>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">We used the online software David [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR40" data-toggle="citation">40</a></span>] to determine whether the three body weight related modules were significantly enriched with regard to known gene ontologies. The brown module is significantly enriched in categories "glycoprotein" (<em xmlns="" class="EmphasisTypeItalic">p</em> = 2 × 10<sup>-24</sup>, Benjamini corrected) and "signal" (<em xmlns="" class="EmphasisTypeItalic">p</em> = 1 × 10<sup>-22</sup>). The red module is enriched in "cell cycle" (<em xmlns="" class="EmphasisTypeItalic">p</em> = 9 × 10<sup>-24</sup>) and "chromosome" (<em xmlns="" class="EmphasisTypeItalic">p</em> = 5 × 10<sup>-20</sup>). The salmon module is most significantly enriched in the category "lipid synthesis" (<em xmlns="" class="EmphasisTypeItalic">p</em> = 1 × 10<sup>-16</sup>). Overall, the high enrichment scores suggest that these modules are indeed biologically meaningful.</p>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">To study the relationships between modules, we correlate their eigengenes. In general, relationships between modules can be studied by using correlation networks between eigengenes (i.e. the nodes correspond to eigengenes). In these meta-networks between modules, the adjacency between modules reflects the correlation between the module eigengenes, and modules of eigengenes are referred to as meta-modules [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR12" data-toggle="citation">12</a></span>]. A sample trait such as body weight can be incorporated as an additional node of the eigengene network. The adjacency between the sample trait and an eigengene is sometimes referred to as the eigengene significance [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR11" data-toggle="citation">11</a></span>]. Figures <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig4">4C, D</a></span> depict the eigengene network using a dendrogram (hierarchical cluster tree) and a heatmap plot. We find that eigengenes may exhibit highly significant correlations, e.g. the red and brown modules are highly correlated. Groups of correlated eigengenes corresponds to meta-modules and are recognizable as branches of the eigengene dendrogram, and as reddish squares along the diagonal of the heatmap plot. Figures <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig4">4C</a></span> indicates that there are four meta-modules (branches). Body weight falls within the meta-module grouping together the blue, brown, red, salmon, and yellow modules. In practice, it is difficult to determine whether the modules underlying a meta-module are truly distinct or whether they should be merged. Sometimes gene ontology information can provide some clues.</p>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">It is interesting to find centrally located intramodular hub genes in the body weight related modules since their expression profile represents that of the entire module [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR11" data-toggle="citation">11</a></span>]. To find intramodular hub genes, one can use the module membership measure <em xmlns="" class="EmphasisTypeItalic">K</em>, Equation <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Equ6">6</a></span>. Figure <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig4">4E</a></span> shows a scatterplot between the body weight based gene significance measure <em xmlns="" class="EmphasisTypeItalic">GS</em><sub>
                    <em xmlns="" class="EmphasisTypeItalic">i</em>
                  </sub>, Equation <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Equ2">2</a></span> and module membership in the brown module.</p>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">The high correlation between gene significance and module membership implies that hubgenes in the brown module also tend to be highly correlated with body weight. This suggests that both gene significance and module membership (intramodular connectivity) can be combined in a systems biologic screening method for finding body weight related genes [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR15" data-toggle="citation">15</a></span>]. Figure <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig4">4F</a></span> shows a Visant plot among the most connected genes in the brown module. This brief description illustrates how WGCNA can lead to testable hypotheses that require validation in independent data sets. A tutorial underlying this example and Figure <span class="InternalRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Fig4">4</a></span> can be found on our webpage.</p>
              </section><section id="Sec18" class="Section2 RenderAsSection2">
                <h3 class="Heading">Tutorials</h3>
                <p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">We provide a comprehensive set of online tutorials that guide the user through major steps of correlation network analysis. The tutorials provide R code the user can copy-and-paste into an R session, along with comments and explanations of both the input and output. The code is organized into short sections, each of which addresses a particular task. In particular, the tutorials cover the following topics: correlation network construction, step-by-step and automatic module detection, consensus module detection, eigengene network analysis, differential network analysis, interfacing with external software packages, and data simulation. The tutorials use both simulated and real gene expression data sets.</p>
              </section></div></section>
            <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="Sec19" class="Section1 RenderAsSection1 c-section"><h2 class="Heading js-ToggleCollapseSection c-section__heading" data-sticky-update="true"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Discussion" data-toggle="collapse" data-toggle-disable-clickoff="" data-target="Discussion">Discussion</a></h2><div class="js-CollapseSection CollapseSection c-section__content" id="Discussion" data-test="collapse-section"><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">The WGCNA package complements other network related packages for R, such as the general network structures in Bioconductor [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR6" data-toggle="citation">6</a></span>], gene network enrichment analysis [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR43" data-toggle="citation">43</a></span>], functional analysis of gene co-expression networks [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR44" data-toggle="citation">44</a></span>], and others. While most of the existing packages focus only on unweighted networks, WGCNA implements methods for both weighted and unweighted correlation networks. WGCNA can be used as a data exploratory tool or as a gene screening (ranking) method. For example, WGCNA can be used to explore the module (cluster) structure in a network, to measure the relationships between genes and modules (module membership information), to explore the relationships among modules (eigengene networks), and to rank-order genes or modules (e.g. with regard to their relationship with a sample trait). WGCNA can be used to generate testable hypotheses for validation in independent data sets. For example, WGCNA may suggest that a module (e.g. a putative pathway) is associated with a disease outcome. Since correlation networks are based on correlations between quantitative variables, one can use a correlation test p-value [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR1" data-toggle="citation">1</a></span>] or a regression-based p-value for assessing the statistical significance between pairs of variables. For example, it is straightforward to attach a significance level to the fuzzy module membership measures <span class="InlineEquation" id="IEq7"><span class="MathJax_Preview" style=""></span><span class="MathJax" id="MathJax-Element-13-Frame" tabindex="0" data-mathml="&lt;math xmlns=&quot;http://www.w3.org/1998/Math/MathML&quot;&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;msubsup&gt;&lt;mi&gt;K&lt;/mi&gt;&lt;mrow&gt;&lt;mi&gt;c&lt;/mi&gt;&lt;mi&gt;o&lt;/mi&gt;&lt;mi&gt;r&lt;/mi&gt;&lt;mo&gt;,&lt;/mo&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/mrow&gt;&lt;mrow&gt;&lt;mo stretchy=&quot;false&quot;&gt;(&lt;/mo&gt;&lt;mi&gt;q&lt;/mi&gt;&lt;mo stretchy=&quot;false&quot;&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/msubsup&gt;&lt;/mrow&gt;&lt;annotation encoding=&quot;MathType-MTEF&quot;&gt;          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaaaaa@369C@&lt;/annotation&gt;&lt;/semantics&gt;&lt;/mrow&gt;&lt;/semantics&gt;&lt;/math&gt;" role="presentation" style="position: relative;"><nobr aria-hidden="true"><span class="math" id="MathJax-Span-397" style="width: 2.599em; display: inline-block;"><span style="display: inline-block; position: relative; width: 2.334em; height: 0px; font-size: 111%;"><span style="position: absolute; clip: rect(0.904em, 1002.33em, 2.705em, -999.997em); top: -2.117em; left: 0em;"><span class="mrow" id="MathJax-Span-398"><span class="semantics" id="MathJax-Span-399"><span class="mrow" id="MathJax-Span-400"><span class="semantics" id="MathJax-Span-401"><span class="mrow" id="MathJax-Span-402"><span class="msubsup" id="MathJax-Span-403"><span style="display: inline-block; position: relative; width: 2.334em; height: 0px;"><span style="position: absolute; clip: rect(3.129em, 1000.9em, 4.136em, -999.997em); top: -3.972em; left: 0em;"><span class="mi" id="MathJax-Span-404" style="font-family: MathJax_Math-italic;">K<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.056em;"></span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.288em, 1000.96em, 4.295em, -999.997em); top: -4.502em; left: 0.957em;"><span class="mrow" id="MathJax-Span-405"><span class="mo" id="MathJax-Span-406" style="font-size: 70.7%; font-family: MathJax_Main;">(</span><span class="mi" id="MathJax-Span-407" style="font-size: 70.7%; font-family: MathJax_Math-italic;">q<span style="display: inline-block; overflow: hidden; height: 1px; width: 0.003em;"></span></span><span class="mo" id="MathJax-Span-408" style="font-size: 70.7%; font-family: MathJax_Main;">)</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span><span style="position: absolute; clip: rect(3.341em, 1001.49em, 4.295em, -999.997em); top: -3.654em; left: 0.851em;"><span class="mrow" id="MathJax-Span-409"><span class="mi" id="MathJax-Span-410" style="font-size: 70.7%; font-family: MathJax_Math-italic;">c</span><span class="mi" id="MathJax-Span-411" style="font-size: 70.7%; font-family: MathJax_Math-italic;">o</span><span class="mi" id="MathJax-Span-412" style="font-size: 70.7%; font-family: MathJax_Math-italic;">r</span><span class="mo" id="MathJax-Span-413" style="font-size: 70.7%; font-family: MathJax_Main;">,</span><span class="mi" id="MathJax-Span-414" style="font-size: 70.7%; font-family: MathJax_Math-italic;">i</span></span><span style="display: inline-block; width: 0px; height: 3.977em;"></span></span></span></span></span></span></span></span></span><span style="display: inline-block; width: 0px; height: 2.122em;"></span></span></span><span style="display: inline-block; overflow: hidden; vertical-align: -0.526em; border-left: 0px solid; width: 0px; height: 1.768em;"></span></span></nobr><span class="MJX_Assistive_MathML" role="presentation"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><semantics><mrow><msubsup><mi>K</mi><mrow><mi>c</mi><mi>o</mi><mi>r</mi><mo>,</mo><mi>i</mi></mrow><mrow><mo stretchy="false">(</mo><mi>q</mi><mo stretchy="false">)</mo></mrow></msubsup></mrow><annotation encoding="MathType-MTEF">          MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaaaaa@369C@</annotation></semantics></mrow></semantics></math></span></span><script type="math/mml" id="MathJax-Element-13"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow>
                      <semantics>
                        <mrow>
                          <msubsup>
                            <mi>K</mi>
                            <mrow>
                              <mi>c</mi>
                              <mi>o</mi>
                              <mi>r</mi>
                              <mo>,</mo>
                              <mi>i</mi>
                            </mrow>
                            <mrow>
                              <mo stretchy="false">(</mo>
                              <mi>q</mi>
                              <mo stretchy="false">)</mo>
                            </mrow>
                          </msubsup>
                        </mrow>
                        <annotation encoding="MathType-MTEF">MathType@MTEF@5@5@+=feaagaart1ev2aaatCvAUfKttLearuWrP9MDH5MBPbIqV92AaeXatLxBI9gBaebbnrfifHhDYfgasaacPC6xNi=xH8viVGI8Gi=hEeeu0xXdbba9frFj0xb9qqpG0dXdb9aspeI8k8fiI+fsY=rqGqVepae9pg0db9vqaiVgFr0xfr=xfr=xc9adbaqaaeGaciGaaiaabeqaaeqabiWaaaGcbaGaem4saS0aa0baaSqaaiabdogaJjabd+gaVjabdkhaYjabcYcaSiabdMgaPbqaaiabcIcaOiabdghaXjabcMcaPaaaaaa@369C@</annotation>
                      </semantics>
                    </mrow></semantics></math></script></span>. The relationship between standard microarray data mining techniques and gene co-expression network analysis is discussed in [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR11" data-toggle="citation">11</a></span>].</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Users should be aware of the limitations of the methods implemented in the WGCNA package. First, WGCNA assumes that the microarray data have been properly pre-processed and normalized. To normalize the expression data, several R functions have been implemented in the Bioconductor packages [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR45" data-toggle="citation">45</a></span>]. Although all normalization methods are mathematically compatible with WGCNA, we recommend to use the biologically most meaningful normalization method with respect to the application under consideration. Second, similar to most other data mining methods, the results of WGCNA can be biased or invalid when dealing with technical artefacts, tissue contaminations, or poor experimental design. Third, although several co-expression module detection methods are implemented, the package does not provide means to determine which method is best. While the default hierarchical clustering methods have performed well in several real data applications, it would be desirable to compare these and other methods on multiple real benchmark data sets. Fourth, this package is limited to undirected networks. Methods for orienting edges and constructing directed networks have been presented in the literature, for example in [<span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR46" data-toggle="citation">46</a></span><span>, <span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR47" data-toggle="citation">47</a></span>, </span><span class="CitationRef"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#CR48" data-toggle="citation">48</a></span>].</p></div></section>
            <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="Sec20" class="Section1 RenderAsSection1 c-section"><h2 class="Heading js-ToggleCollapseSection c-section__heading" data-sticky-update="true"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Conclusion" data-toggle="collapse" data-toggle-disable-clickoff="" data-target="Conclusion">Conclusion</a></h2><div class="js-CollapseSection CollapseSection c-section__content" id="Conclusion" data-test="collapse-section"><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">The WGCNA R package provides a comprehensive set of functions for performing weighted correlation network analysis. The WGCNA package can also be used to describe the correlation structure between gene expression profiles, image data, genetic marker data, proteomics data, and other high-dimensional data.</p></div></section>
            <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="Sec21" class="Section1 RenderAsSection1 c-section"><h2 class="Heading js-ToggleCollapseSection c-section__heading" data-sticky-update="true"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Availabilityandrequirements" data-toggle="collapse" data-toggle-disable-clickoff="" data-target="Availabilityandrequirements">Availability and requirements</a></h2><div class="js-CollapseSection CollapseSection c-section__content" id="Availabilityandrequirements" data-test="collapse-section"><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Project name: WGCNA R package</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Project home page: <span class="ExternalRef"><a href="http://www.genetics.ucla.edu/labs/horvath/CoexpressionNetwork/Rpackages/WGCNA" target="_blank" class="ext-link"><span class="RefSource">http://www.genetics.ucla.edu/labs/horvath/CoexpressionNetwork/Rpackages/WGCNA</span></a></span></p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Operating system(s): Platform independent</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Programming language: R</p><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Licence: GNU GPL 3</p></div></section>
            <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" id="Sec22" class="Section1 RenderAsSection1 c-section"><h2 class="Heading js-ToggleCollapseSection c-section__heading" data-sticky-update="true"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Acknowledgements" data-toggle="collapse" data-toggle-disable-clickoff="" data-target="Acknowledgements">Acknowledgements</a></h2><div class="js-CollapseSection CollapseSection c-section__content" id="Acknowledgements" data-test="collapse-section"><p xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">We would like to thank Jun Dong, Tova Fuller, Dan Geschwind, Winden Kellen, Wen Lin, Jake Lusis, Mike Mason, Jeremy Miller, Paul Mischel, Stan Nelson, Mike Oldham, Angela Presson, Atila Van Nas, and Lin Wang for helpful discussions and suggestions. The work was supported in part by grants P50CA092131, 5P30CA016042-28, and NS050151-01.</p></div></section>
          <section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="Section1 RenderAsSection1 c-section" data-test="declarations-section" id="Decs"><h2 class="Heading js-ToggleCollapseSection c-section__heading" data-sticky-update="true"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Declarations" data-toggle="collapse" data-toggle-disable-clickoff="" data-target="Declarations">Declarations</a></h2><div class="js-CollapseSection CollapseSection c-section__content" data-test="collapse-section" id="Declarations"><aside class="Appendix" id="App1">
              
                <h3 class="Heading">Authors' original submitted files for images</h3><div id="widgetContainer" data-test="figshare-container"></div>
                <div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Para">Below are the links to the authors’ original submitted files for images.<div class="MediaObject" id="MOESM1">
                    <div xmlns="" class="esm-item "><div xmlns="http://www.w3.org/1999/xhtml" class="caption-container"><a class="filename" href="https://static-content.springer.com/esm/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_2544_MOESM1_ESM.pdf">12859_2008_2544_MOESM1_ESM.pdf</a>
                          <span xmlns="" class="SimplePara">Authors’ original file for figure 1</span>
                        </div></div>
                  </div><div class="MediaObject" id="MOESM2">
                    <div xmlns="" class="esm-item "><div xmlns="http://www.w3.org/1999/xhtml" class="caption-container"><a class="filename" href="https://static-content.springer.com/esm/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_2544_MOESM2_ESM.pdf">12859_2008_2544_MOESM2_ESM.pdf</a>
                          <span xmlns="" class="SimplePara">Authors’ original file for figure 2</span>
                        </div></div>
                  </div><div class="MediaObject" id="MOESM3">
                    <div xmlns="" class="esm-item "><div xmlns="http://www.w3.org/1999/xhtml" class="caption-container"><a class="filename" href="https://static-content.springer.com/esm/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_2544_MOESM3_ESM.pdf">12859_2008_2544_MOESM3_ESM.pdf</a>
                          <span xmlns="" class="SimplePara">Authors’ original file for figure 3</span>
                        </div></div>
                  </div><div class="MediaObject" id="MOESM4">
                    <div xmlns="" class="esm-item "><div xmlns="http://www.w3.org/1999/xhtml" class="caption-container"><a class="filename" href="https://static-content.springer.com/esm/art%3A10.1186%2F1471-2105-9-559/MediaObjects/12859_2008_2544_MOESM4_ESM.pdf">12859_2008_2544_MOESM4_ESM.pdf</a>
                          <span xmlns="" class="SimplePara">Authors’ original file for figure 4</span>
                        </div></div>
                  </div></div>
              
            </aside><aside xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="ArticleNote ArticleNoteMisc"><h3 xmlns="" class="Heading">Authors' contributions</h3><p xmlns="" class="SimplePara">Both authors jointly developed the methods and wrote the article. PL packaged the functions into an R package. Both authors read and approved the final manuscript.</p></aside></div></section></div>
                        <div class="FulltextWrapper"><section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="Section1 RenderAsSection1 c-section Affiliations" id="Aff"><h2 class="Heading js-ToggleCollapseSection c-section__heading" data-sticky-update="true"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#Author-Affiliations" data-toggle="collapse" data-toggle-disable-clickoff="" data-target="Author-Affiliations">Authors’ Affiliations</a></h2><div class="js-CollapseSection CollapseSection c-section__content c-section__content" data-test="collapse-section" id="Author-Affiliations"><div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Affiliation" id="Aff1"><span class="AffiliationNumber">(1)</span><div class="AffiliationText">Department of Human Genetics, University of California, Los Angeles, CA&nbsp;90095, USA</div></div><div xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="Affiliation" id="Aff2"><span class="AffiliationNumber">(2)</span><div class="AffiliationText">Department of Human Genetics and Department of Biostatistics, University of California, Los Angeles, CA&nbsp;90095, USA</div></div></div></section></div>
                        <div class="FulltextWrapper"><section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="Section1 RenderAsSection1 c-section SectionTypeMaterialsAndMethods" id="Bib1"><h2 class="Heading js-ToggleCollapseSection c-section__heading" data-sticky-update="true"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#References" data-toggle="collapse" data-toggle-disable-clickoff="" data-target="References">References</a></h2><div class="js-CollapseSection CollapseSection c-section__content" data-test="collapse-section" id="References"><ol class="BibliographyWrapper"><li class="Citation"><cite class="CitationContent" id="CR1" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Fisher RA: On the 'probable error' of a coefficient of correlation deduced from a small sample. <em class="EmphasisTypeItalic">Metron</em> 1915, 1: 1–32.</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=On%20the%20%27probable%20error%27%20of%20a%20coefficient%20of%20correlation%20deduced%20from%20a%20small%20sample&amp;author=RA.%20Fisher&amp;journal=Metron&amp;volume=1&amp;pages=1-32&amp;publication_year=1915"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR2" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Zhou X, Kao MC, Wong W: Transitive Functional Annotation by Shortest-path Analysis of Gene Expression Data. <em class="EmphasisTypeItalic">Proc Natl Acad Sci USA</em> 2002, 99(20):12783–12788.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC130537"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=12196633"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Transitive%20Functional%20Annotation%20by%20Shortest-path%20Analysis%20of%20Gene%20Expression%20Data&amp;author=X.%20Zhou&amp;author=MC.%20Kao&amp;author=W.%20Wong&amp;journal=Proc%20Natl%20Acad%20Sci%20USA&amp;volume=99&amp;issue=20&amp;pages=12783-12788&amp;publication_year=2002"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR3" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Steffen M, Petti A, Aach J, D'haeseleer P, Church G: Automated modelling of signal transduction networks. <em class="EmphasisTypeItalic">BMC Bioinformatics</em> 2002, 3: 34.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC137599"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=12413400"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Automated%20modelling%20of%20signal%20transduction%20networks&amp;author=M.%20Steffen&amp;author=A.%20Petti&amp;author=J.%20Aach&amp;author=P.%20D%27haeseleer&amp;author=G.%20Church&amp;journal=BMC%20Bioinformatics&amp;volume=3&amp;pages=34&amp;publication_year=2002"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR4" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Stuart JM, Segal E, Koller D, Kim SK: A Gene-Coexpression Network for Global Discovery of Conserved Genetic Modules. <em class="EmphasisTypeItalic">Science</em> 2003, 302(5643):249–255.</span><span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=12934013"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=A%20Gene-Coexpression%20Network%20for%20Global%20Discovery%20of%20Conserved%20Genetic%20Modules&amp;author=JM.%20Stuart&amp;author=E.%20Segal&amp;author=D.%20Koller&amp;author=SK.%20Kim&amp;journal=Science&amp;volume=302&amp;issue=5643&amp;pages=249-255&amp;publication_year=2003"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR5" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Zhang B, Horvath S: A General Framework for Weighted Gene Co-expression Network Analysis. <em class="EmphasisTypeItalic">Stat Appl Genet Mol Biol</em> 2005, 4: Article 17.</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=A%20General%20Framework%20for%20Weighted%20Gene%20Co-expression%20Network%20Analysis&amp;author=B.%20Zhang&amp;author=S.%20Horvath&amp;journal=Stat%20Appl%20Genet%20Mol%20Biol&amp;volume=4&amp;pages=Article%2017&amp;publication_year=2005"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR6" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Carey VJ, Gentry J, Whalen E, Gentleman R: Network structures and algorithms in Bioconductor. <em class="EmphasisTypeItalic">Bioinformatics</em> 2005, 21: 135–136.</span><span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org//bth458"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=15297301"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Network%20structures%20and%20algorithms%20in%20Bioconductor&amp;author=VJ.%20Carey&amp;author=J.%20Gentry&amp;author=E.%20Whalen&amp;author=R.%20Gentleman&amp;journal=Bioinformatics&amp;volume=21&amp;pages=135-136&amp;publication_year=2005"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR7" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Schaefer J, Strimmer K: An empirical Bayes approach to inferring large-scale gene association networks. <em class="EmphasisTypeItalic">Bioinformatics</em> 2005, 21(6):754–764.</span><span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org//bti062"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=An%20empirical%20Bayes%20approach%20to%20inferring%20large-scale%20gene%20association%20networks&amp;author=J.%20Schaefer&amp;author=K.%20Strimmer&amp;journal=Bioinformatics&amp;volume=21&amp;issue=6&amp;pages=754-764&amp;publication_year=2005"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR8" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Chuang CL, Jen CH, Chen CM, Shieh GS: A pattern recognition approach to infer time-lagged genetic interactions. <em class="EmphasisTypeItalic">Bioinformatics</em> 2008, 24(9):1183–1190.</span><span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org//btn098"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18337258"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=A%20pattern%20recognition%20approach%20to%20infer%20time-lagged%20genetic%20interactions&amp;author=CL.%20Chuang&amp;author=CH.%20Jen&amp;author=CM.%20Chen&amp;author=GS.%20Shieh&amp;journal=Bioinformatics&amp;volume=24&amp;issue=9&amp;pages=1183-1190&amp;publication_year=2008"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR9" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Cokus S, Rose S, Haynor D, Gronbech-Jensen N, Pellegrini M: Modelling the network of cell cycle transcription factors in the yeast Saccharomyces cerevisiae. <em class="EmphasisTypeItalic">BMC Bioinformatics</em> 2006, 7: 381.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1570153"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=16914048"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Modelling%20the%20network%20of%20cell%20cycle%20transcription%20factors%20in%20the%20yeast%20Saccharomyces%20cerevisiae&amp;author=S.%20Cokus&amp;author=S.%20Rose&amp;author=D.%20Haynor&amp;author=N.%20Gronbech-Jensen&amp;author=M.%20Pellegrini&amp;journal=BMC%20Bioinformatics&amp;volume=7&amp;pages=381&amp;publication_year=2006"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR10" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Horvath S, Zhang B, Carlson M, Lu K, Zhu S, Felciano R, Laurance M, Zhao W, Shu Q, Lee Y, Scheck A, Liau L, Wu H, Geschwind D, Febbo P, Kornblum H, Cloughesy T, Nelson S, Mischel P: Analysis of Oncogenic Signaling Networks in Glioblastoma Identifies ASPM as a Novel Molecular Target. <em class="EmphasisTypeItalic">Proc Natl Acad Sci USA</em> 2006, 103(46):17402–17407.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1635024"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17090670"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Analysis%20of%20Oncogenic%20Signaling%20Networks%20in%20Glioblastoma%20Identifies%20ASPM%20as%20a%20Novel%20Molecular%20Target&amp;author=S.%20Horvath&amp;author=B.%20Zhang&amp;author=M.%20Carlson&amp;author=K.%20Lu&amp;author=S.%20Zhu&amp;author=R.%20Felciano&amp;author=M.%20Laurance&amp;author=W.%20Zhao&amp;author=Q.%20Shu&amp;author=Y.%20Lee&amp;author=A.%20Scheck&amp;author=L.%20Liau&amp;author=H.%20Wu&amp;author=D.%20Geschwind&amp;author=P.%20Febbo&amp;author=H.%20Kornblum&amp;author=T.%20Cloughesy&amp;author=S.%20Nelson&amp;author=P.%20Mischel&amp;journal=Proc%20Natl%20Acad%20Sci%20USA&amp;volume=103&amp;issue=46&amp;pages=17402-17407&amp;publication_year=2006"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR11" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Horvath S, Dong J: Geometric interpretation of Gene Co-expression Network Analysis. <em class="EmphasisTypeItalic">PLoS Computational Biology</em> 2008.</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Geometric%20interpretation%20of%20Gene%20Co-expression%20Network%20Analysis&amp;author=S.%20Horvath&amp;author=J.%20Dong&amp;publication_year=2008"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR12" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Langfelder P, Horvath S: Eigengene networks for studying the relationships between co-expression modules. <em class="EmphasisTypeItalic">BMC Systems Biology</em> 2007, 1: 54.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2267703"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18031580"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Eigengene%20networks%20for%20studying%20the%20relationships%20between%20co-expression%20modules&amp;author=P.%20Langfelder&amp;author=S.%20Horvath&amp;journal=BMC%20Systems%20Biology&amp;volume=1&amp;pages=54&amp;publication_year=2007"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR13" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Carlson MR, Zhang B, Fang Z, Horvath S, Mishel PS, Nelson SF: Gene Connectivity, Function, and Sequence Conservation: Predictions from Modular Yeast Co-expression Networks. <em class="EmphasisTypeItalic">BMC Genomics</em> 2006., 7(40):</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Gene%20Connectivity%2C%20Function%2C%20and%20Sequence%20Conservation%3A%20Predictions%20from%20Modular%20Yeast%20Co-expression%20Networks&amp;author=MR.%20Carlson&amp;author=B.%20Zhang&amp;author=Z.%20Fang&amp;author=S.%20Horvath&amp;author=PS.%20Mishel&amp;author=SF.%20Nelson&amp;publication_year=2006"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR14" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Ghazalpour A, Doss S, Zhang B, Plaisier C, Wang S, Schadt E, Thomas A, Drake T, Lusis A, Horvath S: Integrating Genetics and Network Analysis to Characterize Genes Related to Mouse Weight. <em class="EmphasisTypeItalic">PloS Genetics</em> 2006, 2(8):e130.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1550283"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=16934000"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Integrating%20Genetics%20and%20Network%20Analysis%20to%20Characterize%20Genes%20Related%20to%20Mouse%20Weight&amp;author=A.%20Ghazalpour&amp;author=S.%20Doss&amp;author=B.%20Zhang&amp;author=C.%20Plaisier&amp;author=S.%20Wang&amp;author=E.%20Schadt&amp;author=A.%20Thomas&amp;author=T.%20Drake&amp;author=A.%20Lusis&amp;author=S.%20Horvath&amp;journal=PloS%20Genetics&amp;volume=2&amp;issue=8&amp;pages=e130&amp;publication_year=2006"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR15" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Fuller T, Ghazalpour A, Aten J, Drake T, Lusis A, Horvath S: Weighted Gene Co-expression Network Analysis Strategies Applied to Mouse Weight. <em class="EmphasisTypeItalic">Mammalian Genome</em> 2007, 6(18):463–472.</span><span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Weighted%20Gene%20Co-expression%20Network%20Analysis%20Strategies%20Applied%20to%20Mouse%20Weight&amp;author=T.%20Fuller&amp;author=A.%20Ghazalpour&amp;author=J.%20Aten&amp;author=T.%20Drake&amp;author=A.%20Lusis&amp;author=S.%20Horvath&amp;journal=Mammalian%20Genome&amp;volume=6&amp;issue=18&amp;pages=463-472&amp;publication_year=2007"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR16" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Emilsson V, Thorleifsson G, Zhang B, Leonardson A, Zink F, Zhu J, Carlson S, Helgason A, Walters G, Gunnarsdottir S, Mouy M, Steinthorsdottir V, Eiriksdottir G, Bjornsdottir G, Reynisdottir I, Gudbjartsson D, Helgadottir A, Jonasdottir A, Jonasdottir A, Styrkarsdottir U, Gretarsdottir S, Magnusson K, Stefansson H, Fossdal R, Kristjansson K, Gislason H, Stefansson T, Leifsson B, Thorsteinsdottir U, Lamb J, Gulcher MJ, Reitman , Kong A, Schadt E, Stefansson K: Genetics of gene expression and its effect on disease. <em class="EmphasisTypeItalic">Nature</em> 2008, 452(7186):423–8.</span><span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18344981"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Genetics%20of%20gene%20expression%20and%20its%20effect%20on%20disease&amp;author=V.%20Emilsson&amp;author=G.%20Thorleifsson&amp;author=B.%20Zhang&amp;author=A.%20Leonardson&amp;author=F.%20Zink&amp;author=J.%20Zhu&amp;author=S.%20Carlson&amp;author=A.%20Helgason&amp;author=G.%20Walters&amp;author=S.%20Gunnarsdottir&amp;author=M.%20Mouy&amp;author=V.%20Steinthorsdottir&amp;author=G.%20Eiriksdottir&amp;author=G.%20Bjornsdottir&amp;author=I.%20Reynisdottir&amp;author=D.%20Gudbjartsson&amp;author=A.%20Helgadottir&amp;author=A.%20Jonasdottir&amp;author=A.%20Jonasdottir&amp;author=U.%20Styrkarsdottir&amp;author=S.%20Gretarsdottir&amp;author=K.%20Magnusson&amp;author=H.%20Stefansson&amp;author=R.%20Fossdal&amp;author=K.%20Kristjansson&amp;author=H.%20Gislason&amp;author=T.%20Stefansson&amp;author=B.%20Leifsson&amp;author=U.%20Thorsteinsdottir&amp;author=J.%20Lamb&amp;author=MJ.%20Gulcher&amp;author=.%20Reitman&amp;author=A.%20Kong&amp;author=E.%20Schadt&amp;author=K.%20Stefansson&amp;journal=Nature&amp;volume=452&amp;issue=7186&amp;pages=423-8&amp;publication_year=2008"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR17" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">van Nas A, Guhathakurta D, Wang S, Yehya S, Horvath S, Zhang B, Ingram Drake L, Chaudhuri G, Schadt E, Drake T, Arnold A, Lusis A: Elucidating the Role of Gonadal Hormones in Sexually Dimorphic Gene Co-Expression Networks. <em class="EmphasisTypeItalic">Endocrinology</em> 2008.</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Elucidating%20the%20Role%20of%20Gonadal%20Hormones%20in%20Sexually%20Dimorphic%20Gene%20Co-Expression%20Networks&amp;author=A.%20van%20Nas&amp;author=D.%20Guhathakurta&amp;author=S.%20Wang&amp;author=S.%20Yehya&amp;author=S.%20Horvath&amp;author=B.%20Zhang&amp;author=L.%20Ingram%20Drake&amp;author=G.%20Chaudhuri&amp;author=E.%20Schadt&amp;author=T.%20Drake&amp;author=A.%20Arnold&amp;author=A.%20Lusis&amp;publication_year=2008"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR18" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Oldham M, Horvath S, Geschwind D: Conservation and Evolution of Gene Co-expression Networks in Human and Chimpanzee Brains. <em class="EmphasisTypeItalic">Proc Natl Acad Sci USA</em> 2006, 103(47):17973–17978.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1693857"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17101986"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Conservation%20and%20Evolution%20of%20Gene%20Co-expression%20Networks%20in%20Human%20and%20Chimpanzee%20Brains&amp;author=M.%20Oldham&amp;author=S.%20Horvath&amp;author=D.%20Geschwind&amp;journal=Proc%20Natl%20Acad%20Sci%20USA&amp;volume=103&amp;issue=47&amp;pages=17973-17978&amp;publication_year=2006"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR19" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Miller JA, Oldham MC, Geschwind DH: A Systems Level Analysis of Transcriptional Changes in Alzheimer's Disease and Normal Aging. <em class="EmphasisTypeItalic">J Neurosci</em> 2008, 28(6):1410–1420.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2902235"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18256261"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=A%20Systems%20Level%20Analysis%20of%20Transcriptional%20Changes%20in%20Alzheimer%27s%20Disease%20and%20Normal%20Aging&amp;author=JA.%20Miller&amp;author=MC.%20Oldham&amp;author=DH.%20Geschwind&amp;journal=J%20Neurosci&amp;volume=28&amp;issue=6&amp;pages=1410-1420&amp;publication_year=2008"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR20" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Oldham MC, Konopka G, Iwamoto K, Langfelder P, Kato T, Horvath S, Geschwind DH: Functional organization of the transcriptome in human brain. <em class="EmphasisTypeItalic">Nature Neuroscience</em> 2008, 11(11):1271–1282.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2756411"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18849986"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Functional%20organization%20of%20the%20transcriptome%20in%20human%20brain&amp;author=MC.%20Oldham&amp;author=G.%20Konopka&amp;author=K.%20Iwamoto&amp;author=P.%20Langfelder&amp;author=T.%20Kato&amp;author=S.%20Horvath&amp;author=DH.%20Geschwind&amp;journal=Nature%20Neuroscience&amp;volume=11&amp;issue=11&amp;pages=1271-1282&amp;publication_year=2008"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR21" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Keller MP, Choi Y, Wang P, Belt Davis D, Rabaglia ME, Oler AT, Stapleton DS, Argmann C, Schueler KL, Edwards S, Steinberg HA, Chaibub Neto E, Kleinhanz R, Turner S, Hellerstein MK, Schadt EE, Yandell BS, Kendziorski C, Attie AD: A gene expression network model of type 2 diabetes links cell cycle regulation in islets with diabetes susceptibility. <em class="EmphasisTypeItalic">Genome Res</em> 2008, 18(5):706–716.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2336811"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18347327"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=A%20gene%20expression%20network%20model%20of%20type%202%20diabetes%20links%20cell%20cycle%20regulation%20in%20islets%20with%20diabetes%20susceptibility&amp;author=MP.%20Keller&amp;author=Y.%20Choi&amp;author=P.%20Wang&amp;author=D.%20Belt%20Davis&amp;author=ME.%20Rabaglia&amp;author=AT.%20Oler&amp;author=DS.%20Stapleton&amp;author=C.%20Argmann&amp;author=KL.%20Schueler&amp;author=S.%20Edwards&amp;author=HA.%20Steinberg&amp;author=E.%20Chaibub%20Neto&amp;author=R.%20Kleinhanz&amp;author=S.%20Turner&amp;author=MK.%20Hellerstein&amp;author=EE.%20Schadt&amp;author=BS.%20Yandell&amp;author=C.%20Kendziorski&amp;author=AD.%20Attie&amp;journal=Genome%20Res&amp;volume=18&amp;issue=5&amp;pages=706-716&amp;publication_year=2008"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR22" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Presson A, Sobel E, Papp J, Suarez C, Whistler T, Rajeevan M, Vernon S, Horvath S: Integrated weighted gene co-expression network analysis with an application to chronic fatigue syndrome. <em class="EmphasisTypeItalic">BMC Systems Biology</em> 2008., 2(95):</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Integrated%20weighted%20gene%20co-expression%20network%20analysis%20with%20an%20application%20to%20chronic%20fatigue%20syndrome&amp;author=A.%20Presson&amp;author=E.%20Sobel&amp;author=J.%20Papp&amp;author=C.%20Suarez&amp;author=T.%20Whistler&amp;author=M.%20Rajeevan&amp;author=S.%20Vernon&amp;author=S.%20Horvath&amp;publication_year=2008"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR23" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Weston D, Gunter L, Rogers A, Wullschleger S: Connecting genes, coexpression modules, and molecular signatures to environmental stress phenotypes in plants. <em class="EmphasisTypeItalic">BMC Systems Biology</em> 2008., 2:</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Connecting%20genes%2C%20coexpression%20modules%2C%20and%20molecular%20signatures%20to%20environmental%20stress%20phenotypes%20in%20plants&amp;author=D.%20Weston&amp;author=L.%20Gunter&amp;author=A.%20Rogers&amp;author=S.%20Wullschleger&amp;publication_year=2008"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR24" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Wilcox RR: <em class="EmphasisTypeItalic">Introduction to Robust Estimation and Hypothesis Testing</em>. Academic Press; 1997.</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Introduction%20to%20Robust%20Estimation%20and%20Hypothesis%20Testing&amp;author=RR.%20Wilcox&amp;publication_year=1997"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR25" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Yip A, Horvath S: Gene network interconnectedness and the generalized topological overlap measure. <em class="EmphasisTypeItalic">BMC Bioinformatics</em> 2007, 8: 22.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1797055"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17250769"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Gene%20network%20interconnectedness%20and%20the%20generalized%20topological%20overlap%20measure&amp;author=A.%20Yip&amp;author=S.%20Horvath&amp;journal=BMC%20Bioinformatics&amp;volume=8&amp;pages=22&amp;publication_year=2007"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR26" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Ravasz E, Somera A, Mongru D, Oltvai Z, Barabási A: Hierarchical Organization of Modularity in Metabolic Networks. <em class="EmphasisTypeItalic">Science</em> 2002, 297(5586):1551–1555.</span><span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=12202830"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Hierarchical%20Organization%20of%20Modularity%20in%20Metabolic%20Networks&amp;author=E.%20Ravasz&amp;author=A.%20Somera&amp;author=D.%20Mongru&amp;author=Z.%20Oltvai&amp;author=A.%20Barab%C3%A1si&amp;journal=Science&amp;volume=297&amp;issue=5586&amp;pages=1551-1555&amp;publication_year=2002"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR27" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Li A, Horvath S: Network Neighborhood Analysis With the Multi-node Topological Overlap Measure. <em class="EmphasisTypeItalic">Bioinformatics</em> 2007, 23(2):222–231.</span><span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org//btl581"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17110366"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Network%20Neighborhood%20Analysis%20With%20the%20Multi-node%20Topological%20Overlap%20Measure&amp;author=A.%20Li&amp;author=S.%20Horvath&amp;journal=Bioinformatics&amp;volume=23&amp;issue=2&amp;pages=222-231&amp;publication_year=2007"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR28" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Kaufman L, Rousseeuw P: <em class="EmphasisTypeItalic">Finding Groups in Data: An Introduction to Cluster Analysis</em>. New York: John Wiley &amp; Sons, Inc; 1990.</span><span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Finding%20Groups%20in%20Data%3A%20An%20Introduction%20to%20Cluster%20Analysis&amp;author=L.%20Kaufman&amp;author=P.%20Rousseeuw&amp;publication_year=1990"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR29" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Langfelder P, Zhang B, Horvath S: Defining clusters from a hierarchical cluster tree: the Dynamic Tree Cut package for R. <em class="EmphasisTypeItalic">Bioinformatics</em> 2008, 24(5):719–720.</span><span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org//btm563"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18024473"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Defining%20clusters%20from%20a%20hierarchical%20cluster%20tree%3A%20the%20Dynamic%20Tree%20Cut%20package%20for%20R&amp;author=P.%20Langfelder&amp;author=B.%20Zhang&amp;author=S.%20Horvath&amp;journal=Bioinformatics&amp;volume=24&amp;issue=5&amp;pages=719-720&amp;publication_year=2008"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR30" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Dudoit S, Fridlyand J: A prediction-based resampling method for estimating the number of clusters in a dataset. <em class="EmphasisTypeItalic">Genome Biol</em> 2002, 3(7):RESEARCH0036.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC126241"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=12184810"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=A%20prediction-based%20resampling%20method%20for%20estimating%20the%20number%20of%20clusters%20in%20a%20dataset&amp;author=S.%20Dudoit&amp;author=J.%20Fridlyand&amp;journal=Genome%20Biol&amp;volume=3&amp;issue=7&amp;pages=RESEARCH0036&amp;publication_year=2002"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR31" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Hastie T, Tibshirani R, Sherlock G, Eisen M, Brown P, Botstein D: Imputing Missing Data for Gene Expression Arrays. <em class="EmphasisTypeItalic">Technical report, Stanford Statistics Department</em> 1999.</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Imputing%20Missing%20Data%20for%20Gene%20Expression%20Arrays&amp;author=T.%20Hastie&amp;author=R.%20Tibshirani&amp;author=G.%20Sherlock&amp;author=M.%20Eisen&amp;author=P.%20Brown&amp;author=D.%20Botstein&amp;publication_year=1999"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR32" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Troyanskaya O, Cantor M, Sherlock G, Brown P, Hastie T, Tibshirani R, Botstein D, Altman RB: Missing value estimation methods for DNA microarrays. <em class="EmphasisTypeItalic">Bioinformatics</em> 2001, 17(6):520–525.</span><span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org//17.6.520"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=11395428"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Missing%20value%20estimation%20methods%20for%20DNA%20microarrays&amp;author=O.%20Troyanskaya&amp;author=M.%20Cantor&amp;author=G.%20Sherlock&amp;author=P.%20Brown&amp;author=T.%20Hastie&amp;author=R.%20Tibshirani&amp;author=D.%20Botstein&amp;author=RB.%20Altman&amp;journal=Bioinformatics&amp;volume=17&amp;issue=6&amp;pages=520-525&amp;publication_year=2001"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR33" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Dong J, Horvath S: Understanding network concepts in modules. <em class="EmphasisTypeItalic">BMC Systems Biology</em> 2007, 1: 24.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3238286"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17547772"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Understanding%20network%20concepts%20in%20modules&amp;author=J.%20Dong&amp;author=S.%20Horvath&amp;journal=BMC%20Systems%20Biology&amp;volume=1&amp;pages=24&amp;publication_year=2007"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR34" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Watts DJ, Strogatz SH: Collective dynamics of 'small-world' networks. <em class="EmphasisTypeItalic">Nature</em> 1998, 393(6684):440–2.</span><span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=9623998"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Collective%20dynamics%20of%20%27small-world%27%20networks&amp;author=DJ.%20Watts&amp;author=SH.%20Strogatz&amp;journal=Nature&amp;volume=393&amp;issue=6684&amp;pages=440-2&amp;publication_year=1998"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR35" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Dudoit S, Yang Y, Callow M, Speed T: Statistical methods for identifying differentially expressed genes in replicated cDNA microarray experiments. <em class="EmphasisTypeItalic">Statistica Sinica</em> 2002.</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Statistical%20methods%20for%20identifying%20differentially%20expressed%20genes%20in%20replicated%20cDNA%20microarray%20experiments&amp;author=S.%20Dudoit&amp;author=Y.%20Yang&amp;author=M.%20Callow&amp;author=T.%20Speed&amp;publication_year=2002"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR36" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">(2006) FDT: fields: Tools for Spatial Data. <em class="EmphasisTypeItalic">Tech. rep., National Center for Atmospheric Research, Boulder, CO</em> 2007. [<span xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://www.image.ucar.edu/GSP/Software/Fields" target="_blank" class="ext-link"><span class="RefSource">http://www.image.ucar.edu/GSP/Software/Fields</span></a></span>]</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=fields%3A%20Tools%20for%20Spatial%20Data&amp;publication_year=2007"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR37" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Hu Z, Snitkin ES, DeLisi C: VisANT: an integrative framework for networks in systems biology. <em class="EmphasisTypeItalic">Brief Bioinform</em> 2008, 9(4):317–325.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2743399"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org//bbn020"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18463131"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=VisANT%3A%20an%20integrative%20framework%20for%20networks%20in%20systems%20biology&amp;author=Z.%20Hu&amp;author=ES.%20Snitkin&amp;author=C.%20DeLisi&amp;journal=Brief%20Bioinform&amp;volume=9&amp;issue=4&amp;pages=317-325&amp;publication_year=2008"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR38" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Shannon P, Markiel A, Ozier O, Baliga NS, Wang JT, Ramage D, Amin N, Schwikowski B, Ideker T: Cytoscape: A Software Environment for Integrated Models of Biomolecular Interaction Networks. <em class="EmphasisTypeItalic">Genome Research</em> 2003, 13(11):2498–2504.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC403769"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=14597658"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Cytoscape%3A%20A%20Software%20Environment%20for%20Integrated%20Models%20of%20Biomolecular%20Interaction%20Networks&amp;author=P.%20Shannon&amp;author=A.%20Markiel&amp;author=O.%20Ozier&amp;author=NS.%20Baliga&amp;author=JT.%20Wang&amp;author=D.%20Ramage&amp;author=N.%20Amin&amp;author=B.%20Schwikowski&amp;author=T.%20Ideker&amp;journal=Genome%20Research&amp;volume=13&amp;issue=11&amp;pages=2498-2504&amp;publication_year=2003"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR39" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Frohlich H, Speer N, Poustka A, BeiSZbarth T: GOSim – an R-package for computation of information theoretic GO similarities between terms and gene products. <em class="EmphasisTypeItalic">BMC Bioinformatics</em> 2007., 8:</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=GOSim%20%E2%80%93%20an%20R-package%20for%20computation%20of%20information%20theoretic%20GO%20similarities%20between%20terms%20and%20gene%20products&amp;author=H.%20Frohlich&amp;author=N.%20Speer&amp;author=A.%20Poustka&amp;author=T.%20BeiSZbarth&amp;publication_year=2007"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR40" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Dennis G, Sherman B, Hosack D, Yang J, Gao W, Lane H, Lempicki R: DAVID: Database for Annotation, Visualization, and Integrated Discovery. <em class="EmphasisTypeItalic">Genome Biol</em> 2003, 4(5):P3.</span><span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=12734009"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=DAVID%3A%20Database%20for%20Annotation%2C%20Visualization%2C%20and%20Integrated%20Discovery&amp;author=G.%20Dennis&amp;author=B.%20Sherman&amp;author=D.%20Hosack&amp;author=J.%20Yang&amp;author=W.%20Gao&amp;author=H.%20Lane&amp;author=R.%20Lempicki&amp;journal=Genome%20Biol&amp;volume=4&amp;issue=5&amp;pages=P3&amp;publication_year=2003"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR41" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Ashburner M, Ball CA, Blake JA, Botstein D, Butler H, Cherry JM, Davis AP, Dolinski K, Dwight SS, Eppig JT, Harris MA, Hill DP, Issel-Tarver L, Kasarskis A, Lewis S, Matese JC, Richardson JE, Ringwald M, Rubin GM, Sherlock G: Gene Ontology: tool for the unification of biology. <em class="EmphasisTypeItalic">Nat Genet</em> 2000, 25: 25–29.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3037419"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=10802651"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Gene%20Ontology%3A%20tool%20for%20the%20unification%20of%20biology&amp;author=M.%20Ashburner&amp;author=CA.%20Ball&amp;author=JA.%20Blake&amp;author=D.%20Botstein&amp;author=H.%20Butler&amp;author=JM.%20Cherry&amp;author=AP.%20Davis&amp;author=K.%20Dolinski&amp;author=SS.%20Dwight&amp;author=JT.%20Eppig&amp;author=MA.%20Harris&amp;author=DP.%20Hill&amp;author=L.%20Issel-Tarver&amp;author=A.%20Kasarskis&amp;author=S.%20Lewis&amp;author=JC.%20Matese&amp;author=JE.%20Richardson&amp;author=M.%20Ringwald&amp;author=GM.%20Rubin&amp;author=G.%20Sherlock&amp;journal=Nat%20Genet&amp;volume=25&amp;pages=25-29&amp;publication_year=2000"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR42" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Zhang B, Kirov S, Snoddy J: WebGestalt: an integrated system for exploring gene sets in various biological contexts. <em class="EmphasisTypeItalic">Nucleic Acids Res</em> 2005, 33(Web Server issue):W741-W748.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1160236"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org//gki475"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=15980575"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=WebGestalt%3A%20an%20integrated%20system%20for%20exploring%20gene%20sets%20in%20various%20biological%20contexts&amp;author=B.%20Zhang&amp;author=S.%20Kirov&amp;author=J.%20Snoddy&amp;journal=Nucleic%20Acids%20Res&amp;volume=33&amp;issue=Web%20Server%20issue&amp;pages=W741-W748&amp;publication_year=2005"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR43" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Liu M, Liberzon A, Kong SW, Lai WR, Park PJ, Kohane IS, Kasif S: Network-Based Analysis of Affected Biological Processes in Type 2 Diabetes Models. <em class="EmphasisTypeItalic">PLoS Genet</em> 2007, 3(6):e96.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1904360"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=17571924"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Network-Based%20Analysis%20of%20Affected%20Biological%20Processes%20in%20Type%202%20Diabetes%20Models&amp;author=M.%20Liu&amp;author=A.%20Liberzon&amp;author=SW.%20Kong&amp;author=WR.%20Lai&amp;author=PJ.%20Park&amp;author=IS.%20Kohane&amp;author=S.%20Kasif&amp;journal=PLoS%20Genet&amp;volume=3&amp;issue=6&amp;pages=e96&amp;publication_year=2007"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR44" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Henegar C, Clement K, Zucker JD: Unsupervised Multiple-Instance Learning for Functional Profiling of Genomic Data. In <em class="EmphasisTypeItalic">Machine Learning: ECML 2006</em>. Springer Berlin/Heidelberg; 2006:186–197.</span><span class="Occurrences"><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/_21"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Unsupervised%20Multiple-Instance%20Learning%20for%20Functional%20Profiling%20of%20Genomic%20Data&amp;author=C.%20Henegar&amp;author=K.%20Clement&amp;author=JD.%20Zucker&amp;pages=186-197&amp;publication_year=2006"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR45" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Gentleman R, Huber W, Carey V, Irizarry R, Dudoit S: Bioinformatics and Computational Biology Solutions Using R and Bioconductor. In <em class="EmphasisTypeItalic">Book</em>. Springer-Verlag New York; 2005.</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Bioinformatics%20and%20Computational%20Biology%20Solutions%20Using%20R%20and%20Bioconductor&amp;author=R.%20Gentleman&amp;author=W.%20Huber&amp;author=V.%20Carey&amp;author=R.%20Irizarry&amp;author=S.%20Dudoit&amp;publication_year=2005"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR46" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Opgen-Rhein R, Strimmer K: From correlation to causation networks: a simple approximate learning algorithm and its application to high-dimensional plant gene expression data. <em class="EmphasisTypeItalic">BMC Systems Biology</em> 2007., 1:</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=From%20correlation%20to%20causation%20networks%3A%20a%20simple%20approximate%20learning%20algorithm%20and%20its%20application%20to%20high-dimensional%20plant%20gene%20expression%20data&amp;author=R.%20Opgen-Rhein&amp;author=K.%20Strimmer&amp;publication_year=2007"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR47" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Aten J, Fuller T, Lusis A, Horvath S: Using genetic markers to orient the edges in quantitative trait networks: The NEO software. <em class="EmphasisTypeItalic">BMC Systems Biology</em> 2008., 2:</span><span class="Occurrences"><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Using%20genetic%20markers%20to%20orient%20the%20edges%20in%20quantitative%20trait%20networks%3A%20The%20NEO%20software&amp;author=J.%20Aten&amp;author=T.%20Fuller&amp;author=A.%20Lusis&amp;author=S.%20Horvath&amp;publication_year=2008"><span><span>Google Scholar</span></span></a></span></span></cite></li><li class="Citation"><cite class="CitationContent" id="CR48" itemprop="citation" itemscope="" itemtype="http://schema.org/ScholarlyArticle"><span itemprop="headline">Chaibub Neto E, Ferrara CT, Attie AD, Yandell BS: Inferring Causal Phenotype Networks From Segregating Populations. <em class="EmphasisTypeItalic">Genetics</em> 2008, 179(2):1089–1100.</span><span class="Occurrences"><span class="Occurrence OccurrencePMCID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2429862"><span><span>PubMed Central</span></span></a></span><span class="Occurrence OccurrenceDOI"><a target="_blank" rel="noopener" href="https://doi.org/"><span><span>View Article</span></span></a></span><span class="Occurrence OccurrencePID"><a target="_blank" rel="noopener nofollow" href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=18505877"><span><span>PubMed</span></span></a></span><span class="Occurrence OccurrenceGS"><a target="_blank" rel="noopener" href="http://scholar.google.com/scholar_lookup?title=Inferring%20Causal%20Phenotype%20Networks%20From%20Segregating%20Populations&amp;author=E.%20Chaibub%20Neto&amp;author=CT.%20Ferrara&amp;author=AD.%20Attie&amp;author=BS.%20Yandell&amp;journal=Genetics&amp;volume=179&amp;issue=2&amp;pages=1089-1100&amp;publication_year=2008"><span><span>Google Scholar</span></span></a></span></span></cite></li></ol></div></section><section xmlns="" xmlns:fn="http://www.w3.org/2005/xpath-functions" xmlns:meta="http://www.springer.com/app/meta" class="Section1 RenderAsSection1 c-section" id="CopyrightMessage"><h2 class="Heading js-ToggleCollapseSection c-section__heading" data-sticky-update="true"><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559#LangfelderandHorvathlicenseeBioMedCentralLtd" data-toggle="collapse" data-toggle-disable-clickoff="" data-target="LangfelderandHorvathlicenseeBioMedCentralLtd">Copyright</a></h2><div class="js-CollapseSection CollapseSection c-section__content" data-test="collapse-section" id="LangfelderandHorvathlicenseeBioMedCentralLtd"><div xmlns="http://www.w3.org/1999/xhtml" class="ArticleCopyright">©&nbsp;Langfelder and Horvath; licensee BioMed Central Ltd.&nbsp;2008<div class="CopyrightComment">
                <p xmlns="" class="SimplePara">This article is published under license to BioMed Central Ltd. This is an Open Access article distributed under the terms of the Creative Commons Attribution License (<span xmlns:func="http://oscar.fig.bmc.com" xmlns="http://www.w3.org/1999/xhtml" class="ExternalRef"><a href="http://creativecommons.org/licenses/by/2.0" target="_blank" class="ext-link"><span class="RefSource">http://creativecommons.org/licenses/by/2.0</span></a></span>), which permits unrestricted use, distribution, and reproduction in any medium, provided the original work is properly cited.</p>
              </div></div></div></section></div>
                        


                    </article>
                </main>

                <div class="c-page-layout__side u-text-sm u-margin-top-md-at-lt-md">
                    <aside>
                        <div class="c-separator">
                            
        <p class="u-margin-bottom-sm">
            <a id="articlePdf" data-test="download-pdf" class="c-button c-button--primary c-button--full-width" target="_blank" rel="noopener" href="https://bmcbioinformatics.biomedcentral.com/track/pdf/10.1186/1471-2105-9-559" data-track="click" data-track-category="PDF Download" data-track-action="Click download article PDF" data-track-label="10.1186/1471-2105-9-559">Download PDF</a>
        </p>
    
                            
                            
    <button type="button" class="c-button c-button--expandable" data-toggle="collapse" data-target="export-citations" data-test="export-citations-button" data-sticky-update="true" aria-controls="export-citations" aria-expanded="false">
        <span class="c-dropdown__toggle">Export citations</span>
    </button>
    <div id="export-citations" class="c-export-citations" data-test="export-citations-content">
        <div class="c-export-citations__section">
            <h4 class="c-export-citations__title">Papers, Zotero, Reference Manager, RefWorks (<strong>.RIS</strong>)</h4>
            <ul class="c-export-citations__list" id="articleReferences">
                <li><a data-track="click" data-track-category="Article Page" data-track-action="Download Papers, Zotero, Reference Manager, RefWorks article citations" data-track-label="10.1186/1471-2105-9-559" href="http://citation-needed.springer.com/v2/references/10.1186/1471-2105-9-559?format=refman&amp;flavour=citation">
                    <svg class="c-export-citations__icon" width="12" height="12" aria-hidden="true">
                        <use xlink:href="#icon-download"></use>
                    </svg>
                    <span class="u-visually-hidden">Download </span>Citations
                </a></li>
                <li><a data-track="click" data-track-category="Article Page" data-track-action="Download Papers, Zotero, Reference Manager, RefWorks references" data-track-label="10.1186/1471-2105-9-559" href="http://citation-needed.springer.com/v2/references/10.1186/1471-2105-9-559?format=refman&amp;flavour=references">
                    <svg class="c-export-citations__icon" width="12" height="12" aria-hidden="true">
                        <use xlink:href="#icon-download"></use>
                    </svg>
                    <span class="u-visually-hidden">Download </span>References
                </a></li>
                <li><a data-track="click" data-track-category="Article Page" data-track-action="Download Papers, Zotero, Reference Manager, RefWorks citations and references" data-track-label="10.1186/1471-2105-9-559" href="http://citation-needed.springer.com/v2/references/10.1186/1471-2105-9-559?format=refman&amp;flavour=full">
                    <svg class="c-export-citations__icon" width="12" height="12" aria-hidden="true">
                        <use xlink:href="#icon-download"></use>
                    </svg>
                    <span class="u-visually-hidden">Download </span>Both
                </a></li>
            </ul>
        </div>
        <div class="c-export-citations__section">
            <h4 class="c-export-citations__title">EndNote (<strong>.ENW</strong>)</h4>
            <ul class="c-export-citations__list">
                <li><a data-track="click" data-track-category="Article Page" data-track-action="Download EndNote article citations" data-track-label="10.1186/1471-2105-9-559" href="http://citation-needed.springer.com/v2/references/10.1186/1471-2105-9-559?format=endnote&amp;flavour=citation">
                    <svg class="c-export-citations__icon" width="12" height="12" aria-hidden="true">
                        <use xlink:href="#icon-download"></use>
                    </svg>
                    <span class="u-visually-hidden">Download </span>Citations
                </a></li>
                <li><a data-track="click" data-track-category="Article Page" data-track-action="Download EndNote references" data-track-label="10.1186/1471-2105-9-559" href="http://citation-needed.springer.com/v2/references/10.1186/1471-2105-9-559?format=endnote&amp;flavour=references">
                    <svg class="c-export-citations__icon" width="12" height="12" aria-hidden="true">
                        <use xlink:href="#icon-download"></use>
                    </svg>
                    <span class="u-visually-hidden">Download </span>References
                </a></li>
                <li><a data-track="click" data-track-category="Article Page" data-track-action="Download EndNote citations and references" data-track-label="10.1186/1471-2105-9-559" href="http://citation-needed.springer.com/v2/references/10.1186/1471-2105-9-559?format=endnote&amp;flavour=full">
                    <svg class="c-export-citations__icon" width="12" height="12" aria-hidden="true">
                        <use xlink:href="#icon-download"></use>
                    </svg>
                    <span class="u-visually-hidden">Download </span>Both
                </a></li>
            </ul>
        </div>
        <div class="c-export-citations__section">
            <h4 class="c-export-citations__title">Mendeley, JabRef (<strong>.BIB</strong>)</h4>
            <ul class="c-export-citations__list">
                <li><a data-track="click" data-track-category="Article Page" data-track-action="Download Mendeley, JabRef article citations" data-track-label="10.1186/1471-2105-9-559" href="http://citation-needed.springer.com/v2/references/10.1186/1471-2105-9-559?format=bibtex&amp;flavour=citation">
                    <svg class="c-export-citations__icon" width="12" height="12" aria-hidden="true">
                        <use xlink:href="#icon-download"></use>
                    </svg>
                    <span class="u-visually-hidden">Download </span>Citations
                </a></li>
                <li><a data-track="click" data-track-category="Article Page" data-track-action="Download Mendeley, JabRef references" data-track-label="10.1186/1471-2105-9-559" href="http://citation-needed.springer.com/v2/references/10.1186/1471-2105-9-559?format=bibtex&amp;flavour=references">
                    <svg class="c-export-citations__icon" width="12" height="12" aria-hidden="true">
                        <use xlink:href="#icon-download"></use>
                    </svg>
                    <span class="u-visually-hidden">Download </span>References
                </a></li>
                <li><a data-track="click" data-track-category="Article Page" data-track-action="Download Mendeley, JabRef citations and references" data-track-label="10.1186/1471-2105-9-559" href="http://citation-needed.springer.com/v2/references/10.1186/1471-2105-9-559?format=bibtex&amp;flavour=full">
                    <svg class="c-export-citations__icon" width="12" height="12" aria-hidden="true">
                        <use xlink:href="#icon-download"></use>
                    </svg>
                    <span class="u-visually-hidden">Download </span>Both
                </a></li>
            </ul>
        </div>
    </div>

                        </div>
                        
    
 
                        
    
 
                        <div class="c-separator">
                            <h3 class="p-section-title">Metrics</h3>
                            <ul class="c-list-group c-list-group--md">
                                
            <li class="c-list-group__item">
                Article accesses:
                
                    162902
                
            </li>
        
                                
            <li class="c-list-group__item">
                Citations: 2566 <a href="http://citations.springer.com/item?doi=10.1186/1471-2105-9-559" rel="noopener" target="_blank">more information</a>
            </li>
        
                                
    <li class="c-list-group__item">
        <p>Altmetric Attention Score: 16</p>
        <div class="altmetric">
            <a class="altmetric__link" href="http://www.altmetric.com/details.php?citation_id=132979&amp;domain=www.biomedcentral.com" target="_blank" rel="noopener" data-track="click" data-track-category="Article Page" data-track-action="Click Altmetric" data-track-label="10.1186/1471-2105-9-559">
                <img class="altmetric__img" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/saved_resource" alt="Altmetric Attention Score">
            </a>
        </div>
    </li>

                            </ul>
                        </div>
                        
    <div class="c-separator">
        <div class="js-shown">
            <h3 class="p-section-title">Share this article</h3>
            <ul data-test="social-media-share-buttons" class="c-social-links">
                <li class="c-social-links__item">
                    <button id="shareOnTwitter" class="js-btnShareOnTwitter" data-track="click" data-track-category="Article Page" data-track-action="Share Twitter" data-track-label="10.1186/1471-2105-9-559">
                        <span class="u-visually-hidden" data-track="click" data-track-category="Article Page" data-track-action="Share Twitter" data-track-label="10.1186/1471-2105-9-559">Share on Twitter</span>
                        <svg class="c-icon c-social-links__icon" width="24" height="24" aria-hidden="true" data-track="click" data-track-category="Article Page" data-track-action="Share Twitter" data-track-label="10.1186/1471-2105-9-559">
                            <use xlink:href="#icon-twitter"></use>
                        </svg>
                    </button>
                </li>
                <li class="c-social-links__item">
                    <button id="shareOnFacebook" class="js-btnShareOnFacebook" data-track="click" data-track-category="Article Page" data-track-action="Share Facebook" data-track-label="10.1186/1471-2105-9-559">
                        <span class="u-visually-hidden" data-track="click" data-track-category="Article Page" data-track-action="Share Facebook" data-track-label="10.1186/1471-2105-9-559">Share on Facebook</span>
                        <svg class="c-icon c-social-links__icon" width="24" height="24" aria-hidden="true" data-track="click" data-track-category="Article Page" data-track-action="Share Facebook" data-track-label="10.1186/1471-2105-9-559">
                            <use xlink:href="#icon-facebook"></use>
                        </svg>
                    </button>
                </li>
                <li class="c-social-links__item">
                    <button id="shareOnLinkedIn" class="js-btnShareOnLinkedIn" data-track="click" data-track-category="Article Page" data-track-action="Share LinkedIn" data-track-label="10.1186/1471-2105-9-559">
                        <span class="u-visually-hidden" data-track="click" data-track-category="Article Page" data-track-action="Share LinkedIn" data-track-label="10.1186/1471-2105-9-559">Share on LinkedIn</span>
                        <svg class="c-icon c-social-links__icon" width="24" height="24" aria-hidden="true" data-track="click" data-track-category="Article Page" data-track-action="Share LinkedIn" data-track-label="10.1186/1471-2105-9-559">
                            <use xlink:href="#icon-linkedin"></use>
                        </svg>
                    </button>
                </li>
                <li class="c-social-links__item">
                    <button id="shareOnWeibo" class="js-btnShareOnWeibo" data-track="click" data-track-category="Article Page" data-track-action="Share Weibo" data-track-label="10.1186/1471-2105-9-559">
                        <span class="u-visually-hidden" data-track="click" data-track-category="Article Page" data-track-action="Share Weibo" data-track-label="10.1186/1471-2105-9-559">Share on Weibo</span>
                        <svg class="c-icon c-social-links__icon" width="24" height="24" aria-hidden="true" data-track="click" data-track-category="Article Page" data-track-action="Share Weibo" data-track-label="10.1186/1471-2105-9-559">
                            <use xlink:href="#icon-weibo"></use>
                        </svg>
                    </button>
                </li>
                <li class="c-social-links__item">
                    <button id="shareOnReddit" class="js-btnShareOnReddit" data-track="click" data-track-category="Article Page" data-track-action="Share Reddit" data-track-label="10.1186/1471-2105-9-559">
                        <span class="u-visually-hidden" data-track="click" data-track-category="Article Page" data-track-action="Share Reddit" data-track-label="10.1186/1471-2105-9-559">Share on Reddit</span>
                        <svg class="c-icon c-social-links__icon" width="24" height="24" aria-hidden="true" data-track="click" data-track-category="Article Page" data-track-action="Share Reddit" data-track-label="10.1186/1471-2105-9-559">
                            <use xlink:href="#icon-reddit"></use>
                        </svg>
                    </button>
                </li>
            </ul>
        </div>
        
            <div data-sharedit="container">
                <a class="c-button u-margin-top-sm" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-9-559/sharedit" rel="nofollow" data-sharedit="button" data-track="click" data-track-category="sharedit" data-track-action="click article sharedit button" data-track-label="10.1186/1471-2105-9-559">Get shareable link</a>
            </div>
        
    </div>
 
                        
    
 
                        <div class="c-separator">
                            <h3 class="p-section-title">Other Actions</h3>
                            <ul class="c-list-group c-list-group--bordered c-list-group c-list-group--md">
                                <li class="c-list-group__item">
                                    
    <a data-track="click" data-track-category="Article Page" data-track-action="Order article reprints" data-track-label="10.1186/1471-2105-9-559" href="https://www.odysseypress.com/onlinehost/reprint_order.php?type=A&amp;page=0&amp;journal=738&amp;doi=10.1186%2F1471-2105-9-559&amp;volume=9&amp;issue=1&amp;title=WGCNA%3A+an+R+package+for+weighted+correlation+network+analysis&amp;author_name=Peter+Langfelder&amp;start_page=1&amp;end_page=13">Order reprint</a>

                                </li>
                            </ul>
                        </div>
                    </aside>
                    
    <div id="FulltextAd" class="adsbox c-ad c-ad--skyscraper">
        <div class="c-ad__inner" data-component="sticky" data-test="fulltext-ad">
            <p class="c-ad__label">Advertisement</p>
            <div class="c-ad__content">
                <span id="pbgrd-43415139"></span>
            </div>
        </div>
    </div>

                </div>
            </div>
        </div>


        
            <div class="c-journal-footer">
                <div class="c-journal-footer__inner">
                    <div class="c-journal-footer__summary">
                        <h4 class="c-journal-title c-journal-title--footer">
                            
                            <span class="c-journal-title__text">BMC Bioinformatics</span>
                        </h4>
                         <p class="c-journal-footer__issn">ISSN: 1471-2105</p>
                    </div>
                    
                        <div class="c-journal-footer__contact">
                            <h4 class="c-journal-footer__contact-title ">Contact us</h4>
                            <ul class="c-journal-footer__contact-list">
                                
                                    <li class="c-journal-footer__contact-item">Submission enquiries: <a href="mailto:bmcbioinformatics@biomedcentral.com">bmcbioinformatics@biomedcentral.com</a></li>
                                
                                
                                    <li class="c-journal-footer__contact-item">General enquiries: <a href="mailto:info@biomedcentral.com">info@biomedcentral.com</a></li>
                                
                            </ul>
                        </div>
                    
                </div>
            </div>
        
        
    <img rel="nofollow" class="tracker" style="display:none" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/1471-2105-9-559" alt="">

         
    <footer>
        
            <div class="c-publisher-footer" data-test="publisher-footer">
    <div class="u-container">
        
        <div class="c-flex c-flex--wrap c-flex--space-between" data-test="publisher-footer-menu">
            <div class="c-flex">
                
                    
                        <ul class="c-list-group c-list-group--sm u-margin-right-lg u-margin-bottom-md">
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="http://blogs.biomedcentral.com/">Read more on our blogs</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://www.biomedcentral.com/login">Receive BMC newsletters</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://www.biomedcentral.com/account">Manage article alerts</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://authorservices.springernature.com/go/10BMC">Language editing for authors</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="http://authorservices.springernature.com/scientific-editing/">Scientific editing for authors</a>
                                </li>
                            
                        </ul>
                    
                        <ul class="c-list-group c-list-group--sm u-margin-right-lg u-margin-bottom-md">
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://www.biomedcentral.com/about/policies">Policies</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://www.biomedcentral.com/accessibility">Accessibility</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://www.biomedcentral.com/about/press-centre">Press center</a>
                                </li>
                            
                        </ul>
                    
                        <ul class="c-list-group c-list-group--sm u-margin-right-lg u-margin-bottom-md">
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://www.biomedcentral.com/about/contact-us">Support and Contact</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://biomedcentral.typeform.com/to/VLXboo">Leave feedback</a>
                                </li>
                            
                                <li class="c-list-group__item">
                                    <a class="u-gray-link" href="https://www.biomedcentral.com/about/jobs">Careers</a>
                                </li>
                            
                        </ul>
                    
                
            </div>
            <div class="u-margin-bottom-lg">
                <h3 id="social-menu" class="u-text-sm u-reset-margin u-text-normal">Follow BMC</h3>
                <ul class="c-flex u-reset-list" data-test="footer-social-links">
                    
                        <li class="u-margin-top-xs u-margin-right-xs">
                            <a href="https://twitter.com/biomedcentral" class="u-gray-link">
                                <span class="u-visually-hidden">BMC Twitter page</span>
                                <svg class="c-icon" width="24" height="24" aria-hidden="true">
                                    <use xlink:href="#icon-twitter-bordered"></use>
                                </svg>
                            </a>
                        </li>
                    
                        <li class="u-margin-top-xs u-margin-right-xs">
                            <a href="https://www.facebook.com/BioMedCentral" class="u-gray-link">
                                <span class="u-visually-hidden">BMC Facebook page</span>
                                <svg class="c-icon" width="24" height="24" aria-hidden="true">
                                    <use xlink:href="#icon-facebook-bordered"></use>
                                </svg>
                            </a>
                        </li>
                    
                        <li class="u-margin-top-xs u-margin-right-xs">
                            <a href="http://www.weibo.com/biomedcentral" class="u-gray-link">
                                <span class="u-visually-hidden">BMC Weibo page</span>
                                <svg class="c-icon" width="24" height="24" aria-hidden="true">
                                    <use xlink:href="#icon-weibo-bordered"></use>
                                </svg>
                            </a>
                        </li>
                    
                </ul>
            </div>
        </div>
        <p class="u-reset-margin">
            By using this website, you agree to our
            <a class="u-gray-link" href="https://www.biomedcentral.com/terms-and-conditions">Terms and Conditions</a>,
            <a class="u-gray-link" href="https://www.biomedcentral.com/privacy-statement">Privacy
                statement</a> and
            <a class="u-gray-link" href="https://www.biomedcentral.com/cookies" data-test="cookie-link">Cookies</a> policy.
            
                <a class="optanon-toggle-display u-gray-link" href="javascript:void(0);">Manage the cookies</a> we use in the preference centre.
            
        </p>
    </div>
</div>

        
        <div class="c-sn-footer">
    <div class="u-container">
        <img src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/logo-springernature-8bd4e8d82f.svg" class="c-sn-footer__logo" alt="Springer Nature" itemprop="logo">
        <p data-test="copyright"> © 2019 BioMed Central Ltd unless otherwise stated. Part of
            <a href="https://www.springernature.com/" itemscope="" itemtype="http://schema.org/Organization" itemid="#parentOrganization">Springer Nature</a>.
        </p>
    </div>
</div>

        
    </footer>

    </div>

    
    <script type="text/javascript" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/bmc.js.download" async="async"></script>
    <noscript>
	<img hidden src="https://verify.nature.com/verify/nature.png" border="0" width="0" height="0" style="display: none">
</noscript>

        
<script>
    (function() {
        if (window.config && window.config.mustardcut) {
            var appScript = document.createElement('script');
            
            appScript.src = '/static/js/app-bundle-332105ab27.js';
            
            appScript.async = false;
            document.body.appendChild(appScript);
        }
    })();
</script><script src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/app-bundle-332105ab27.js.download"></script>

    


<script type="text/javascript" id="polyfill-matches">Element.prototype.matches||(Element.prototype.matches=Element.prototype.matchesSelector||Element.prototype.mozMatchesSelector||Element.prototype.msMatchesSelector||Element.prototype.oMatchesSelector||Element.prototype.webkitMatchesSelector||function(a){a=(this.document||this.ownerDocument).querySelectorAll(a);for(var b=a.length;0<=--b&&a.item(b)!==this;);return-1<b});</script><script type="text/javascript" id="gpt-control" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/gpt.js.download"></script><script type="text/javascript" id="krux-control-bmc">window.Krux||((Krux=function(){Krux.q.push(arguments)}).q=[]);(function(){var a=document.createElement("script");a.type="text/javascript";a.async=!0;a.src=("https:"===location.protocol?"https:":"http:")+"//cdn.krxd.net/controltag/KDqylSLE.js";var b=document.getElementsByTagName("script")[0];b.parentNode.insertBefore(a,b)})();</script><script type="text/javascript" id="krux-interchange-bmc">window.Krux||((Krux=function(){Krux.q.push(arguments)}).q=[]);(function(){function c(a){a="kxmacmillan_"+a;try{var b=window.localStorage}catch(e){b=null}return b?b[a]||"":navigator.cookieEnabled?(b=document.cookie.match(a+"\x3d([^;]*)"))&&unescape(b[1])||"":""}Krux.user=c("user");Krux.segments=c("segs")?c("segs").split(","):[];var a=[];Krux.user&&a.push("kuid\x3d"+Krux.user);for(var d=0;d<Krux.segments.length;d++)a.push("ksg\x3d"+Krux.segments[d]);Krux.dfppKeyValues=a.length?a.join(";")+";":""})();</script><script type="text/javascript" id="krux-consent">var allowed=google_tag_manager["GTM-TDGJHK"].macro(27);allowed=!0===allowed?1:0;Krux("consent:set",{dc:allowed,al:allowed,tg:allowed,cd:!1,sh:!1,re:!1},function(a,b){});</script><script type="text/javascript" id="gpt-retrieve-ads">function splitKeys(a){var d=[],e="",g=[],h=a.split(";");for(a=0;a<h.length;++a){void 0!==k&&(e=k);var b=h[a].split("\x3d");var k=b[0];k!==e&&(0<e.length&&0<d.length&&g.push([e,d]),d=[]);if(2===b.length&&""!==b[0]&&""!==b[1]){var c=b[1].split(",");for(b=0;b<c.length;++b)d.push(c[b])}}0<k.length&&0<d.length&&g.push([k,d]);return g}
function splitSizes(a){var d=[];if(null!==a){var e=0<=a.indexOf("|")?a.split("|"):a.split(",");for(a=0;a<e.length;++a){var g=e[a].split("x");var h=parseInt(g[0],10);var b=parseInt(g[1],10);2===g.length&&!isNaN(h)&&!isNaN(b)&&0<=h&&0<=b&&d.push([h,b])}}return d}function debounce(a,d){var e=null,g=null;return function(){var h=this,b=Number(new Date),k=arguments;e&&b<e+d?(clearTimeout(g),g=setTimeout(function(){e=b;a.apply(h,k)},d)):(e=b,a.apply(h,k))}}
function addScrollEvent(a){window.addEventListener?window.addEventListener("scroll",a,!1):window.attachEvent("onscroll",a)}function removeScrollEvent(a){window.removeEventListener?window.removeEventListener("scroll",a,!1):window.detachEvent("scroll",a)}function getAdContainers(){return document.querySelectorAll("div[data-gpt-unitpath]")}function isLoggedIn(){return 0<=document.cookie.indexOf("OSCAR_SESSION_COOKIE")?"logged\x3dy;":"logged\x3dn;"}
(function(a,d){function e(f){var a="test";a=a.replace(/[\[\]]/g,"\\$\x26");a=new RegExp("[?\x26]"+a+"(\x3d([^\x26#]*)|\x26|#|$)");a=(a=a.exec(window.location.href))?a[2]?decodeURIComponent(a[2].replace(/\+/g," ")):"":null;a=a&&"ads"===a?"adtype\x3dtest;":"";f=f.getAttribute("data-gpt-targeting");var c=document.querySelector("[data-ad-targeting-search-terms]")?"search\x3d"+document.querySelector("[data-ad-targeting-search-terms]").innerText+";":!1;var d=isLoggedIn();a&&-1===f.indexOf(a)&&(f+=a);c&&
(f+=c);return f+d+b}function g(a,c){for(var f=a.length,b=[];f--;)c(a[f],f)&&(b.push(a[f].slot),a.splice(f,1));b.length&&googletag.pubads().refresh(b);return a}function h(f){var c=Math.max(document.documentElement.clientHeight,a.innerHeight||0);return g(f,function(a){a=document.getElementById(a.divId);a=a.getBoundingClientRect();a=a.top-300;return c>a})}var b,k=getAdContainers(),c=[];d&&(b=d.dfppKeyValues?d.dfppKeyValues:"");a.googletag=a.googletag||{};a.googletag.cmd=a.googletag.cmd||[];for(var m=
0;k[m];++m){var l=k[m];splitKeys(e(l));c.push({divId:l.getAttribute("id"),adUnitPath:l.getAttribute("data-gpt-unitpath"),sizeArray:splitSizes(l.getAttribute("data-gpt-sizes")),targeting:splitKeys(e(l))})}googletag.cmd.push(function(){googletag.pubads().setRequestNonPersonalizedAds(google_tag_manager["GTM-TDGJHK"].macro(31)?0:1);googletag.pubads().disableInitialLoad();googletag.enableServices()});googletag.cmd.push(function(){for(var a=0;c[a];++a)try{c[a].slot=googletag.defineSlot(c[a].adUnitPath,c[a].sizeArray,c[a].divId).addService(googletag.pubads());
for(var b=0,d=c[a].targeting.length;b<d;++b)2===c[a].targeting[b].length&&""!==c[a].targeting[b][0]&&""!==c[a].targeting[b][1]&&c[a].slot.setTargeting(c[a].targeting[b][0],c[a].targeting[b][1])}catch(p){console.log("failed to create slot for",c[a])}});googletag.cmd.push(function(){for(var a=0;c[a];++a)googletag.display(c[a].divId)});googletag.cmd.push(function(){c=h(c)});var n=debounce(function(){googletag.cmd.push(function(){c=h(c);c.length||removeScrollEvent(n)})},250);addScrollEvent(n)})(window,
window.Krux);</script><script type="text/javascript" id="ga-data-track-event-listeners">var setupHandler=google_tag_manager["GTM-TDGJHK"].macro(39);setupHandler("click");setupHandler("submit");</script><script type="text/javascript" id="crossmark-script" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/widget.js.download"></script><script type="text/javascript" id="recommended-script" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/entry-point.js.download"></script><iframe src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/saved_resource.html" title="EventTracker" id="event-tracker" height="0" width="0" style="display:none; visibility:hidden" aria-hidden="true" tabindex="-1"></iframe><iframe id="google_osd_static_frame_7244199846195" name="google_osd_static_frame" style="display: none; width: 0px; height: 0px;" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/saved_resource(1).html"></iframe><div style="position: absolute; width: 0px; height: 0px; overflow: hidden; padding: 0px; border: 0px; margin: 0px;"><div id="MathJax_Font_Test" style="position: absolute; visibility: hidden; top: 0px; left: 0px; width: auto; min-width: 0px; max-width: none; padding: 0px; border: 0px; margin: 0px; white-space: nowrap; text-align: left; text-indent: 0px; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; font-size: 40px; font-weight: normal; font-style: normal; font-family: MathJax_Size2, sans-serif;"></div></div><iframe frameborder="0" scrolling="no" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/client.html" name="uptodate-client" id="uptodate-client" title="Personalised recommendations" allowtransparency="true" style="width: 80px; height: 80px; bottom: 0px; right: 0px; z-index: 100000; position: fixed;"></iframe><div class="kxhead" data-id="KDqylSLE" style="display:none !important;"><span class="kxtag kxinvisible" data-id="20636"><script>
// this tag is intentionally blank
</script></span><span class="kxtag kxinvisible" data-id="20637"></span><span class="kxtag kxinvisible" data-id="20638"></span><span class="kxtag kxinvisible" data-id="20651"><script type="text/javascript">Krux('social.init');</script></span><span class="kxtag kxinvisible" data-id="36251"><script>
(function() {
    /* Selective Attribute DataLayer Library Tag */
    var _, allAttr, allowedList, attr, attributes, dataLayerIngester, dataObj,
        isAllowed, keepCase, normalizeName, omitKeys, optimizeNames, pageAttr, prefix,
        toSet, trim, userAttr, value,
        hasProp = {}.hasOwnProperty;
    _ = Krux('require:underscore');
    dataLayerIngester = Krux('require:scrape').ingestDataLayer;

    /* Safe copy of dataLayer object */
    dataObj = Krux('scrape.javascript', 'dataLayer[0]');

    /* String trimming helper function */
    trim = function(attr) {
        return ("" + attr).replace(/^\s+|\s+$/g, '');
    };

    /* Attribute configs */
    pageAttr = _.map('content.article.keywords,content.journal.title,content.category.publishingSegment,content.contentInfo.imprint,content.contentInfo.title,content.category.pmc.primarySubject,content.article.doi,content.article.articleType,content.category.contentType,content.category.contentSubType1,content.contentInfo.author,content.category.subjectType,content.category.subjectSubType1,content.authorization.type,content.attributes.keywords,content.contentInfo.collection[n].collectionName,content.article.journalIssueName,product.productInfo.brand,product.category.productType,product.productInfo.productName,product.productInfo.sku'.split(','), trim);
    userAttr = _.map('user.segment.subscriberMind,user.segment.subscriberSA,user.segment.customer,user.segment.registered,user.segment.appUserSAuser.segment.appUserMind'.split(','), trim);

    /* Create a array of attributes striping any empty strings */
    allAttr = _.without(pageAttr.concat(userAttr), '');

    /* Configuration settings */
    keepCase = 'true' === 'true';
    omitKeys = 'undefined'.split(',');
    prefix = "_";
    var domain = location.host.split('.'); //Assumption that the array can have max length of 4 eg. www.subdomain.domain.com
    if (domain.length === 4 && domain[0] === 'www') {
        prefix = domain[2] + '_';
    } else if (domain.length === 2) {
        prefix = domain[0] + '_';
    } else {
        prefix = domain[1] + '_';
    }


    /* Function to varify if attribute should be used */
    isAllowed = function(value, whitelist) {
        var i, len, str, x;
        str = "" + value;
        if (!((value !== null) && str.length > 0)) {
            return false;
        }
        for (i = 0, len = whitelist.length; i < len; i++) {
            x = whitelist[i];
            if (value.match(x) !== null) {
                return true;
            }
        }
        return false;
    };

    /* Function to standardise the attribute names */
    normalizeName = function(name, optimize) {
        if (optimize === null) {
            optimize = false;
        }
        if (keepCase) {
            return name;
        }
        return ("" + name).replace(/([A-Z])([A-Z]+)/g, function(full, first, rest) {
            if (optimize === true) {
                return "" + first + (rest.toLowerCase());
            }
            return "" + full;
        }).replace(/_*([A-Z])/g, '_$1').replace(/^_/, '').toLowerCase();
    };

    /* Creates a pattern/replacement config for dataLayer tool to clean up names */
    optimizeNames = function(names) {
        var config, i, len, name;
        config = [{
            pattern: /(\.)_/,
            replacement: '$1'
        }];
        for (i = 0, len = names.length; i < len; i++) {
            name = names[i];
            config.push({
                pattern: normalizeName(name),
                replacement: normalizeName(name, true)
            });
        }
        if (!prefix.match(/^_$|null|undefined|false/)) {
            config.push({
                pattern: /((?:page|user)_attr_)/,
                replacement: "$1" + prefix
            });

        }
        return config;
    };

    /* Get a full list of attributes usting the dataLayer tool */
    attributes = dataLayerIngester(dataObj, {
        omitKeys: _.without(omitKeys.concat([/gtm\./, /sessionid/i,
            /\.phpsessid$/i, /\.sid$/i, /\.zenid$/i,
            /\.requestid$/i
        ]), ''),
        omitValues: [/.*@.*(?:\..*)+/, /gtm\./, /^(https?:)?\/\/[^\/]+/,
            /^\/[^\/]+/, /.{255}/
        ],
        caseSensitive: keepCase,
        useFullPath: 'true' === 'true',
        useLastValue: 'false' === 'true',
        customDelimited: [/./],
        altDelimiter: ',',
        userKeys: _.map(userAttr, function(exp) {
            return new RegExp("(^|\\.)" + exp + "$");
        }),
        convertAttrNames: optimizeNames(allAttr)
    });


    /* Only set Attributes in the allowed list */
    allowedList = _.map(allAttr, function(name) {
        return new RegExp("(_attr_|_attr_" + prefix + "|\\.)" + (
            normalizeName(name, true)) + "$");
    });
    toSet = {};
    for (attr in attributes) {
        if (!hasProp.call(attributes, attr)) continue;
        value = attributes[attr];
        if (isAllowed(attr, allowedList)) {
            toSet[attr] = value;
        }
    }
    Krux('set', toSet);

})();
</script></span></div></body><span class="gr__tooltip"><span class="gr__tooltip-content"></span><i class="gr__tooltip-logo"></i><span class="gr__triangle"></span></span><iframe id="kx-proxy-KDqylSLE" src="./WGCNA_ an R package for weighted correlation network analysis _ BMC Bioinformatics _ Full Text_files/proxy.3d2100fd7107262ecb55ce6847f01fa5.html" style="display: none; visibility: hidden; height: 0; width: 0;"></iframe></html>